// @coreui/angular/esm2015/lib/layout/header/header-brand.component.js
import { Component, Input } from '@angular/core';
import { SidebarService } from '../../sidebar/sidebar.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../sidebar/sidebar.service';
import * as ɵngcc2 from '@angular/router';
import * as ɵngcc3 from '@angular/common';

function HeaderBrandComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0);
} }
function HeaderBrandComponent_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0, 1);
} }
const _c0 = ["*", "*"];
export class HeaderBrandComponent {
    constructor(sidebarService) {
        this.sidebarService = sidebarService;
        this.routerLink = '';
        this.minimized = false;
    }
    ngOnInit() {
        this.stateToggleSubscribe();
    }
    ngOnDestroy() {
        this.stateToggleSubscribe(false);
    }
    minimize(force) {
        this.minimized = (force === 'toggle') ? !this.minimized : force;
    }
    stateToggleSubscribe(subscribe = true) {
        if (subscribe) {
            this.stateToggleSubscription = this.sidebarService.sidebarState$.subscribe((state) => {
                if (!state.id) {
                    if ('minimize' in state) {
                        this.minimize(state.minimize);
                    }
                }
            });
        }
        else {
            this.stateToggleSubscription.unsubscribe();
        }
    }
}
HeaderBrandComponent.ɵfac = function HeaderBrandComponent_Factory(t) { return new (t || HeaderBrandComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.SidebarService)); };
HeaderBrandComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: HeaderBrandComponent, selectors: [["c-header-brand"]], inputs: { routerLink: "routerLink", minimized: "minimized" }, ngContentSelectors: _c0, decls: 3, vars: 3, consts: [["role", "button", 1, "c-header-brand", 3, "routerLink"], [3, "ngIf"]], template: function HeaderBrandComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵelementStart(0, "a", 0);
        ɵngcc0.ɵɵtemplate(1, HeaderBrandComponent_ng_template_1_Template, 1, 0, "ng-template", 1);
        ɵngcc0.ɵɵtemplate(2, HeaderBrandComponent_ng_template_2_Template, 1, 0, "ng-template", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("routerLink", ctx.routerLink);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.minimized);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.minimized);
    } }, directives: [ɵngcc2.RouterLinkWithHref, ɵngcc3.NgIf], encapsulation: 2 });
HeaderBrandComponent.ctorParameters = () => [
    { type: SidebarService }
];
HeaderBrandComponent.propDecorators = {
    routerLink: [{ type: Input }],
    minimized: [{ type: Input }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(HeaderBrandComponent, [{
        type: Component,
        args: [{
                selector: 'c-header-brand',
                template: "<a class=\"c-header-brand\" [routerLink]=\"routerLink\" role=\"button\">\r\n  <ng-template [ngIf]=\"minimized\">\r\n    <ng-content></ng-content>\r\n  </ng-template>\r\n  <ng-template [ngIf]=\"!minimized\">\r\n    <ng-content></ng-content>\r\n  </ng-template>\r\n</a>\r\n\r\n"
            }]
    }], function () { return [{ type: ɵngcc1.SidebarService }]; }, { routerLink: [{
            type: Input
        }], minimized: [{
            type: Input
        }] }); })();
