// @coreui/angular/esm2015/lib/layout/header/header.component.js
import { Component, Input, HostBinding } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class HeaderComponent {
    constructor() {
        this.colorScheme = 'light';
        this.fixed = true;
        this.role = 'header';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames(this.ngClass, this.class, {
            'c-header': true,
            'c-header-fixed': this.fixed,
        }, `c-header-${this.colorScheme}`);
        return classes;
    }
}
HeaderComponent.ɵfac = function HeaderComponent_Factory(t) { return new (t || HeaderComponent)(); };
HeaderComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: HeaderComponent, selectors: [["c-header"], ["", "c-header", ""]], hostVars: 3, hostBindings: function HeaderComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("role", ctx.role);
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { colorScheme: "colorScheme", fixed: "fixed", class: "class", ngClass: "ngClass" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function HeaderComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
HeaderComponent.ctorParameters = () => [];
HeaderComponent.propDecorators = {
    colorScheme: [{ type: Input }],
    fixed: [{ type: Input }],
    class: [{ type: Input }],
    ngClass: [{ type: Input }],
    role: [{ type: HostBinding, args: ['attr.role',] }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(HeaderComponent, [{
        type: Component,
        args: [{
                selector: 'c-header, [c-header]',
                template: `<ng-content></ng-content>`,
                styles: [""]
            }]
    }], function () { return []; }, { colorScheme: [{
            type: Input
        }], fixed: [{
            type: Input
        }], role: [{
            type: HostBinding,
            args: ['attr.role']
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], class: [{
            type: Input
        }], ngClass: [{
            type: Input
        }] }); })();
