// @coreui/angular/esm2015/lib/form/form-group/form-group.component.js
import { Component, HostBinding, Input } from '@angular/core';
import { chkBoolProp } from '../../shared/check-bool-prop';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class FormGroupComponent {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = {
            'form-group': true,
            disabled: chkBoolProp(this.disabled)
        };
        return classes;
    }
    ngOnInit() {
    }
}
FormGroupComponent.ɵfac = function FormGroupComponent_Factory(t) { return new (t || FormGroupComponent)(); };
FormGroupComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FormGroupComponent, selectors: [["c-form-group"]], hostVars: 2, hostBindings: function FormGroupComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { inline: "inline", disabled: "disabled" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function FormGroupComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]:not(.row):not(.form-check-inline):not(.custom-control-inline){display:block}"] });
FormGroupComponent.ctorParameters = () => [];
FormGroupComponent.propDecorators = {
    inline: [{ type: Input }],
    disabled: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FormGroupComponent, [{
        type: Component,
        args: [{
                selector: 'c-form-group',
                template: "<ng-content></ng-content>\n",
                styles: [":host:not(.row):not(.form-check-inline):not(.custom-control-inline){display:block}"]
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], inline: [{
            type: Input
        }], disabled: [{
            type: Input
        }] }); })();
