// @coreui/angular/esm2015/lib/tabset/tab-content/tab-content.component.js
import { Component, ContentChildren, HostBinding } from '@angular/core';
import { TabPaneComponent } from '../tab-pane/tab-pane.component';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class TabContentComponent {
    constructor() {
        this._hasPanes = false;
        this._tabContent = true;
    }
    set hasPanes(hasPanes) {
        this._hasPanes = hasPanes;
    }
    get hasPanes() {
        return this._hasPanes ? null : true;
    }
    ngOnInit() {
    }
    ngAfterContentChecked() {
        this.hasPanes = this.panes.length > 0;
    }
}
TabContentComponent.ɵfac = function TabContentComponent_Factory(t) { return new (t || TabContentComponent)(); };
TabContentComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TabContentComponent, selectors: [["c-tab-content"]], contentQueries: function TabContentComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, TabPaneComponent, 0);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.panes = _t);
    } }, hostVars: 3, hostBindings: function TabContentComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("hidden", ctx.hasPanes);
        ɵngcc0.ɵɵclassProp("tab-content", ctx._tabContent);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function TabContentComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]{display:block}"] });
TabContentComponent.ctorParameters = () => [];
TabContentComponent.propDecorators = {
    _tabContent: [{ type: HostBinding, args: ['class.tab-content',] }],
    hasPanes: [{ type: HostBinding, args: ['attr.hidden',] }],
    panes: [{ type: ContentChildren, args: [TabPaneComponent,] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TabContentComponent, [{
        type: Component,
        args: [{
                selector: 'c-tab-content',
                template: "<ng-content></ng-content>\n",
                styles: [":host{display:block}"]
            }]
    }], function () { return []; }, { _tabContent: [{
            type: HostBinding,
            args: ['class.tab-content']
        }], hasPanes: [{
            type: HostBinding,
            args: ['attr.hidden']
        }], panes: [{
            type: ContentChildren,
            args: [TabPaneComponent]
        }] }); })();
