// @coreui/angular/esm2015/lib/tabset/tab-list/tab-list.component.js
import { Component, HostBinding, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class TabListComponent {
    constructor() {
        this._navClass = true;
    }
    set variant(variant) {
        this._navTabsClass = variant !== 'pills';
        this._navPillsClass = variant === 'pills';
    }
    set vertical(vertical) {
        this._navFlexColumn = !!vertical;
        this._navH100 = !!vertical;
    }
    set fill(fill) {
        this._navFill = !!fill;
    }
    set justified(justified) {
        this._navJustified = !!justified;
    }
}
TabListComponent.ɵfac = function TabListComponent_Factory(t) { return new (t || TabListComponent)(); };
TabListComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TabListComponent, selectors: [["c-tablist"]], hostVars: 14, hostBindings: function TabListComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("nav", ctx._navClass)("nav-tabs", ctx._navTabsClass)("nav-pills", ctx._navPillsClass)("flex-column", ctx._navFlexColumn)("h-100", ctx._navH100)("nav-fill", ctx._navFill)("nav-justified", ctx._navJustified);
    } }, inputs: { variant: "variant", vertical: "vertical", fill: "fill", justified: "justified" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function TabListComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
TabListComponent.ctorParameters = () => [];
TabListComponent.propDecorators = {
    variant: [{ type: Input }],
    vertical: [{ type: Input }],
    fill: [{ type: Input }],
    justified: [{ type: Input }],
    _navClass: [{ type: HostBinding, args: ['class.nav',] }],
    _navTabsClass: [{ type: HostBinding, args: ['class.nav-tabs',] }],
    _navPillsClass: [{ type: HostBinding, args: ['class.nav-pills',] }],
    _navFill: [{ type: HostBinding, args: ['class.nav-fill',] }],
    _navJustified: [{ type: HostBinding, args: ['class.nav-justified',] }],
    _navFlexColumn: [{ type: HostBinding, args: ['class.flex-column',] }],
    _navH100: [{ type: HostBinding, args: ['class.h-100',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TabListComponent, [{
        type: Component,
        args: [{
                selector: 'c-tablist',
                template: "<ng-content></ng-content>\n",
                styles: [""]
            }]
    }], function () { return []; }, { _navClass: [{
            type: HostBinding,
            args: ['class.nav']
        }], variant: [{
            type: Input
        }], _navTabsClass: [{
            type: HostBinding,
            args: ['class.nav-tabs']
        }], _navPillsClass: [{
            type: HostBinding,
            args: ['class.nav-pills']
        }], vertical: [{
            type: Input
        }], _navFlexColumn: [{
            type: HostBinding,
            args: ['class.flex-column']
        }], _navH100: [{
            type: HostBinding,
            args: ['class.h-100']
        }], fill: [{
            type: Input
        }], _navFill: [{
            type: HostBinding,
            args: ['class.nav-fill']
        }], justified: [{
            type: Input
        }], _navJustified: [{
            type: HostBinding,
            args: ['class.nav-justified']
        }] }); })();
