// @coreui/angular/esm2015/lib/progress/progress.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProgressComponent } from './progress.component';
import { ProgressBarComponent } from './progress-bar.component';
import * as ɵngcc0 from '@angular/core';
export class ProgressModule {
}
ProgressModule.ɵfac = function ProgressModule_Factory(t) { return new (t || ProgressModule)(); };
ProgressModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ProgressModule });
ProgressModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ProgressModule, { declarations: function () { return [ProgressComponent, ProgressBarComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ProgressComponent,
        ProgressBarComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ProgressModule, [{
        type: NgModule,
        args: [{
                declarations: [ProgressComponent, ProgressBarComponent],
                exports: [
                    ProgressComponent,
                    ProgressBarComponent
                ],
                imports: [
                    CommonModule
                ]
            }]
    }], null, null); })();
