// @coreui/angular/esm2015/lib/media/media-body.component.js
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class MediaBodyComponent {
    constructor() {
        this.mediaBodyClass = true;
    }
}
MediaBodyComponent.ɵfac = function MediaBodyComponent_Factory(t) { return new (t || MediaBodyComponent)(); };
MediaBodyComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MediaBodyComponent, selectors: [["c-media-body"]], hostVars: 2, hostBindings: function MediaBodyComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("media-body", ctx.mediaBodyClass);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function MediaBodyComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
MediaBodyComponent.propDecorators = {
    mediaBodyClass: [{ type: HostBinding, args: ['class.media-body',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(MediaBodyComponent, [{
        type: Component,
        args: [{
                selector: 'c-media-body',
                template: "<ng-content></ng-content>\n",
                styles: [""]
            }]
    }], function () { return []; }, { mediaBodyClass: [{
            type: HostBinding,
            args: ['class.media-body']
        }] }); })();
