// @coreui/angular/esm2015/lib/media/media.component.js
import { Component } from '@angular/core';
import { MediaDirective } from './media.directive';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class MediaComponent extends MediaDirective {
    constructor() {
        super();
    }
}
MediaComponent.ɵfac = function MediaComponent_Factory(t) { return new (t || MediaComponent)(); };
MediaComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MediaComponent, selectors: [["c-media"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function MediaComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
MediaComponent.ctorParameters = () => [];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(MediaComponent, [{
        type: Component,
        args: [{
                selector: 'c-media',
                template: "<ng-content></ng-content>\n",
                styles: [""]
            }]
    }], function () { return []; }, null); })();
