// @coreui/angular/esm2015/lib/media/media.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MediaComponent } from './media.component';
import { MediaBodyComponent } from './media-body.component';
import { MediaDirective } from './media.directive';
import * as ɵngcc0 from '@angular/core';
export class MediaModule {
}
MediaModule.ɵfac = function MediaModule_Factory(t) { return new (t || MediaModule)(); };
MediaModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MediaModule });
MediaModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MediaModule, { declarations: function () { return [MediaComponent, MediaBodyComponent, MediaDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [MediaDirective, MediaComponent, MediaBodyComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(MediaModule, [{
        type: NgModule,
        args: [{
                declarations: [MediaComponent, MediaBodyComponent, MediaDirective],
                imports: [CommonModule],
                exports: [MediaDirective, MediaComponent, MediaBodyComponent]
            }]
    }], null, null); })();
