// @coreui/angular/esm2015/lib/chart/chart.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NoZoneBaseChartDirective } from './no-zone-basechart.directive';
import * as ɵngcc0 from '@angular/core';
export class ChartModule {
}
ChartModule.ɵfac = function ChartModule_Factory(t) { return new (t || ChartModule)(); };
ChartModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ChartModule });
ChartModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ChartModule, { declarations: function () { return [NoZoneBaseChartDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NoZoneBaseChartDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ChartModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    NoZoneBaseChartDirective
                ],
                imports: [
                    CommonModule
                ],
                exports: [
                    NoZoneBaseChartDirective
                ]
            }]
    }], null, null); })();
