// @coreui/angular/esm2015/lib/chart/no-zone-basechart.directive.js
import { BaseChartDirective, ThemeService } from 'ng2-charts';
import { Directive, ElementRef, NgZone } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from 'ng2-charts';
export class NoZoneBaseChartDirective extends BaseChartDirective {
    constructor(element, themeService, zone) {
        super(element, themeService);
        this.zone = zone;
        const refresh = this.refresh;
        this.refresh = () => {
            this.zone.runOutsideAngular(refresh.bind(this));
        };
    }
}
NoZoneBaseChartDirective.ɵfac = function NoZoneBaseChartDirective_Factory(t) { return new (t || NoZoneBaseChartDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ThemeService), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.NgZone)); };
NoZoneBaseChartDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NoZoneBaseChartDirective, selectors: [["canvas", "cNoZoneBaseChart", ""]], exportAs: ["base-chart"], features: [ɵngcc0.ɵɵInheritDefinitionFeature] });
NoZoneBaseChartDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: ThemeService },
    { type: NgZone }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NoZoneBaseChartDirective, [{
        type: Directive,
        args: [{
                selector: 'canvas[cNoZoneBaseChart]',
                exportAs: 'base-chart'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc1.ThemeService }, { type: ɵngcc0.NgZone }]; }, null); })();
