// @coreui/angular/esm2015/lib/form/form.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormDirective } from './form/form.directive';
import { FormGroupComponent } from './form-group/form-group.component';
import { FormFeedbackComponent } from './form-feedback/form-feedback.component';
import { FormTextComponent } from './form-text/form-text.component';
import { InputDirective } from './input/input.directive';
import { InputGroupComponent } from './input-group/input-group.component';
import { InputGroupTextComponent } from './input-group-text/input-group-text.component';
import { InputGroupAddonComponent, InputGroupPrependComponent, InputGroupAppendComponent } from './input-group-addon/input-group-addon.component';
import { TextareaDirective } from './textarea/textarea.directive';
import { SelectDirective } from './select/select.directive';
import { LabelDirective } from './label/label.directive';
import { FormCheckComponent } from './form-check/form-check.component';
import * as ɵngcc0 from '@angular/core';
export class FormModule {
}
FormModule.ɵfac = function FormModule_Factory(t) { return new (t || FormModule)(); };
FormModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FormModule });
FormModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FormModule, { declarations: function () { return [FormDirective,
        FormCheckComponent,
        FormFeedbackComponent,
        FormGroupComponent,
        FormTextComponent,
        InputDirective,
        InputGroupComponent,
        InputGroupTextComponent,
        InputGroupAddonComponent,
        InputGroupPrependComponent,
        InputGroupAppendComponent,
        TextareaDirective,
        SelectDirective,
        LabelDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [FormDirective,
        FormCheckComponent,
        FormFeedbackComponent,
        FormGroupComponent,
        FormTextComponent,
        InputDirective,
        InputGroupComponent,
        InputGroupTextComponent,
        InputGroupPrependComponent,
        InputGroupAppendComponent,
        TextareaDirective,
        SelectDirective,
        LabelDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FormModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    FormDirective,
                    FormCheckComponent,
                    FormFeedbackComponent,
                    FormGroupComponent,
                    FormTextComponent,
                    InputDirective,
                    InputGroupComponent,
                    InputGroupTextComponent,
                    InputGroupAddonComponent,
                    InputGroupPrependComponent,
                    InputGroupAppendComponent,
                    TextareaDirective,
                    SelectDirective,
                    LabelDirective
                ],
                imports: [
                    CommonModule
                ],
                exports: [
                    FormDirective,
                    FormCheckComponent,
                    FormFeedbackComponent,
                    FormGroupComponent,
                    FormTextComponent,
                    InputDirective,
                    InputGroupComponent,
                    InputGroupTextComponent,
                    InputGroupPrependComponent,
                    InputGroupAppendComponent,
                    TextareaDirective,
                    SelectDirective,
                    LabelDirective
                ]
            }]
    }], null, null); })();
