// @coreui/angular/esm2015/lib/carousel/carousel-caption/carousel-caption.component.js
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class CarouselCaptionComponent {
    constructor() {
        this.carouselCaptionClass = true;
    }
    ngOnInit() {
    }
}
CarouselCaptionComponent.ɵfac = function CarouselCaptionComponent_Factory(t) { return new (t || CarouselCaptionComponent)(); };
CarouselCaptionComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CarouselCaptionComponent, selectors: [["c-carousel-caption"]], hostVars: 2, hostBindings: function CarouselCaptionComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("carousel-caption", ctx.carouselCaptionClass);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function CarouselCaptionComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
CarouselCaptionComponent.ctorParameters = () => [];
CarouselCaptionComponent.propDecorators = {
    carouselCaptionClass: [{ type: HostBinding, args: ['class.carousel-caption',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CarouselCaptionComponent, [{
        type: Component,
        args: [{
                selector: 'c-carousel-caption',
                template: "<ng-content></ng-content>\n\n",
                styles: [""]
            }]
    }], function () { return []; }, { carouselCaptionClass: [{
            type: HostBinding,
            args: ['class.carousel-caption']
        }] }); })();
