// @coreui/angular/esm2015/lib/carousel/carousel-state.js
import { Injectable } from '@angular/core';
import { CarouselService } from './carousel.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './carousel.service';
export class CarouselState {
    constructor(carouselService) {
        this.carouselService = carouselService;
        this._state = {
            activeItemIndex: -1,
            animate: true,
            items: [],
            direction: 'next',
        };
    }
    get state() {
        return this._state;
    }
    set state(state) {
        const prevState = Object.assign({}, this._state);
        const nextState = Object.assign(Object.assign({}, this._state), state);
        this._state = nextState;
        if (prevState.activeItemIndex !== nextState.activeItemIndex) {
            this.carouselService.setIndex({ active: nextState.activeItemIndex });
        }
    }
    setItems(newItems) {
        if (newItems.length) {
            const itemsArray = newItems.toArray();
            itemsArray.forEach((item, i) => {
                item.index = i;
            });
            this.state = {
                items: itemsArray,
            };
        }
        else {
            this.reset();
        }
    }
    setNextIndex(nextIndex) {
        this.carouselService.setIndex(nextIndex);
    }
    direction(direction = 'next') {
        this.state = { direction };
        const { activeItemIndex, items } = this.state;
        const itemsCount = items.length;
        return direction === 'next' ?
            (activeItemIndex === itemsCount - 1 ? 0 : activeItemIndex + 1) :
            (activeItemIndex === 0 ? itemsCount - 1 : activeItemIndex - 1);
    }
    reset() {
        this.state = {
            activeItemIndex: -1,
            animate: true,
            items: [],
            direction: 'next',
        };
    }
}
CarouselState.ɵfac = function CarouselState_Factory(t) { return new (t || CarouselState)(ɵngcc0.ɵɵinject(ɵngcc1.CarouselService)); };
CarouselState.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: CarouselState, factory: CarouselState.ɵfac });
CarouselState.ctorParameters = () => [
    { type: CarouselService }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CarouselState, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc1.CarouselService }]; }, null); })();
