// @coreui/angular/esm2015/lib/card/card.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CardComponent } from './card.component';
import { CardBodyComponent } from './card-body.component';
import { CardFooterComponent } from './card-footer.component';
import { CardHeaderComponent } from './card-header.component';
import { CardHeaderActionsComponent } from './card-header-actions.component';
import { CardGroupComponent } from './card-group.component';
import * as ɵngcc0 from '@angular/core';
export class CardModule {
}
CardModule.ɵfac = function CardModule_Factory(t) { return new (t || CardModule)(); };
CardModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CardModule });
CardModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CardModule, { declarations: function () { return [CardComponent,
        CardBodyComponent,
        CardFooterComponent,
        CardGroupComponent,
        CardHeaderComponent,
        CardHeaderActionsComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CardComponent,
        CardBodyComponent,
        CardFooterComponent,
        CardGroupComponent,
        CardHeaderComponent,
        CardHeaderActionsComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CardModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    CardComponent,
                    CardBodyComponent,
                    CardFooterComponent,
                    CardGroupComponent,
                    CardHeaderComponent,
                    CardHeaderActionsComponent
                ],
                imports: [
                    CommonModule,
                ],
                exports: [
                    CardComponent,
                    CardBodyComponent,
                    CardFooterComponent,
                    CardGroupComponent,
                    CardHeaderComponent,
                    CardHeaderActionsComponent
                ]
            }]
    }], null, null); })();
