// @coreui/angular/esm2015/lib/card/card-group.component.js
import { Component, HostBinding, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class CardGroupComponent {
    constructor() {
        this.deck = false;
        this.columns = false;
    }
    get hostClasses() {
        return this.getClasses();
    }
    getClasses() {
        return `card-${this.columns ? 'columns' : this.deck ? 'deck' : 'group'}`;
    }
}
CardGroupComponent.ɵfac = function CardGroupComponent_Factory(t) { return new (t || CardGroupComponent)(); };
CardGroupComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CardGroupComponent, selectors: [["c-card-group"], ["", "c-card-group", ""]], hostVars: 2, hostBindings: function CardGroupComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.hostClasses);
    } }, inputs: { deck: "deck", columns: "columns" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function CardGroupComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
CardGroupComponent.propDecorators = {
    deck: [{ type: Input }],
    columns: [{ type: Input }],
    hostClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CardGroupComponent, [{
        type: Component,
        args: [{
                selector: 'c-card-group, [c-card-group]',
                template: '<ng-content></ng-content>'
            }]
    }], function () { return []; }, { deck: [{
            type: Input
        }], columns: [{
            type: Input
        }], hostClasses: [{
            type: HostBinding,
            args: ['class']
        }] }); })();
