// angular2-ladda/bundles/angular2-ladda.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('ladda')) :
    typeof define === 'function' && define.amd ? define('angular2-ladda', ['exports', '@angular/core', '@angular/common', 'ladda'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['angular2-ladda'] = {}, global.ng.core, global.ng.common, global.ladda));
}(this, (function (exports, core, common, ladda) { 'use strict';

    var LaddaConfigArgs = /** @class */ (function () {
        function LaddaConfigArgs() {
        }
        return LaddaConfigArgs;
    }());
    var configAttributes = {
        "data-style": "style",
        "data-spinner-size": "spinnerSize",
        "data-spinner-color": "spinnerColor",
        "data-spinner-lines": "spinnerLines",
    };
    var LaddaConfig = /** @class */ (function () {
        function LaddaConfig(config) {
            if (config === void 0) { config = {}; }
            Object.assign(this, config);
        }
        return LaddaConfig;
    }());
    LaddaConfig.decorators = [
        { type: core.Injectable }
    ];
    LaddaConfig.ctorParameters = function () { return [
        { type: LaddaConfigArgs }
    ]; };

    var LaddaDirective = /** @class */ (function () {
        function LaddaDirective(el, config, platformId) {
            this.platformId = platformId;
            this.ladda = undefined;
            this.disabled = false;
            this.el = el.nativeElement;
            if (!config) {
                return;
            }
            // apply default styles if they aren't overwritten by an attribute
            for (var attribute in configAttributes) {
                var configValue = config[configAttributes[attribute]];
                if (!configValue) {
                    continue; // don't waste time reading the attribute
                }
                if (!this.el.getAttribute(attribute)) {
                    // attribute isn't set - apply the default config value
                    var value = (typeof configValue === "number") ? configValue.toString() : configValue;
                    this.el.setAttribute(attribute, value);
                }
            }
        }
        LaddaDirective.prototype.ngOnChanges = function (changes) {
            if (!this.ladda) {
                return; // needed since ngOnChanges is called before ngOnInit
            }
            if (changes.loading) {
                this.updateLadda(changes.loading.previousValue);
            }
            if (changes.disabled) {
                this.updateDisabled();
            }
        };
        LaddaDirective.prototype.ngOnInit = function () {
            var _this = this;
            if (!common.isPlatformBrowser(this.platformId)) {
                return;
            }
            this.ladda = ladda.create(this.el);
            // if the initial loading value isn't false, a timeout of 0 ms
            // is necessary for the calculated spinner size to be correct.
            setTimeout(function () { _this.updateLadda(false); }, 0);
        };
        LaddaDirective.prototype.ngOnDestroy = function () {
            if (this.ladda) {
                this.ladda.remove();
            }
        };
        LaddaDirective.prototype.updateLadda = function (previousValue) {
            if (!this.ladda) {
                return;
            }
            var loading = typeof this.loading === 'number' || !!this.loading;
            var wasLoading = typeof previousValue === 'number' || !!previousValue;
            if (!loading) {
                if (wasLoading) {
                    this.ladda.stop();
                }
                return this.updateDisabled();
            }
            if (!wasLoading) {
                this.ladda.start();
            }
            if (typeof this.loading === 'number') {
                this.ladda.setProgress(this.loading);
            }
        };
        LaddaDirective.prototype.updateDisabled = function () {
            this.el.disabled = this.disabled;
        };
        return LaddaDirective;
    }());
    LaddaDirective.decorators = [
        { type: core.Directive, args: [{
                    selector: '[ladda]'
                },] }
    ];
    LaddaDirective.ctorParameters = function () { return [
        { type: core.ElementRef },
        { type: LaddaConfigArgs, decorators: [{ type: core.Inject, args: [LaddaConfig,] }, { type: core.Optional }] },
        { type: Object, decorators: [{ type: core.Inject, args: [core.PLATFORM_ID,] }] }
    ]; };
    LaddaDirective.propDecorators = {
        loading: [{ type: core.Input, args: ['ladda',] }],
        disabled: [{ type: core.Input }]
    };

    var LaddaModule = /** @class */ (function () {
        function LaddaModule() {
        }
        LaddaModule.forRoot = function (config) {
            return {
                ngModule: LaddaModule,
                providers: [
                    { provide: LaddaConfig, useValue: config }
                ]
            };
        };
        return LaddaModule;
    }());
    LaddaModule.decorators = [
        { type: core.NgModule, args: [{
                    declarations: [LaddaDirective],
                    exports: [LaddaDirective],
                },] }
    ];

    /*
     * Public API Surface of ladda
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.LaddaConfig = LaddaConfig;
    exports.LaddaConfigArgs = LaddaConfigArgs;
    exports.LaddaDirective = LaddaDirective;
    exports.LaddaModule = LaddaModule;
    exports.configAttributes = configAttributes;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
