// @coreui/angular/esm2015/lib/form/textarea/textarea.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import { chkBoolProp } from '../../shared/check-bool-prop';
import * as ɵngcc0 from '@angular/core';
export class TextareaDirective {
    constructor() {
        this.sizing = '';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const isPlainText = this.isPlainText;
        const classes = classNames({
            'form-control': !isPlainText,
            'form-control-plaintext': isPlainText,
            [`form-control-${this.sizing}`]: !!this.sizing,
            'is-valid': this.valid === true,
            'is-invalid': this.valid === false,
        });
        return classes;
    }
    get isPlainText() {
        return chkBoolProp(this.plaintext);
    }
}
TextareaDirective.ɵfac = function TextareaDirective_Factory(t) { return new (t || TextareaDirective)(); };
TextareaDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: TextareaDirective, selectors: [["", "cTextarea", ""]], hostVars: 2, hostBindings: function TextareaDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { sizing: "sizing", valid: "valid", plaintext: "plaintext" } });
TextareaDirective.propDecorators = {
    valid: [{ type: Input }],
    plaintext: [{ type: Input }],
    sizing: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TextareaDirective, [{
        type: Directive,
        args: [{
                selector: '[cTextarea]'
            }]
    }], function () { return []; }, { sizing: [{
            type: Input
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], valid: [{
            type: Input
        }], plaintext: [{
            type: Input
        }] }); })();
