// @coreui/angular/esm2015/lib/form/form-check/form-check.component.js
import { Component, HostBinding, Input } from '@angular/core';
import { chkBoolProp } from '../../shared/check-bool-prop';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class FormCheckComponent {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        if (this.variant === 'switch') {
            this.custom = true;
        }
        const variantClass = this.custom ? 'custom-control' : 'form-check';
        const isCustom = chkBoolProp(this.custom);
        const classes = {
            'form-check': !isCustom,
            'custom-control': isCustom,
            [`custom-${this.variant}`]: !!this.variant,
            [`${variantClass}-inline`]: chkBoolProp(this.inline),
            disabled: chkBoolProp(this.disabled)
        };
        return classes;
    }
    ngOnInit() {
    }
}
FormCheckComponent.ɵfac = function FormCheckComponent_Factory(t) { return new (t || FormCheckComponent)(); };
FormCheckComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FormCheckComponent, selectors: [["c-form-check"]], hostVars: 2, hostBindings: function FormCheckComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { custom: "custom", disabled: "disabled", inline: "inline", variant: "variant" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function FormCheckComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
FormCheckComponent.ctorParameters = () => [];
FormCheckComponent.propDecorators = {
    custom: [{ type: Input }],
    disabled: [{ type: Input }],
    inline: [{ type: Input }],
    variant: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FormCheckComponent, [{
        type: Component,
        args: [{
                selector: 'c-form-check',
                template: "<ng-content></ng-content>\n",
                styles: [""]
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], custom: [{
            type: Input
        }], disabled: [{
            type: Input
        }], inline: [{
            type: Input
        }], variant: [{
            type: Input
        }] }); })();
