// @coreui/angular/esm2015/lib/layout/body/body.component.js
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class BodyComponent {
    constructor() {
        this.bodyClass = true;
    }
}
BodyComponent.ɵfac = function BodyComponent_Factory(t) { return new (t || BodyComponent)(); };
BodyComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: BodyComponent, selectors: [["c-body"], ["", "c-body", ""]], hostVars: 2, hostBindings: function BodyComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("c-body", ctx.bodyClass);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function BodyComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]{display:flex;flex-direction:column}"] });
BodyComponent.propDecorators = {
    bodyClass: [{ type: HostBinding, args: ['class.c-body',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BodyComponent, [{
        type: Component,
        args: [{
                selector: 'c-body, [c-body]',
                template: "<ng-content></ng-content>\n",
                styles: [":host{display:flex;flex-direction:column}"]
            }]
    }], function () { return []; }, { bodyClass: [{
            type: HostBinding,
            args: ['class.c-body']
        }] }); })();
