// @coreui/angular/esm2015/lib/carousel/carousel-control/carousel-control.component.js
import { ChangeDetectorRef, Component, HostBinding, HostListener, Input, ViewChild, } from '@angular/core';
import classNames from 'classnames';
import { CarouselState } from '../carousel-state';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../carousel-state';
import * as ɵngcc2 from '@angular/common';

const _c0 = ["content"];
function CarouselControlComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", null, 2);
    ɵngcc0.ɵɵprojection(2);
    ɵngcc0.ɵɵelementEnd();
} }
function CarouselControlComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span");
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap(ctx_r2.carouselControlIconClass);
    ɵngcc0.ɵɵattribute("aria-label", ctx_r2.direction);
} }
const _c1 = ["*"];
export class CarouselControlComponent {
    constructor(changeDetector, carouselState) {
        this.changeDetector = changeDetector;
        this.carouselState = carouselState;
        this.direction = 'next';
        this.hasContent = true;
    }
    get hostRole() {
        return 'button';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get carouselControlClass() {
        return `carousel-control-${this.direction}`;
    }
    get carouselControlIconClass() {
        return `carousel-control-${this.direction}-icon`;
    }
    get hostClasses() {
        const classes = classNames(this.carouselControlClass);
        return classes;
    }
    onKeyUp($event) {
        if ($event.key === 'Enter') {
            this.play();
        }
        if ($event.key === 'ArrowLeft') {
            this.play('prev');
        }
        if ($event.key === 'ArrowRight') {
            this.play('next');
        }
    }
    onClick($event) {
        this.play();
    }
    ngOnInit() { }
    ngAfterViewInit() {
        this.hasContent = this.content.nativeElement.childNodes.length;
        this.changeDetector.detectChanges();
    }
    play(direction = this.direction) {
        const nextIndex = this.carouselState.direction(direction);
        this.carouselState.state = { activeItemIndex: nextIndex };
    }
}
CarouselControlComponent.ɵfac = function CarouselControlComponent_Factory(t) { return new (t || CarouselControlComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CarouselState)); };
CarouselControlComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CarouselControlComponent, selectors: [["c-carousel-control"]], viewQuery: function CarouselControlComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, 1);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.content = _t.first);
    } }, hostVars: 3, hostBindings: function CarouselControlComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("keyup", function CarouselControlComponent_keyup_HostBindingHandler($event) { return ctx.onKeyUp($event); })("click", function CarouselControlComponent_click_HostBindingHandler($event) { return ctx.onClick($event); });
    } if (rf & 2) {
        ɵngcc0.ɵɵattribute("role", ctx.hostRole);
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { direction: "direction" }, ngContentSelectors: _c1, decls: 3, vars: 2, consts: [[4, "ngIf", "ngIfElse"], ["default", ""], ["content", ""]], template: function CarouselControlComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵtemplate(0, CarouselControlComponent_div_0_Template, 3, 0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, CarouselControlComponent_ng_template_1_Template, 1, 3, "ng-template", null, 1, ɵngcc0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        const _r1 = ɵngcc0.ɵɵreference(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasContent)("ngIfElse", _r1);
    } }, directives: [ɵngcc2.NgIf], styles: [""] });
CarouselControlComponent.ctorParameters = () => [
    { type: ChangeDetectorRef },
    { type: CarouselState }
];
CarouselControlComponent.propDecorators = {
    direction: [{ type: Input }],
    content: [{ type: ViewChild, args: ['content',] }],
    hostRole: [{ type: HostBinding, args: ['attr.role',] }],
    getClasses: [{ type: HostBinding, args: ['class',] }],
    onKeyUp: [{ type: HostListener, args: ['keyup', ['$event'],] }],
    onClick: [{ type: HostListener, args: ['click', ['$event'],] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CarouselControlComponent, [{
        type: Component,
        args: [{
                selector: 'c-carousel-control',
                template: "<div #content *ngIf = \"hasContent; else default\"><ng-content></ng-content></div>\n<ng-template #default>\n  <span [class]=\"carouselControlIconClass\" [attr.aria-label]=\"direction\"></span>\n</ng-template>\n",
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc0.ChangeDetectorRef }, { type: ɵngcc1.CarouselState }]; }, { direction: [{
            type: Input
        }], hostRole: [{
            type: HostBinding,
            args: ['attr.role']
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], onKeyUp: [{
            type: HostListener,
            args: ['keyup', ['$event']]
        }], onClick: [{
            type: HostListener,
            args: ['click', ['$event']]
        }], content: [{
            type: ViewChild,
            args: ['content']
        }] }); })();
