// @coreui/angular/esm2015/lib/sidebar/public_api.js
export { SidebarBackdropComponent } from './sidebar-backdrop.component';
export { SidebarBrandComponent } from './sidebar-brand.component';
export { SidebarFooterComponent } from './sidebar-footer.component';
export { SidebarFormComponent } from './sidebar-form.component';
export { SidebarHeaderComponent } from './sidebar-header.component';
export { SidebarMinimizerComponent } from './sidebar-minimizer.component';
export { SidebarMinimizeDirective } from './sidebar-minimize.directive';
export { SidebarToggleDirective } from './sidebar-toggle.directive';
export { SidebarComponent } from './sidebar.component';
export { SidebarService } from './sidebar.service';
export { SidebarModule } from './sidebar.module';
export { SidebarNavTitleComponent, SidebarNavItemClassPipe, SidebarNavIconPipe, SidebarNavDropdownToggleDirective, SidebarNavDropdownComponent, SidebarNavDividerComponent, SidebarNavBadgePipe, SidebarNavService, SidebarNavComponent, SidebarNavItemsComponent, SidebarNavLinkContentComponent, SidebarNavLinkPipe, SidebarNavLinkComponent, SidebarNavLabelComponent } from './sidebar-nav';
