// @coreui/angular/esm2015/lib/sidebar/sidebar-toggle.directive.js
import { Directive, HostListener, Input } from '@angular/core';
import { SidebarService } from './sidebar.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './sidebar.service';
export class SidebarToggleDirective {
    constructor(sidebarService) {
        this.sidebarService = sidebarService;
    }
    toggleOpen($event) {
        $event.preventDefault();
        this.sidebarService.toggle({ open: 'toggle', id: this.id });
    }
}
SidebarToggleDirective.ɵfac = function SidebarToggleDirective_Factory(t) { return new (t || SidebarToggleDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.SidebarService)); };
SidebarToggleDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SidebarToggleDirective, selectors: [["", "cSidebarToggle", ""]], hostBindings: function SidebarToggleDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function SidebarToggleDirective_click_HostBindingHandler($event) { return ctx.toggleOpen($event); });
    } }, inputs: { id: ["cSidebarToggle", "id"] }, exportAs: ["cSidebarToggle"] });
SidebarToggleDirective.ctorParameters = () => [
    { type: SidebarService }
];
SidebarToggleDirective.propDecorators = {
    id: [{ type: Input, args: ['cSidebarToggle',] }],
    toggleOpen: [{ type: HostListener, args: ['click', ['$event'],] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarToggleDirective, [{
        type: Directive,
        args: [{
                selector: '[cSidebarToggle]',
                exportAs: 'cSidebarToggle'
            }]
    }], function () { return [{ type: ɵngcc1.SidebarService }]; }, { toggleOpen: [{
            type: HostListener,
            args: ['click', ['$event']]
        }], id: [{
            type: Input,
            args: ['cSidebarToggle']
        }] }); })();
