// @coreui/angular/esm2015/lib/sidebar/sidebar-minimizer.component.js
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class SidebarMinimizerComponent {
    constructor() {
        this.role = 'button';
        this._sidebarMinimizerClass = true;
    }
}
SidebarMinimizerComponent.ɵfac = function SidebarMinimizerComponent_Factory(t) { return new (t || SidebarMinimizerComponent)(); };
SidebarMinimizerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SidebarMinimizerComponent, selectors: [["c-sidebar-minimizer"]], hostVars: 3, hostBindings: function SidebarMinimizerComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("role", ctx.role);
        ɵngcc0.ɵɵclassProp("c-sidebar-minimizer", ctx._sidebarMinimizerClass);
    } }, decls: 0, vars: 0, template: function SidebarMinimizerComponent_Template(rf, ctx) { }, encapsulation: 2 });
SidebarMinimizerComponent.propDecorators = {
    role: [{ type: HostBinding, args: ['attr.role',] }],
    _sidebarMinimizerClass: [{ type: HostBinding, args: ['class.c-sidebar-minimizer',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarMinimizerComponent, [{
        type: Component,
        args: [{
                selector: 'c-sidebar-minimizer',
                template: ``
            }]
    }], function () { return []; }, { role: [{
            type: HostBinding,
            args: ['attr.role']
        }], _sidebarMinimizerClass: [{
            type: HostBinding,
            args: ['class.c-sidebar-minimizer']
        }] }); })();
