// @coreui/angular/esm2015/lib/sidebar/sidebar-brand.component.js
import { Component, HostBinding, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/router';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../shared/html-attr.directive';

function SidebarBrandComponent_ng_template_1_img_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 3);
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("cHtmlAttr", ctx_r2.brandFull)("ngClass", "c-sidebar-brand-full");
} }
function SidebarBrandComponent_ng_template_1_img_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 3);
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵproperty("cHtmlAttr", ctx_r3.brandMinimized)("ngClass", "c-sidebar-brand-minimized");
} }
function SidebarBrandComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, SidebarBrandComponent_ng_template_1_img_0_Template, 1, 2, "img", 2);
    ɵngcc0.ɵɵtemplate(1, SidebarBrandComponent_ng_template_1_img_1_Template, 1, 2, "img", 2);
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.brandFull);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r0.brandMinimized);
} }
function SidebarBrandComponent_ng_template_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 4);
    ɵngcc0.ɵɵelement(1, "div", 5);
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("innerHTML", ctx_r1.brandText.text, ɵngcc0.ɵɵsanitizeHtml);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("innerHTML", ctx_r1.brandText.icon, ɵngcc0.ɵɵsanitizeHtml);
} }
export class SidebarBrandComponent {
    constructor() {
        this.brandText = { icon: '🅲', text: '🅲 CoreUI' };
        this.routerLink = '';
        this._sidebarBrand = true;
    }
    ngOnInit() {
        this.brandImg = Boolean(this.brand || this.brandFull || this.brandMinimized);
    }
}
SidebarBrandComponent.ɵfac = function SidebarBrandComponent_Factory(t) { return new (t || SidebarBrandComponent)(); };
SidebarBrandComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SidebarBrandComponent, selectors: [["c-sidebar-brand"]], hostVars: 2, hostBindings: function SidebarBrandComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("c-sidebar-brand", ctx._sidebarBrand);
    } }, inputs: { brandText: "brandText", routerLink: "routerLink", brand: "brand", brandFull: "brandFull", brandMinimized: "brandMinimized" }, decls: 3, vars: 3, consts: [[3, "routerLink"], [3, "ngIf"], [3, "cHtmlAttr", "ngClass", 4, "ngIf"], [3, "cHtmlAttr", "ngClass"], [1, "c-sidebar-brand-full", 3, "innerHTML"], [1, "c-sidebar-brand-minimized", 3, "innerHTML"]], template: function SidebarBrandComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "a", 0);
        ɵngcc0.ɵɵtemplate(1, SidebarBrandComponent_ng_template_1_Template, 2, 2, "ng-template", 1);
        ɵngcc0.ɵɵtemplate(2, SidebarBrandComponent_ng_template_2_Template, 2, 2, "ng-template", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("routerLink", ctx.routerLink);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.brandImg);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.brandImg);
    } }, directives: [ɵngcc1.RouterLinkWithHref, ɵngcc2.NgIf, ɵngcc3.HtmlAttributesDirective, ɵngcc2.NgClass], encapsulation: 2 });
SidebarBrandComponent.ctorParameters = () => [];
SidebarBrandComponent.propDecorators = {
    brand: [{ type: Input }],
    brandFull: [{ type: Input }],
    brandMinimized: [{ type: Input }],
    brandText: [{ type: Input }],
    routerLink: [{ type: Input }],
    _sidebarBrand: [{ type: HostBinding, args: ['class.c-sidebar-brand',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarBrandComponent, [{
        type: Component,
        args: [{
                selector: 'c-sidebar-brand',
                template: "<a [routerLink]=\"routerLink\">\r\n  <ng-template [ngIf]=\"brandImg\">\r\n<!--    <img *ngIf=\"brand\"-->\r\n<!--         [cHtmlAttr]=\"brand\"-->\r\n<!--         [ngClass]=\"'c-sidebar-brand'\">-->\r\n    <img *ngIf=\"brandFull\"\r\n         [cHtmlAttr]=\"brandFull\"\r\n         [ngClass]=\"'c-sidebar-brand-full'\">\r\n    <img *ngIf=\"brandMinimized\"\r\n         [cHtmlAttr]=\"brandMinimized\"\r\n         [ngClass]=\"'c-sidebar-brand-minimized'\">\r\n  </ng-template>\r\n  <ng-template [ngIf]=\"!brandImg\">\r\n    <div class=\"c-sidebar-brand-full\" [innerHTML]=\"brandText.text\"></div>\r\n    <div class=\"c-sidebar-brand-minimized\" [innerHTML]=\"brandText.icon\"></div>\r\n  </ng-template>\r\n</a>\r\n"
            }]
    }], function () { return []; }, { brandText: [{
            type: Input
        }], routerLink: [{
            type: Input
        }], _sidebarBrand: [{
            type: HostBinding,
            args: ['class.c-sidebar-brand']
        }], brand: [{
            type: Input
        }], brandFull: [{
            type: Input
        }], brandMinimized: [{
            type: Input
        }] }); })();
