// @coreui/angular/esm2015/lib/sidebar/sidebar-minimize.directive.js
import { Directive, HostListener, Input } from '@angular/core';
import { SidebarService } from './sidebar.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './sidebar.service';
export class SidebarMinimizeDirective {
    constructor(sidebarService) {
        this.sidebarService = sidebarService;
    }
    toggleMinimize($event) {
        $event.preventDefault();
        this.sidebarService.toggle({ minimize: 'toggle', id: this.id });
    }
}
SidebarMinimizeDirective.ɵfac = function SidebarMinimizeDirective_Factory(t) { return new (t || SidebarMinimizeDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.SidebarService)); };
SidebarMinimizeDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SidebarMinimizeDirective, selectors: [["", "cSidebarMinimize", ""]], hostBindings: function SidebarMinimizeDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function SidebarMinimizeDirective_click_HostBindingHandler($event) { return ctx.toggleMinimize($event); });
    } }, inputs: { id: ["cSidebarMinimize", "id"] }, exportAs: ["cSidebarMinimize"] });
SidebarMinimizeDirective.ctorParameters = () => [
    { type: SidebarService }
];
SidebarMinimizeDirective.propDecorators = {
    id: [{ type: Input, args: ['cSidebarMinimize',] }],
    toggleMinimize: [{ type: HostListener, args: ['click', ['$event'],] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarMinimizeDirective, [{
        type: Directive,
        args: [{
                selector: '[cSidebarMinimize]',
                exportAs: 'cSidebarMinimize'
            }]
    }], function () { return [{ type: ɵngcc1.SidebarService }]; }, { toggleMinimize: [{
            type: HostListener,
            args: ['click', ['$event']]
        }], id: [{
            type: Input,
            args: ['cSidebarMinimize']
        }] }); })();
