// @coreui/angular/esm2015/lib/sidebar/sidebar-backdrop.component.js
import { Component, ElementRef, HostBinding, HostListener } from '@angular/core';
import { SidebarService } from './sidebar.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './sidebar.service';
export class SidebarBackdropComponent {
    constructor(elementRef, sidebarService) {
        this.elementRef = elementRef;
        this.sidebarService = sidebarService;
        this._sidebarBackdrop = true;
    }
    onClick($event) {
        const targetElement = $event.target;
        if (targetElement && !this.elementRef.nativeElement.contains(targetElement)) {
            this.sidebarService.toggle({ open: false });
        }
    }
    ngOnInit() {
    }
}
SidebarBackdropComponent.ɵfac = function SidebarBackdropComponent_Factory(t) { return new (t || SidebarBackdropComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.SidebarService)); };
SidebarBackdropComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SidebarBackdropComponent, selectors: [["c-sidebar-backdrop"]], hostVars: 2, hostBindings: function SidebarBackdropComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function SidebarBackdropComponent_click_HostBindingHandler($event) { return ctx.onClick($event); });
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("c-sidebar-backdrop", ctx._sidebarBackdrop);
    } }, decls: 0, vars: 0, template: function SidebarBackdropComponent_Template(rf, ctx) { }, styles: [""] });
SidebarBackdropComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: SidebarService }
];
SidebarBackdropComponent.propDecorators = {
    _sidebarBackdrop: [{ type: HostBinding, args: ['class.c-sidebar-backdrop',] }],
    onClick: [{ type: HostListener, args: ['click', ['$event'],] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarBackdropComponent, [{
        type: Component,
        args: [{
                selector: 'c-sidebar-backdrop',
                template: "",
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc1.SidebarService }]; }, { _sidebarBackdrop: [{
            type: HostBinding,
            args: ['class.c-sidebar-backdrop']
        }], onClick: [{
            type: HostListener,
            args: ['click', ['$event']]
        }] }); })();
