// @coreui/angular/esm2015/lib/sidebar/sidebar.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { SharedModule } from '../shared';
import { OutClickService } from '../shared/out-click.service';
import { SidebarComponent } from './sidebar.component';
import { SidebarBrandComponent } from './sidebar-brand.component';
import { SidebarFooterComponent } from './sidebar-footer.component';
import { SidebarFormComponent } from './sidebar-form.component';
import { SidebarHeaderComponent } from './sidebar-header.component';
import { SidebarMinimizerComponent } from './sidebar-minimizer.component';
import { SidebarBackdropComponent } from './sidebar-backdrop.component';
import { SidebarService } from './sidebar.service';
import { SidebarToggleDirective } from './sidebar-toggle.directive';
import { SidebarMinimizeDirective } from './sidebar-minimize.directive';
import { SidebarNavComponent } from './sidebar-nav/sidebar-nav.component';
import { SidebarNavDividerComponent } from './sidebar-nav/sidebar-nav-divider.component';
import { SidebarNavDropdownComponent } from './sidebar-nav/sidebar-nav-dropdown.component';
import { SidebarNavItemsComponent } from './sidebar-nav/sidebar-nav-items.component';
import { SidebarNavLinkComponent } from './sidebar-nav/sidebar-nav-link.component';
import { SidebarNavLinkContentComponent } from './sidebar-nav/sidebar-nav-link.component';
import { SidebarNavTitleComponent } from './sidebar-nav/sidebar-nav-title.component';
import { SidebarNavHelper } from './sidebar-nav/sidebar-nav.service';
import { SidebarNavLabelComponent } from './sidebar-nav/sidebar-nav-label.component';
import { SidebarNavIconPipe } from './sidebar-nav/sidebar-nav-icon.pipe';
import { SidebarNavBadgePipe } from './sidebar-nav/sidebar-nav-badge.pipe';
import { SidebarNavItemClassPipe } from './sidebar-nav/sidebar-nav-item-class.pipe';
import { SidebarNavLinkPipe } from './sidebar-nav/sidebar-nav-link.pipe';
import { SidebarNavDropdownToggleDirective } from './sidebar-nav/sidebar-nav-dropdown.directive';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../shared/html-attr.directive';
import * as ɵngcc2 from '@angular/common';
export class SidebarModule {
}
SidebarModule.ɵfac = function SidebarModule_Factory(t) { return new (t || SidebarModule)(); };
SidebarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SidebarModule });
SidebarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ providers: [
        OutClickService,
        SidebarService,
        SidebarNavHelper,
    ], imports: [[
            CommonModule,
            RouterModule,
            SharedModule
        ], SharedModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SidebarModule, { declarations: function () { return [SidebarBrandComponent,
        SidebarFooterComponent,
        SidebarFormComponent,
        SidebarHeaderComponent,
        SidebarMinimizerComponent,
        SidebarComponent,
        SidebarNavItemsComponent,
        SidebarNavComponent,
        SidebarNavDividerComponent,
        SidebarNavDropdownComponent,
        SidebarNavLinkComponent,
        SidebarNavLinkContentComponent,
        SidebarNavTitleComponent,
        SidebarNavDropdownToggleDirective,
        SidebarNavLabelComponent,
        SidebarNavIconPipe,
        SidebarNavBadgePipe,
        SidebarNavItemClassPipe,
        SidebarNavLinkPipe,
        SidebarBackdropComponent,
        SidebarToggleDirective,
        SidebarMinimizeDirective]; }, imports: function () { return [CommonModule,
        RouterModule,
        SharedModule]; }, exports: function () { return [SidebarBrandComponent,
        SidebarFooterComponent,
        SidebarFormComponent,
        SidebarHeaderComponent,
        SidebarMinimizerComponent,
        SidebarComponent,
        SidebarNavItemsComponent,
        SidebarNavComponent,
        SidebarNavDividerComponent,
        SidebarNavDropdownComponent,
        SidebarNavLinkComponent,
        SidebarNavLinkContentComponent,
        SidebarNavTitleComponent,
        SidebarToggleDirective,
        SidebarMinimizeDirective,
        SharedModule]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    RouterModule,
                    SharedModule
                ],
                exports: [
                    SidebarBrandComponent,
                    SidebarFooterComponent,
                    SidebarFormComponent,
                    SidebarHeaderComponent,
                    SidebarMinimizerComponent,
                    SidebarComponent,
                    SidebarNavItemsComponent,
                    SidebarNavComponent,
                    SidebarNavDividerComponent,
                    SidebarNavDropdownComponent,
                    SidebarNavLinkComponent,
                    SidebarNavLinkContentComponent,
                    SidebarNavTitleComponent,
                    SidebarToggleDirective,
                    SidebarMinimizeDirective,
                    SharedModule
                ],
                declarations: [
                    SidebarBrandComponent,
                    SidebarFooterComponent,
                    SidebarFormComponent,
                    SidebarHeaderComponent,
                    SidebarMinimizerComponent,
                    SidebarComponent,
                    SidebarNavItemsComponent,
                    SidebarNavComponent,
                    SidebarNavDividerComponent,
                    SidebarNavDropdownComponent,
                    SidebarNavLinkComponent,
                    SidebarNavLinkContentComponent,
                    SidebarNavTitleComponent,
                    SidebarNavDropdownToggleDirective,
                    SidebarNavLabelComponent,
                    SidebarNavIconPipe,
                    SidebarNavBadgePipe,
                    SidebarNavItemClassPipe,
                    SidebarNavLinkPipe,
                    SidebarBackdropComponent,
                    SidebarToggleDirective,
                    SidebarMinimizeDirective
                ],
                providers: [
                    OutClickService,
                    SidebarService,
                    SidebarNavHelper,
                ]
            }]
    }], null, null); })();
ɵngcc0.ɵɵsetComponentScope(SidebarNavDropdownComponent, [SidebarNavDropdownToggleDirective, ɵngcc1.HtmlAttributesDirective, ɵngcc2.NgIf, ɵngcc2.NgClass, SidebarNavComponent], [SidebarNavIconPipe,
    SidebarNavBadgePipe]);
