// @coreui/angular/esm2015/lib/sidebar/sidebar-form.component.js
import { Component, ElementRef, Renderer2 } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class SidebarFormComponent {
    constructor(renderer, hostElement) {
        this.renderer = renderer;
        this.hostElement = hostElement;
        renderer.addClass(hostElement.nativeElement, 'c-sidebar-form');
    }
    ngOnInit() { }
}
SidebarFormComponent.ɵfac = function SidebarFormComponent_Factory(t) { return new (t || SidebarFormComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
SidebarFormComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SidebarFormComponent, selectors: [["c-sidebar-form"]], ngContentSelectors: _c0, decls: 1, vars: 0, template: function SidebarFormComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
SidebarFormComponent.ctorParameters = () => [
    { type: Renderer2 },
    { type: ElementRef }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarFormComponent, [{
        type: Component,
        args: [{
                selector: 'c-sidebar-form',
                template: `<ng-content></ng-content>`
            }]
    }], function () { return [{ type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ElementRef }]; }, null); })();
