// @coreui/angular/esm2015/lib/form/select/select.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import { chkBoolProp } from '../../shared/check-bool-prop';
import * as ɵngcc0 from '@angular/core';
export class SelectDirective {
    constructor() {
        this.sizing = '';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const isCustom = chkBoolProp(this.custom);
        const classes = classNames({
            'form-control': !isCustom,
            [`form-control-${this.sizing}`]: !!this.sizing && !isCustom,
            'custom-select': isCustom,
            [`custom-select-${this.sizing}`]: !!this.sizing && isCustom,
            'is-valid': this.valid === true,
            'is-invalid': this.valid === false,
        });
        return classes;
    }
}
SelectDirective.ɵfac = function SelectDirective_Factory(t) { return new (t || SelectDirective)(); };
SelectDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SelectDirective, selectors: [["", "cSelect", ""]], hostVars: 2, hostBindings: function SelectDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { sizing: "sizing", valid: "valid", custom: "custom" } });
SelectDirective.ctorParameters = () => [];
SelectDirective.propDecorators = {
    valid: [{ type: Input }],
    custom: [{ type: Input }],
    sizing: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SelectDirective, [{
        type: Directive,
        args: [{
                selector: '[cSelect]'
            }]
    }], function () { return []; }, { sizing: [{
            type: Input
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], valid: [{
            type: Input
        }], custom: [{
            type: Input
        }] }); })();
