// @coreui/angular/esm2015/lib/widget/widget-progress/widget-progress.component.js
import { Component, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../../progress/progress.component';
import * as ɵngcc2 from '../../progress/progress-bar.component';
import * as ɵngcc3 from '@angular/common';

function WidgetProgressComponent_small_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "small", 6);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.footer, " ");
} }
const _c0 = function (a2) { return { "my-3": true, "mb-0": true, "progress-white": a2 }; };
export class WidgetProgressComponent {
    constructor() {
        this.value = 25;
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames({
            card: true,
            'text-white': this.inverse,
            [`bg-${this.color}`]: this.inverse && !!this.color
        });
        return classes;
    }
    ngOnInit() {
    }
}
WidgetProgressComponent.ɵfac = function WidgetProgressComponent_Factory(t) { return new (t || WidgetProgressComponent)(); };
WidgetProgressComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: WidgetProgressComponent, selectors: [["c-widget-progress"]], hostVars: 2, hostBindings: function WidgetProgressComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { value: "value", color: "color", footer: "footer", header: "header", text: "text", inverse: "inverse" }, decls: 8, vars: 9, consts: [[1, "card-body"], ["v-if", "header", 1, "h4", "m-0"], ["v-if", "text"], ["size", "xs"], [3, "color", "value"], ["class", "text-muted", 4, "ngIf"], [1, "text-muted"]], template: function WidgetProgressComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵtext(4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "c-progress", 3);
        ɵngcc0.ɵɵelement(6, "c-progress-bar", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(7, WidgetProgressComponent_small_7_Template, 2, 1, "small", 5);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx.header);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx.text);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMap(ɵngcc0.ɵɵpureFunction1(7, _c0, ctx.inverse));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("color", !ctx.inverse ? ctx.color : "")("value", ctx.value);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !!ctx.footer);
    } }, directives: [ɵngcc1.ProgressComponent, ɵngcc2.ProgressBarComponent, ɵngcc3.NgIf], styles: [""] });
WidgetProgressComponent.ctorParameters = () => [];
WidgetProgressComponent.propDecorators = {
    color: [{ type: Input }],
    footer: [{ type: Input }],
    header: [{ type: Input }],
    text: [{ type: Input }],
    inverse: [{ type: Input }],
    value: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(WidgetProgressComponent, [{
        type: Component,
        args: [{
                selector: 'c-widget-progress',
                template: "<div class=\"card-body\">\n  <div v-if=\"header\" class=\"h4 m-0\">{{header}}</div>\n  <div v-if=\"text\">{{text}}</div>\n  <c-progress size=\"xs\" [class]=\"{'my-3': true, 'mb-0': true, 'progress-white': inverse }\">\n    <c-progress-bar\n      [color]=\"!inverse ? color : ''\"\n      [value]=\"value\"\n    ></c-progress-bar>\n  </c-progress>\n  <small *ngIf=\"!!footer\" class=\"text-muted\">\n    {{footer}}\n  </small>\n</div>\n",
                styles: [""]
            }]
    }], function () { return []; }, { value: [{
            type: Input
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], color: [{
            type: Input
        }], footer: [{
            type: Input
        }], header: [{
            type: Input
        }], text: [{
            type: Input
        }], inverse: [{
            type: Input
        }] }); })();
