// @coreui/angular/esm2015/lib/modal/modal.component.js
import { Component, ContentChild, ElementRef, ChangeDetectorRef, Input, HostListener, Renderer2, Inject, } from '@angular/core';
import { DOCUMENT } from '@angular/common';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

const _c0 = ["modalHeader"];
const _c1 = ["modalBody"];
const _c2 = ["modalFooter"];
function ModalComponent_div_0_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0, 11);
} }
function ModalComponent_div_0_ng_container_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0);
} }
function ModalComponent_div_0_ng_container_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0);
} }
function ModalComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    const _r6 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 2);
    ɵngcc0.ɵɵelementStart(1, "div", 3);
    ɵngcc0.ɵɵelementStart(2, "div", 4);
    ɵngcc0.ɵɵelementStart(3, "div", 5);
    ɵngcc0.ɵɵtemplate(4, ModalComponent_div_0_ng_container_4_Template, 1, 0, "ng-container", 6);
    ɵngcc0.ɵɵelementStart(5, "button", 7);
    ɵngcc0.ɵɵlistener("click", function ModalComponent_div_0_Template_button_click_5_listener() { ɵngcc0.ɵɵrestoreView(_r6); const ctx_r5 = ɵngcc0.ɵɵnextContext(); return ctx_r5.hide(); });
    ɵngcc0.ɵɵtext(6, "\u00D7");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(7, "div", 8);
    ɵngcc0.ɵɵtemplate(8, ModalComponent_div_0_ng_container_8_Template, 1, 0, "ng-container", 9);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(9, "div", 10);
    ɵngcc0.ɵɵtemplate(10, ModalComponent_div_0_ng_container_10_Template, 1, 0, "ng-container", 9);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("show", ctx_r0.visibleAnimate);
    ɵngcc0.ɵɵproperty("ngClass", ctx_r0.modalClasses);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngClass", ctx_r0.dialogClasses);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r0.header);
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r0.body);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r0.footer);
} }
function ModalComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 12);
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ctx_r1.backdropClasses);
} }
export class ModalComponent {
    constructor(document, renderer, elementRef, changeDetectorRef) {
        this.document = document;
        this.renderer = renderer;
        this.elementRef = elementRef;
        this.changeDetectorRef = changeDetectorRef;
        this.closeOnBackdrop = true;
        this.size = '';
        this.color = '';
        this.modalClasses = {
            modal: true,
            fade: true
        };
        this.dialogClasses = {
            'modal-dialog': true
        };
        this.backdropClasses = {
            'modal-backdrop': true,
            fade: true,
            show: true
        };
        this.visible = false;
        this.visibleAnimate = false;
    }
    ngOnInit() {
        this.setClasses();
    }
    ngOnDestroy() {
        this.hide();
    }
    setClasses() {
        this.dialogClasses[`modal-${this.size}`] = !!this.size;
        this.dialogClasses[`modal-${this.color}`] = !!this.color;
    }
    show() {
        this.renderer.addClass(this.document.body, 'modal-open');
        this.visible = true;
        this.modalClasses[`d-block`] = true;
        setTimeout(() => {
            this.visibleAnimate = true;
        });
    }
    hide() {
        this.renderer.removeClass(this.document.body, 'modal-open');
        this.visibleAnimate = false;
        setTimeout(() => {
            this.visible = false;
            this.modalClasses[`d-block`] = false;
            this.changeDetectorRef.markForCheck();
        }, 200);
    }
    onContainerClicked(event) {
        if (event.target.classList.contains('modal') && this.isTopmost() && this.closeOnBackdrop) {
            this.hide();
        }
    }
    onKeyDownHandler(event) {
        if (event.key === 'Escape' && this.isTopmost()) {
            this.hide();
        }
    }
    isTopmost() {
        const isTopmost = !this.elementRef.nativeElement.querySelector(':scope c-modal > .modal');
        return isTopmost;
    }
}
ModalComponent.ɵfac = function ModalComponent_Factory(t) { return new (t || ModalComponent)(ɵngcc0.ɵɵdirectiveInject(DOCUMENT), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef)); };
ModalComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ModalComponent, selectors: [["c-modal"]], contentQueries: function ModalComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, _c0, 3);
        ɵngcc0.ɵɵcontentQuery(dirIndex, _c1, 3);
        ɵngcc0.ɵɵcontentQuery(dirIndex, _c2, 3);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.header = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.body = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.footer = _t.first);
    } }, hostBindings: function ModalComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function ModalComponent_click_HostBindingHandler($event) { return ctx.onContainerClicked($event); })("keydown", function ModalComponent_keydown_HostBindingHandler($event) { return ctx.onKeyDownHandler($event); }, false, ɵngcc0.ɵɵresolveDocument);
    } }, inputs: { closeOnBackdrop: "closeOnBackdrop", size: "size", color: "color" }, decls: 2, vars: 2, consts: [["role", "dialog", "tabindex", "-1", 3, "show", "ngClass", 4, "ngIf"], [3, "ngClass", 4, "ngIf"], ["role", "dialog", "tabindex", "-1", 3, "ngClass"], ["role", "document", 3, "ngClass"], [1, "modal-content"], [1, "modal-header"], ["class", "modal-title", 4, "ngTemplateOutlet"], ["data-dismiss", "modal", "type", "button", "aria-label", "Close", 1, "close", 3, "click"], [1, "modal-body"], [4, "ngTemplateOutlet"], [1, "modal-footer"], [1, "modal-title"], [3, "ngClass"]], template: function ModalComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, ModalComponent_div_0_Template, 11, 7, "div", 0);
        ɵngcc0.ɵɵtemplate(1, ModalComponent_div_1_Template, 1, 1, "div", 1);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.visible);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.visibleAnimate);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgClass, ɵngcc1.NgTemplateOutlet], styles: [""] });
ModalComponent.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [DOCUMENT,] }] },
    { type: Renderer2 },
    { type: ElementRef },
    { type: ChangeDetectorRef }
];
ModalComponent.propDecorators = {
    header: [{ type: ContentChild, args: ['modalHeader', { static: true },] }],
    body: [{ type: ContentChild, args: ['modalBody', { static: true },] }],
    footer: [{ type: ContentChild, args: ['modalFooter', { static: true },] }],
    closeOnBackdrop: [{ type: Input }],
    size: [{ type: Input }],
    color: [{ type: Input }],
    onContainerClicked: [{ type: HostListener, args: ['click', ['$event'],] }],
    onKeyDownHandler: [{ type: HostListener, args: ['document:keydown', ['$event'],] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ModalComponent, [{
        type: Component,
        args: [{
                selector: 'c-modal',
                template: "<div\n  role=\"dialog\"\n  tabindex=\"-1\"\n  [class.show]=\"visibleAnimate\"\n  [ngClass]=\"modalClasses\"\n  *ngIf=\"visible\">\n  <div [ngClass]=\"dialogClasses\" role=\"document\">\n    <div class=\"modal-content\">\n      <div class=\"modal-header\">\n        <ng-container *ngTemplateOutlet=\"header\" class=\"modal-title\"></ng-container>\n        <button class=\"close\" data-dismiss=\"modal\" type=\"button\" aria-label=\"Close\" (click)=\"hide()\">\u00D7</button>\n      </div>\n      <div class=\"modal-body\">\n        <ng-container *ngTemplateOutlet=\"body\"></ng-container>\n      </div>\n      <div class=\"modal-footer\">\n        <ng-container *ngTemplateOutlet=\"footer\"></ng-container>\n      </div>\n    </div>\n  </div>\n</div>\n<div\n  *ngIf=\"visibleAnimate\"\n  [ngClass]=\"backdropClasses\">\n</div>\n",
                styles: [""]
            }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [DOCUMENT]
            }] }, { type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ElementRef }, { type: ɵngcc0.ChangeDetectorRef }]; }, { closeOnBackdrop: [{
            type: Input
        }], size: [{
            type: Input
        }], color: [{
            type: Input
        }], onContainerClicked: [{
            type: HostListener,
            args: ['click', ['$event']]
        }], onKeyDownHandler: [{
            type: HostListener,
            args: ['document:keydown', ['$event']]
        }], header: [{
            type: ContentChild,
            args: ['modalHeader', { static: true }]
        }], body: [{
            type: ContentChild,
            args: ['modalBody', { static: true }]
        }], footer: [{
            type: ContentChild,
            args: ['modalFooter', { static: true }]
        }] }); })();
