// @coreui/angular/esm2015/lib/widget/widget-dropdown/widget-dropdown.component.js
import { Component, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*", [["", "slot", "footer"]]];
const _c1 = ["*", "[slot=footer]"];
export class WidgetDropdownComponent {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames({
            card: true,
            'text-white': true,
            [`bg-${this.color}`]: !!this.color,
        });
        return classes;
    }
    ngOnInit() { }
}
WidgetDropdownComponent.ɵfac = function WidgetDropdownComponent_Factory(t) { return new (t || WidgetDropdownComponent)(); };
WidgetDropdownComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: WidgetDropdownComponent, selectors: [["c-widget-dropdown"]], hostVars: 2, hostBindings: function WidgetDropdownComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { color: "color", header: "header", text: "text" }, ngContentSelectors: _c1, decls: 9, vars: 2, consts: [[1, "card-body", "pb-0", "d-flex", "justify-content-between", "align-items-start"], ["v-if", "header", 1, "text-value-lg"], ["v-if", "text"]], template: function WidgetDropdownComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵelementContainerStart(0);
        ɵngcc0.ɵɵelementStart(1, "div", 0);
        ɵngcc0.ɵɵelementStart(2, "div");
        ɵngcc0.ɵɵelementStart(3, "div", 1);
        ɵngcc0.ɵɵtext(4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 2);
        ɵngcc0.ɵɵtext(6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵprojection(7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵprojection(8, 1);
        ɵngcc0.ɵɵelementContainerEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate(ctx.header);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx.text);
    } }, styles: [""] });
WidgetDropdownComponent.ctorParameters = () => [];
WidgetDropdownComponent.propDecorators = {
    color: [{ type: Input }],
    header: [{ type: Input }],
    text: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(WidgetDropdownComponent, [{
        type: Component,
        args: [{
                selector: 'c-widget-dropdown',
                template: "<ng-container>\n  <div class=\"card-body pb-0 d-flex justify-content-between align-items-start\">\n    <div>\n      <div v-if=\"header\" class=\"text-value-lg\">{{header}}</div>\n      <div v-if=\"text\">{{text}}</div>\n    </div>\n    <ng-content></ng-content>\n<!--    <ng-content select=\"[dropdown]\" ></ng-content>-->\n  </div>\n  <ng-content select=\"[slot=footer]\"></ng-content>\n</ng-container>\n",
                styles: [""]
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], color: [{
            type: Input
        }], header: [{
            type: Input
        }], text: [{
            type: Input
        }] }); })();
