// @coreui/angular/esm2015/lib/form/form/form.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import { chkBoolProp } from '../../shared/check-bool-prop';
import * as ɵngcc0 from '@angular/core';
export class FormDirective {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = {
            'form-inline': chkBoolProp(this.inline),
            'was-validated': chkBoolProp(this.wasValidated),
        };
        return classes;
    }
}
FormDirective.ɵfac = function FormDirective_Factory(t) { return new (t || FormDirective)(); };
FormDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: FormDirective, selectors: [["", "cForm", ""]], hostVars: 2, hostBindings: function FormDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { inline: "inline", wasValidated: "wasValidated" } });
FormDirective.ctorParameters = () => [];
FormDirective.propDecorators = {
    inline: [{ type: Input }],
    wasValidated: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FormDirective, [{
        type: Directive,
        args: [{
                selector: '[cForm]'
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], inline: [{
            type: Input
        }], wasValidated: [{
            type: Input
        }] }); })();
