// @coreui/angular/esm2015/lib/nav/nav-item.component.js
import { Component, HostBinding } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class NavItemComponent {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames({
            'nav-item': true,
        });
        return classes;
    }
    ngOnInit() {
    }
}
NavItemComponent.ɵfac = function NavItemComponent_Factory(t) { return new (t || NavItemComponent)(); };
NavItemComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NavItemComponent, selectors: [["c-nav-item"]], hostVars: 2, hostBindings: function NavItemComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function NavItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]{display:list-item;text-align:match-parent}"] });
NavItemComponent.ctorParameters = () => [];
NavItemComponent.propDecorators = {
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NavItemComponent, [{
        type: Component,
        args: [{
                selector: 'c-nav-item',
                template: "<ng-content></ng-content>\n",
                styles: [":host{display:list-item;text-align:match-parent}"]
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }] }); })();
