// @coreui/angular/esm2015/lib/nav/nav-link.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';
export class NavLinkDirective {
    constructor() {
        this.disabled = null;
        this.tabindex = null;
    }
    get isDisabled() {
        return (this.disabled !== 'false' && this.disabled !== false && this.disabled !== null) ? this.disabled : null;
    }
    get getTabindex() {
        return this.disabled ? null : this.tabindex != null ? this.tabindex : '0';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames({
            'nav-link': true,
            disabled: this.disabled,
            active: this.active
        });
        return classes;
    }
}
NavLinkDirective.ɵfac = function NavLinkDirective_Factory(t) { return new (t || NavLinkDirective)(); };
NavLinkDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NavLinkDirective, selectors: [["", "cNavLink", ""]], hostVars: 4, hostBindings: function NavLinkDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("aria-disabled", ctx.isDisabled)("tabindex", ctx.getTabindex);
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { disabled: "disabled", tabindex: "tabindex", active: "active" } });
NavLinkDirective.ctorParameters = () => [];
NavLinkDirective.propDecorators = {
    disabled: [{ type: Input }],
    tabindex: [{ type: Input }],
    active: [{ type: Input }],
    isDisabled: [{ type: HostBinding, args: ['attr.aria-disabled',] }],
    getTabindex: [{ type: HostBinding, args: ['attr.tabindex',] }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NavLinkDirective, [{
        type: Directive,
        args: [{
                selector: '[cNavLink]'
            }]
    }], function () { return []; }, { disabled: [{
            type: Input
        }], tabindex: [{
            type: Input
        }], isDisabled: [{
            type: HostBinding,
            args: ['attr.aria-disabled']
        }], getTabindex: [{
            type: HostBinding,
            args: ['attr.tabindex']
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], active: [{
            type: Input
        }] }); })();
