// @coreui/angular/esm2015/lib/nav/nav.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NavComponent } from './nav.component';
import { NavItemComponent } from './nav-item.component';
import { NavLinkDirective } from './nav-link.directive';
import * as ɵngcc0 from '@angular/core';
export class NavModule {
}
NavModule.ɵfac = function NavModule_Factory(t) { return new (t || NavModule)(); };
NavModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NavModule });
NavModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NavModule, { declarations: function () { return [NavComponent, NavItemComponent, NavLinkDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NavComponent, NavItemComponent, NavLinkDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NavModule, [{
        type: NgModule,
        args: [{
                declarations: [NavComponent, NavItemComponent, NavLinkDirective],
                imports: [
                    CommonModule
                ],
                exports: [NavComponent, NavItemComponent, NavLinkDirective]
            }]
    }], null, null); })();
