// @coreui/angular/esm2015/lib/nav/nav.component.js
import { Component, HostBinding, Input, } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class NavComponent {
    constructor() {
        this.variant = '';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const verticalSuffix = this.vertical === true ? '' : `-${this.vertical}`;
        const classes = classNames({
            nav: true,
            [`nav-${this.variant}`]: this.variant,
            [`flex${verticalSuffix}-column`]: this.vertical,
            'nav-fill': this.fill,
            'nav-justified': this.justified,
            [`card-header-${this.variant}`]: this.inCard && this.variant
        });
        return classes;
    }
    ngOnInit() { }
}
NavComponent.ɵfac = function NavComponent_Factory(t) { return new (t || NavComponent)(); };
NavComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NavComponent, selectors: [["c-nav"]], hostVars: 2, hostBindings: function NavComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { variant: "variant", fill: "fill", justified: "justified", vertical: "vertical", inCard: "inCard" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function NavComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%] > .nav-link[_ngcontent-%COMP%]:focus{outline:0}"] });
NavComponent.ctorParameters = () => [];
NavComponent.propDecorators = {
    variant: [{ type: Input }],
    fill: [{ type: Input }],
    justified: [{ type: Input }],
    vertical: [{ type: Input }],
    inCard: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NavComponent, [{
        type: Component,
        args: [{
                selector: 'c-nav',
                template: "<ng-content></ng-content>\n",
                styles: [":host>.nav-link:focus{outline:0}"]
            }]
    }], function () { return []; }, { variant: [{
            type: Input
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], fill: [{
            type: Input
        }], justified: [{
            type: Input
        }], vertical: [{
            type: Input
        }], inCard: [{
            type: Input
        }] }); })();
