// @coreui/angular/esm2015/lib/layout/subheader/subheader.component.js
import { Component, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class SubheaderComponent {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames(this.ngClass, this.class, {
            'c-subheader': true,
        });
        return classes;
    }
    ngOnInit() {
    }
}
SubheaderComponent.ɵfac = function SubheaderComponent_Factory(t) { return new (t || SubheaderComponent)(); };
SubheaderComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SubheaderComponent, selectors: [["c-subheader"], ["", "c-subheader", ""]], hostVars: 2, hostBindings: function SubheaderComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { class: "class", ngClass: "ngClass" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function SubheaderComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
SubheaderComponent.ctorParameters = () => [];
SubheaderComponent.propDecorators = {
    class: [{ type: Input }],
    ngClass: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SubheaderComponent, [{
        type: Component,
        args: [{
                selector: 'c-subheader, [c-subheader]',
                template: `<ng-content></ng-content>`
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], class: [{
            type: Input
        }], ngClass: [{
            type: Input
        }] }); })();
