// @coreui/angular/esm2015/lib/carousel/carousel-item/carousel-item.component.js
import { Component, HostBinding, Input, } from '@angular/core';
import classNames from 'classnames';
import { CarouselService } from '../carousel.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../carousel.service';
import * as ɵngcc2 from '@angular/common';

function CarouselItemComponent_ng_content_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0, 0, ["*ngIf", "active"]);
} }
const _c0 = ["*"];
export class CarouselItemComponent {
    constructor(carouselService) {
        this.carouselService = carouselService;
    }
    set active(active) {
        this._active = active;
    }
    get active() {
        return this._active;
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames({
            'carousel-item': true,
            active: this.active,
        });
        return classes;
    }
    ngOnInit() { }
    ngOnDestroy() {
        this.carouselStateSubscribe(false);
    }
    ngAfterViewInit() {
        setTimeout(() => {
            this.carouselStateSubscribe();
        });
    }
    carouselStateSubscribe(subscribe = true) {
        if (subscribe) {
            this.carouselIndexSubscription = this.carouselService.carouselIndex$.subscribe((nextIndex) => {
                if ('active' in nextIndex) {
                    this.active = nextIndex.active === this.index;
                }
            });
        }
        else {
            this.carouselIndexSubscription.unsubscribe();
        }
    }
}
CarouselItemComponent.ɵfac = function CarouselItemComponent_Factory(t) { return new (t || CarouselItemComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CarouselService)); };
CarouselItemComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CarouselItemComponent, selectors: [["c-carousel-item"]], hostVars: 2, hostBindings: function CarouselItemComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { active: "active" }, ngContentSelectors: _c0, decls: 1, vars: 1, consts: [[4, "ngIf"]], template: function CarouselItemComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵtemplate(0, CarouselItemComponent_ng_content_0_Template, 1, 0, "ng-content", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.active);
    } }, directives: [ɵngcc2.NgIf], styles: ["[_nghost-%COMP%]{display:block}"] });
CarouselItemComponent.ctorParameters = () => [
    { type: CarouselService }
];
CarouselItemComponent.propDecorators = {
    active: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CarouselItemComponent, [{
        type: Component,
        args: [{
                selector: 'c-carousel-item',
                template: "<ng-content *ngIf=\"active\"></ng-content>\n",
                styles: [":host{display:block}"]
            }]
    }], function () { return [{ type: ɵngcc1.CarouselService }]; }, { active: [{
            type: Input
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }] }); })();
