// @coreui/angular/esm2015/lib/embed/embed.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { EmbedComponent } from './embed.component';
import { EmbedItemDirective } from './embed-item.directive';
import * as ɵngcc0 from '@angular/core';
export class EmbedModule {
}
EmbedModule.ɵfac = function EmbedModule_Factory(t) { return new (t || EmbedModule)(); };
EmbedModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: EmbedModule });
EmbedModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(EmbedModule, { declarations: function () { return [EmbedComponent, EmbedItemDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [EmbedComponent, EmbedItemDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(EmbedModule, [{
        type: NgModule,
        args: [{
                declarations: [EmbedComponent, EmbedItemDirective],
                imports: [CommonModule],
                exports: [EmbedComponent, EmbedItemDirective]
            }]
    }], null, null); })();
