// @coreui/angular/esm2015/lib/embed/embed.component.js
import { Component, ContentChild, HostBinding, Input, } from '@angular/core';
import classNames from 'classnames';
import { EmbedItemDirective } from './embed-item.directive';
import * as ɵngcc0 from '@angular/core';

const _c0 = [[["iframe"]]];
const _c1 = ["iframe"];
export class EmbedComponent {
    constructor() {
        this.ratio = '16by9';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames({
            'embed-responsive': true,
            [`embed-responsive-${this.ratio}`]: !!this.ratio,
        });
        return classes;
    }
    ngOnInit() { }
    ngAfterContentInit() {
    }
}
EmbedComponent.ɵfac = function EmbedComponent_Factory(t) { return new (t || EmbedComponent)(); };
EmbedComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: EmbedComponent, selectors: [["c-embed"]], contentQueries: function EmbedComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, EmbedItemDirective, 1);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.embedItem = _t.first);
    } }, hostVars: 2, hostBindings: function EmbedComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { ratio: "ratio" }, ngContentSelectors: _c1, decls: 1, vars: 0, template: function EmbedComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
EmbedComponent.ctorParameters = () => [];
EmbedComponent.propDecorators = {
    ratio: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }],
    embedItem: [{ type: ContentChild, args: [EmbedItemDirective,] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(EmbedComponent, [{
        type: Component,
        args: [{
                selector: 'c-embed',
                template: "<ng-content select=\"iframe\"></ng-content>\n",
                styles: [""]
            }]
    }], function () { return []; }, { ratio: [{
            type: Input
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], embedItem: [{
            type: ContentChild,
            args: [EmbedItemDirective]
        }] }); })();
