// @coreui/angular/esm2015/lib/embed/embed-item.directive.js
import { Directive, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class EmbedItemDirective {
    constructor() {
        this.embedItemClass = true;
    }
}
EmbedItemDirective.ɵfac = function EmbedItemDirective_Factory(t) { return new (t || EmbedItemDirective)(); };
EmbedItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: EmbedItemDirective, selectors: [["", "cEmbedItem", ""]], hostVars: 2, hostBindings: function EmbedItemDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("embed-responsive-item", ctx.embedItemClass);
    } } });
EmbedItemDirective.ctorParameters = () => [];
EmbedItemDirective.propDecorators = {
    embedItemClass: [{ type: HostBinding, args: ['class.embed-responsive-item',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(EmbedItemDirective, [{
        type: Directive,
        args: [{
                selector: '[cEmbedItem]'
            }]
    }], function () { return []; }, { embedItemClass: [{
            type: HostBinding,
            args: ['class.embed-responsive-item']
        }] }); })();
