// @pascalhonegger/ng-datatable/bundles/pascalhonegger-ng-datatable.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/common'),exports, require('@angular/core'), require('rxjs'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('@pascalhonegger/ng-datatable', ['@angular/core','@angular/common','exports', '@angular/core', 'rxjs', '@angular/common'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global.ng.core,global.ng.common,(global.pascalhonegger = global.pascalhonegger || {}, global.pascalhonegger['ng-datatable'] = {}), global.ng.core, global.rxjs, global.ng.common));
}(this, (function (ɵngcc0,ɵngcc1,exports, core, rxjs, common) { 
function BootstrapPaginator_ul_2_li_4_Template(rf, ctx) { if (rf & 1) {
    var _r12 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_2_li_4_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r12); ɵngcc0.ɵɵnextContext(2); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setPage(_r0.activePage - 4); });
    ɵngcc0.ɵɵelementStart(1, "a", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext(2);
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(_r0.activePage - 4);
} }
function BootstrapPaginator_ul_2_li_5_Template(rf, ctx) { if (rf & 1) {
    var _r14 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_2_li_5_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r14); ɵngcc0.ɵɵnextContext(2); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setPage(_r0.activePage - 3); });
    ɵngcc0.ɵɵelementStart(1, "a", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext(2);
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(_r0.activePage - 3);
} }
function BootstrapPaginator_ul_2_li_6_Template(rf, ctx) { if (rf & 1) {
    var _r16 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_2_li_6_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r16); ɵngcc0.ɵɵnextContext(2); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setPage(_r0.activePage - 2); });
    ɵngcc0.ɵɵelementStart(1, "a", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext(2);
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(_r0.activePage - 2);
} }
function BootstrapPaginator_ul_2_li_7_Template(rf, ctx) { if (rf & 1) {
    var _r18 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_2_li_7_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r18); ɵngcc0.ɵɵnextContext(2); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setPage(_r0.activePage - 1); });
    ɵngcc0.ɵɵelementStart(1, "a", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext(2);
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(_r0.activePage - 1);
} }
function BootstrapPaginator_ul_2_li_11_Template(rf, ctx) { if (rf & 1) {
    var _r20 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_2_li_11_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r20); ɵngcc0.ɵɵnextContext(2); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setPage(_r0.activePage + 1); });
    ɵngcc0.ɵɵelementStart(1, "a", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext(2);
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(_r0.activePage + 1);
} }
function BootstrapPaginator_ul_2_li_12_Template(rf, ctx) { if (rf & 1) {
    var _r22 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_2_li_12_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r22); ɵngcc0.ɵɵnextContext(2); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setPage(_r0.activePage + 2); });
    ɵngcc0.ɵɵelementStart(1, "a", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext(2);
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(_r0.activePage + 2);
} }
function BootstrapPaginator_ul_2_li_13_Template(rf, ctx) { if (rf & 1) {
    var _r24 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_2_li_13_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r24); ɵngcc0.ɵɵnextContext(2); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setPage(_r0.activePage + 3); });
    ɵngcc0.ɵɵelementStart(1, "a", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext(2);
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(_r0.activePage + 3);
} }
function BootstrapPaginator_ul_2_li_14_Template(rf, ctx) { if (rf & 1) {
    var _r26 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_2_li_14_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r26); ɵngcc0.ɵɵnextContext(2); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setPage(_r0.activePage + 4); });
    ɵngcc0.ɵɵelementStart(1, "a", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext(2);
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(_r0.activePage + 4);
} }
function BootstrapPaginator_ul_2_Template(rf, ctx) { if (rf & 1) {
    var _r28 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "ul", 4);
    ɵngcc0.ɵɵelementStart(1, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_2_Template_li_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r28); ɵngcc0.ɵɵnextContext(); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setPage(1); });
    ɵngcc0.ɵɵelementStart(2, "a", 6);
    ɵngcc0.ɵɵtext(3, "\u00AB");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(4, BootstrapPaginator_ul_2_li_4_Template, 3, 1, "li", 7);
    ɵngcc0.ɵɵtemplate(5, BootstrapPaginator_ul_2_li_5_Template, 3, 1, "li", 7);
    ɵngcc0.ɵɵtemplate(6, BootstrapPaginator_ul_2_li_6_Template, 3, 1, "li", 7);
    ɵngcc0.ɵɵtemplate(7, BootstrapPaginator_ul_2_li_7_Template, 3, 1, "li", 7);
    ɵngcc0.ɵɵelementStart(8, "li", 8);
    ɵngcc0.ɵɵelementStart(9, "a", 6);
    ɵngcc0.ɵɵtext(10);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(11, BootstrapPaginator_ul_2_li_11_Template, 3, 1, "li", 7);
    ɵngcc0.ɵɵtemplate(12, BootstrapPaginator_ul_2_li_12_Template, 3, 1, "li", 7);
    ɵngcc0.ɵɵtemplate(13, BootstrapPaginator_ul_2_li_13_Template, 3, 1, "li", 7);
    ɵngcc0.ɵɵtemplate(14, BootstrapPaginator_ul_2_li_14_Template, 3, 1, "li", 7);
    ɵngcc0.ɵɵelementStart(15, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_2_Template_li_click_15_listener() { ɵngcc0.ɵɵrestoreView(_r28); ɵngcc0.ɵɵnextContext(); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setPage(_r0.lastPage); });
    ɵngcc0.ɵɵelementStart(16, "a", 6);
    ɵngcc0.ɵɵtext(17, "\u00BB");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    ɵngcc0.ɵɵnextContext();
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassProp("disabled", _r0.activePage <= 1);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", _r0.activePage > 4 && _r0.activePage + 1 > _r0.lastPage);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", _r0.activePage > 3 && _r0.activePage + 2 > _r0.lastPage);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", _r0.activePage > 2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", _r0.activePage > 1);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(_r0.activePage);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", _r0.activePage + 1 <= _r0.lastPage);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", _r0.activePage + 2 <= _r0.lastPage);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", _r0.activePage + 3 <= _r0.lastPage && _r0.activePage < 3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", _r0.activePage + 4 <= _r0.lastPage && _r0.activePage < 2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassProp("disabled", _r0.activePage >= _r0.lastPage);
} }
function BootstrapPaginator_ul_3_li_1_Template(rf, ctx) { if (rf & 1) {
    var _r33 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 5);
    ɵngcc0.ɵɵlistener("click", function BootstrapPaginator_ul_3_li_1_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r33); var rows_r31 = ctx.$implicit; ɵngcc0.ɵɵnextContext(2); var _r0 = ɵngcc0.ɵɵreference(1); return _r0.setRowsOnPage(rows_r31); });
    ɵngcc0.ɵɵelementStart(1, "a", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var rows_r31 = ctx.$implicit;
    ɵngcc0.ɵɵnextContext(2);
    var _r0 = ɵngcc0.ɵɵreference(1);
    ɵngcc0.ɵɵclassProp("active", _r0.rowsOnPage === rows_r31);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(rows_r31);
} }
function BootstrapPaginator_ul_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ul", 9);
    ɵngcc0.ɵɵtemplate(1, BootstrapPaginator_ul_3_li_1_Template, 3, 3, "li", 10);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r2.rowsOnPageSet);
} }
var _c0 = ["*"];
function DefaultSorter_span_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 3);
} }
function DefaultSorter_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 4);
} }
'use strict';

    var BootstrapPaginator = /** @class */ (function () {
        function BootstrapPaginator() {
            this.rowsOnPageSet = [];
            this.minRowsOnPage = 0;
        }
        BootstrapPaginator.prototype.ngOnChanges = function (changes) {
            if (changes.rowsOnPageSet) {
                this.minRowsOnPage = this.rowsOnPageSet.reduce(function (previous, current) { return current < previous ? current : previous; });
            }
        };
BootstrapPaginator.ɵfac = function BootstrapPaginator_Factory(t) { return new (t || BootstrapPaginator)(); };
BootstrapPaginator.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: BootstrapPaginator, selectors: [["mfBootstrapPaginator"]], inputs: { rowsOnPageSet: "rowsOnPageSet", mfTable: "mfTable" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], decls: 4, vars: 3, consts: [[3, "mfTable"], ["p", ""], ["class", "pagination", 4, "ngIf"], ["class", "pagination pull-right float-sm-right", 4, "ngIf"], [1, "pagination"], [1, "page-item", 3, "click"], [1, "page-link", 2, "cursor", "pointer"], ["class", "page-item", 3, "click", 4, "ngIf"], [1, "page-item", "active"], [1, "pagination", "pull-right", "float-sm-right"], ["class", "page-item", 3, "active", "click", 4, "ngFor", "ngForOf"]], template: function BootstrapPaginator_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "mfPaginator", 0, 1);
        ɵngcc0.ɵɵtemplate(2, BootstrapPaginator_ul_2_Template, 18, 13, "ul", 2);
        ɵngcc0.ɵɵtemplate(3, BootstrapPaginator_ul_3_Template, 2, 1, "ul", 3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        var _r0 = ɵngcc0.ɵɵreference(1);
        ɵngcc0.ɵɵproperty("mfTable", ctx.mfTable);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", _r0.dataLength > _r0.rowsOnPage);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", _r0.dataLength > ctx.minRowsOnPage);
    } }, directives: function () { return [Paginator, ɵngcc1.NgIf, ɵngcc1.NgForOf]; }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BootstrapPaginator, [{
        type: core.Component,
        args: [{
                selector: "mfBootstrapPaginator",
                template: "\n    <mfPaginator #p [mfTable]=\"mfTable\">\n        <ul class=\"pagination\" *ngIf=\"p.dataLength > p.rowsOnPage\">\n            <li class=\"page-item\" [class.disabled]=\"p.activePage <= 1\" (click)=\"p.setPage(1)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">&laquo;</a>\n            </li>\n            <li class=\"page-item\" *ngIf=\"p.activePage > 4 && p.activePage + 1 > p.lastPage\" (click)=\"p.setPage(p.activePage - 4)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">{{p.activePage-4}}</a>\n            </li>\n            <li class=\"page-item\" *ngIf=\"p.activePage > 3 && p.activePage + 2 > p.lastPage\" (click)=\"p.setPage(p.activePage - 3)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">{{p.activePage-3}}</a>\n            </li>\n            <li class=\"page-item\" *ngIf=\"p.activePage > 2\" (click)=\"p.setPage(p.activePage - 2)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">{{p.activePage-2}}</a>\n            </li>\n            <li class=\"page-item\" *ngIf=\"p.activePage > 1\" (click)=\"p.setPage(p.activePage - 1)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">{{p.activePage-1}}</a>\n            </li>\n            <li class=\"page-item active\">\n                <a class=\"page-link\" style=\"cursor: pointer\">{{p.activePage}}</a>\n            </li>\n            <li class=\"page-item\" *ngIf=\"p.activePage + 1 <= p.lastPage\" (click)=\"p.setPage(p.activePage + 1)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">{{p.activePage+1}}</a>\n            </li>\n            <li class=\"page-item\" *ngIf=\"p.activePage + 2 <= p.lastPage\" (click)=\"p.setPage(p.activePage + 2)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">{{p.activePage+2}}</a>\n            </li>\n            <li class=\"page-item\" *ngIf=\"p.activePage + 3 <= p.lastPage && p.activePage < 3\" (click)=\"p.setPage(p.activePage + 3)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">{{p.activePage+3}}</a>\n            </li>\n            <li class=\"page-item\" *ngIf=\"p.activePage + 4 <= p.lastPage && p.activePage < 2\" (click)=\"p.setPage(p.activePage + 4)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">{{p.activePage+4}}</a>\n            </li>\n            <li class=\"page-item\" [class.disabled]=\"p.activePage >= p.lastPage\" (click)=\"p.setPage(p.lastPage)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">&raquo;</a>\n            </li>\n        </ul>\n        <ul class=\"pagination pull-right float-sm-right\" *ngIf=\"p.dataLength > minRowsOnPage\">\n            <li class=\"page-item\" *ngFor=\"let rows of rowsOnPageSet\" [class.active]=\"p.rowsOnPage===rows\" (click)=\"p.setRowsOnPage(rows)\">\n                <a class=\"page-link\" style=\"cursor: pointer\">{{rows}}</a>\n            </li>\n        </ul>\n    </mfPaginator>\n    "
            }]
    }], function () { return []; }, { rowsOnPageSet: [{
            type: core.Input,
            args: ["rowsOnPageSet"]
        }], mfTable: [{
            type: core.Input,
            args: ["mfTable"]
        }] }); })();
        return BootstrapPaginator;
    }());
    BootstrapPaginator.propDecorators = {
        rowsOnPageSet: [{ type: core.Input, args: ["rowsOnPageSet",] }],
        mfTable: [{ type: core.Input, args: ["mfTable",] }]
    };

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation.

    Permission to use, copy, modify, and/or distribute this software for any
    purpose with or without fee is hereby granted.

    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
    REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
    AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
    INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
    LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
    OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
    PERFORMANCE OF THIS SOFTWARE.
    ***************************************************************************** */
    /* global Reflect, Promise */
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b)
                if (b.hasOwnProperty(p))
                    d[p] = b[p]; };
        return extendStatics(d, b);
    };
    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }
    var __assign = function () {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s)
                    if (Object.prototype.hasOwnProperty.call(s, p))
                        t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
    function __rest(s, e) {
        var t = {};
        for (var p in s)
            if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
                t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }
    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
            r = Reflect.decorate(decorators, target, key, desc);
        else
            for (var i = decorators.length - 1; i >= 0; i--)
                if (d = decorators[i])
                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }
    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); };
    }
    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function")
            return Reflect.metadata(metadataKey, metadataValue);
    }
    function __awaiter(thisArg, _arguments, P, generator) {
        function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }
    var __createBinding = Object.create ? (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        Object.defineProperty(o, k2, { enumerable: true, get: function () { return m[k]; } });
    }) : (function (o, m, k, k2) {
        if (k2 === undefined)
            k2 = k;
        o[k2] = m[k];
    });
    function __exportStar(m, exports) {
        for (var p in m)
            if (p !== "default" && !exports.hasOwnProperty(p))
                __createBinding(exports, m, p);
    }
    function __values(o) {
        var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
        if (m)
            return m.call(o);
        if (o && typeof o.length === "number")
            return {
                next: function () {
                    if (o && i >= o.length)
                        o = void 0;
                    return { value: o && o[i++], done: !o };
                }
            };
        throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
    }
    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m)
            return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done)
                ar.push(r.value);
        }
        catch (error) {
            e = { error: error };
        }
        finally {
            try {
                if (r && !r.done && (m = i["return"]))
                    m.call(i);
            }
            finally {
                if (e)
                    throw e.error;
            }
        }
        return ar;
    }
    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }
    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++)
            s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    }
    ;
    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }
    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n])
            i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try {
            step(g[n](v));
        }
        catch (e) {
            settle(q[0][3], e);
        } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length)
            resume(q[0][0], q[0][1]); }
    }
    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }
    function __asyncValues(o) {
        if (!Symbol.asyncIterator)
            throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function (v) { resolve({ value: v, done: d }); }, reject); }
    }
    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) {
            Object.defineProperty(cooked, "raw", { value: raw });
        }
        else {
            cooked.raw = raw;
        }
        return cooked;
    }
    ;
    var __setModuleDefault = Object.create ? (function (o, v) {
        Object.defineProperty(o, "default", { enumerable: true, value: v });
    }) : function (o, v) {
        o["default"] = v;
    };
    function __importStar(mod) {
        if (mod && mod.__esModule)
            return mod;
        var result = {};
        if (mod != null)
            for (var k in mod)
                if (Object.hasOwnProperty.call(mod, k))
                    __createBinding(result, mod, k);
        __setModuleDefault(result, mod);
        return result;
    }
    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }
    function __classPrivateFieldGet(receiver, privateMap) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to get private field on non-instance");
        }
        return privateMap.get(receiver);
    }
    function __classPrivateFieldSet(receiver, privateMap, value) {
        if (!privateMap.has(receiver)) {
            throw new TypeError("attempted to set private field on non-instance");
        }
        privateMap.set(receiver, value);
        return value;
    }

    var DataTable = /** @class */ (function () {
        function DataTable(differs) {
            this.differs = differs;
            this.inputData = [];
            this.sortBy = "";
            this.sortOrder = "asc";
            this.sortByChange = new core.EventEmitter();
            this.sortOrderChange = new core.EventEmitter();
            this.rowsOnPage = 1000;
            this.activePage = 1;
            this.mustRecalculateData = false;
            this.onSortChange = new rxjs.ReplaySubject(1);
            this.onPageChange = new core.EventEmitter();
            this.diff = differs.find([]).create(null);
        }
        DataTable.prototype.getSort = function () {
            return { sortBy: this.sortBy, sortOrder: this.sortOrder };
        };
        DataTable.prototype.setSort = function (sortBy, sortOrder) {
            if (this.sortBy !== sortBy || this.sortOrder !== sortOrder) {
                this.sortBy = sortBy;
                this.sortOrder = ["asc", "desc"].indexOf(sortOrder) >= 0 ? sortOrder : "asc";
                this.mustRecalculateData = true;
                this.onSortChange.next({ sortBy: this.sortBy, sortOrder: this.sortOrder });
                this.sortByChange.emit(this.sortBy);
                this.sortOrderChange.emit(this.sortOrder);
            }
        };
        DataTable.prototype.getPage = function () {
            return { activePage: this.activePage, rowsOnPage: this.rowsOnPage, dataLength: this.inputData.length };
        };
        DataTable.prototype.setPage = function (activePage, rowsOnPage) {
            if (this.rowsOnPage !== rowsOnPage || this.activePage !== activePage) {
                this.activePage = this.activePage !== activePage ? activePage : this.calculateNewActivePage(this.rowsOnPage, rowsOnPage);
                this.rowsOnPage = rowsOnPage;
                this.mustRecalculateData = true;
                this.onPageChange.emit({
                    activePage: this.activePage,
                    rowsOnPage: this.rowsOnPage,
                    dataLength: this.inputData ? this.inputData.length : 0
                });
            }
        };
        DataTable.prototype.calculateNewActivePage = function (previousRowsOnPage, currentRowsOnPage) {
            var firstRowOnPage = (this.activePage - 1) * previousRowsOnPage + 1;
            var newActivePage = Math.ceil(firstRowOnPage / currentRowsOnPage);
            return newActivePage;
        };
        DataTable.prototype.recalculatePage = function () {
            var lastPage = Math.ceil(this.inputData.length / this.rowsOnPage);
            this.activePage = lastPage < this.activePage ? lastPage : this.activePage;
            this.activePage = this.activePage || 1;
            this.onPageChange.emit({
                activePage: this.activePage,
                rowsOnPage: this.rowsOnPage,
                dataLength: this.inputData.length
            });
        };
        DataTable.prototype.ngOnChanges = function (changes) {
            if (changes.rowsOnPage) {
                this.rowsOnPage = changes.rowsOnPage.previousValue;
                this.setPage(this.activePage, changes.rowsOnPage.currentValue);
                this.mustRecalculateData = true;
            }
            if (changes.sortBy || changes.sortOrder) {
                if (["asc", "desc"].indexOf(this.sortOrder) < 0) {
                    console.warn("ng-datatable: value for input mfSortOrder must be one of ['asc', 'desc'], but is:", this.sortOrder);
                    this.sortOrder = "asc";
                }
                if (this.sortBy) {
                    this.onSortChange.next({ sortBy: this.sortBy, sortOrder: this.sortOrder });
                }
                this.mustRecalculateData = true;
            }
            if (changes.inputData) {
                this.inputData = changes.inputData.currentValue || [];
                this.recalculatePage();
                this.mustRecalculateData = true;
            }
        };
        DataTable.prototype.ngDoCheck = function () {
            var changes = this.diff.diff(this.inputData);
            if (changes) {
                this.recalculatePage();
                this.mustRecalculateData = true;
            }
            if (this.mustRecalculateData) {
                this.fillData();
                this.mustRecalculateData = false;
            }
        };
        DataTable.prototype.fillData = function () {
            // this.activePage = this.activePage;
            // this.rowsOnPage = this.rowsOnPage;
            var offset = (this.activePage - 1) * this.rowsOnPage;
            // let data = this.inputData;
            // const sortBy = this.sortBy;
            // if (typeof sortBy === "string" || sortBy instanceof String) {
            //     data = orderBy(data, this.caseInsensitiveIteratee(sortBy as string), [this.sortOrder]);
            // } else {
            //     data = orderBy(data, sortBy, [this.sortOrder]);
            // }
            // data = slice(data, offset, offset + this.rowsOnPage);
            this.data = __spread(this.inputData).sort(this.sorter(this.sortBy, this.sortOrder))
                .slice(offset, offset + this.rowsOnPage);
        };
        DataTable.prototype.caseInsensitiveIteratee = function (sortBy) {
            return function (row) {
                var e_1, _a;
                var value = row;
                if (typeof sortBy === "string" || sortBy instanceof String) {
                    try {
                        for (var _b = __values(sortBy.split(".")), _c = _b.next(); !_c.done; _c = _b.next()) {
                            var sortByProperty = _c.value;
                            if (value) {
                                value = value[sortByProperty];
                            }
                        }
                    }
                    catch (e_1_1) { e_1 = { error: e_1_1 }; }
                    finally {
                        try {
                            if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
                        }
                        finally { if (e_1) throw e_1.error; }
                    }
                }
                else {
                    value = sortBy(value);
                }
                if (value && typeof value === "string" || value instanceof String) {
                    return value.toLowerCase();
                }
                return value;
            };
        };
        DataTable.prototype.compare = function (left, right) {
            return left === right ? 0 : left == null || left > right ? 1 : -1;
        };
        DataTable.prototype.sorter = function (sortBy, sortOrder) {
            var _this = this;
            var order = sortOrder === "desc" ? -1 : 1;
            if (Array.isArray(sortBy)) {
                var iteratees_1 = sortBy.map(function (entry) { return _this.caseInsensitiveIteratee(entry); });
                return function (left, right) {
                    var e_2, _a;
                    try {
                        for (var iteratees_2 = __values(iteratees_1), iteratees_2_1 = iteratees_2.next(); !iteratees_2_1.done; iteratees_2_1 = iteratees_2.next()) {
                            var iteratee = iteratees_2_1.value;
                            var comparison = _this.compare(iteratee(left), iteratee(right)) * order;
                            if (comparison !== 0) {
                                return comparison;
                            }
                        }
                    }
                    catch (e_2_1) { e_2 = { error: e_2_1 }; }
                    finally {
                        try {
                            if (iteratees_2_1 && !iteratees_2_1.done && (_a = iteratees_2.return)) _a.call(iteratees_2);
                        }
                        finally { if (e_2) throw e_2.error; }
                    }
                    return 0;
                };
            }
            else {
                var iteratee_1 = this.caseInsensitiveIteratee(sortBy);
                return function (left, right) { return _this.compare(iteratee_1(left), iteratee_1(right)) * order; };
            }
        };
DataTable.ɵfac = function DataTable_Factory(t) { return new (t || DataTable)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.IterableDiffers)); };
DataTable.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: DataTable, selectors: [["table", "mfData", ""]], inputs: { inputData: ["mfData", "inputData"], sortBy: ["mfSortBy", "sortBy"], sortOrder: ["mfSortOrder", "sortOrder"], rowsOnPage: ["mfRowsOnPage", "rowsOnPage"], activePage: ["mfActivePage", "activePage"] }, outputs: { sortByChange: "mfSortByChange", sortOrderChange: "mfSortOrderChange" }, exportAs: ["mfDataTable"], features: [ɵngcc0.ɵɵNgOnChangesFeature] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DataTable, [{
        type: core.Directive,
        args: [{
                selector: "table[mfData]",
                exportAs: "mfDataTable"
            }]
    }], function () { return [{ type: ɵngcc0.IterableDiffers }]; }, { inputData: [{
            type: core.Input,
            args: ["mfData"]
        }], sortBy: [{
            type: core.Input,
            args: ["mfSortBy"]
        }], sortOrder: [{
            type: core.Input,
            args: ["mfSortOrder"]
        }], sortByChange: [{
            type: core.Output,
            args: ["mfSortByChange"]
        }], sortOrderChange: [{
            type: core.Output,
            args: ["mfSortOrderChange"]
        }], rowsOnPage: [{
            type: core.Input,
            args: ["mfRowsOnPage"]
        }], activePage: [{
            type: core.Input,
            args: ["mfActivePage"]
        }] }); })();
        return DataTable;
    }());
    DataTable.ctorParameters = function () { return [
        { type: core.IterableDiffers }
    ]; };
    DataTable.propDecorators = {
        inputData: [{ type: core.Input, args: ["mfData",] }],
        sortBy: [{ type: core.Input, args: ["mfSortBy",] }],
        sortOrder: [{ type: core.Input, args: ["mfSortOrder",] }],
        sortByChange: [{ type: core.Output, args: ["mfSortByChange",] }],
        sortOrderChange: [{ type: core.Output, args: ["mfSortOrderChange",] }],
        rowsOnPage: [{ type: core.Input, args: ["mfRowsOnPage",] }],
        activePage: [{ type: core.Input, args: ["mfActivePage",] }]
    };

    var Paginator = /** @class */ (function () {
        function Paginator(injectMfTable) {
            var _this = this;
            this.injectMfTable = injectMfTable;
            this.dataLength = 0;
            this.onPageChangeSubscriber = function (event) {
                _this.activePage = event.activePage;
                _this.rowsOnPage = event.rowsOnPage;
                _this.dataLength = event.dataLength;
                _this.lastPage = Math.ceil(_this.dataLength / _this.rowsOnPage);
            };
        }
        Paginator.prototype.ngOnChanges = function (changes) {
            this.mfTable = this.inputMfTable || this.injectMfTable;
            this.onPageChangeSubscriber(this.mfTable.getPage());
            this.mfTable.onPageChange.subscribe(this.onPageChangeSubscriber);
        };
        Paginator.prototype.setPage = function (pageNumber) {
            this.mfTable.setPage(pageNumber, this.rowsOnPage);
        };
        Paginator.prototype.setRowsOnPage = function (rowsOnPage) {
            this.mfTable.setPage(this.activePage, rowsOnPage);
        };
Paginator.ɵfac = function Paginator_Factory(t) { return new (t || Paginator)(ɵngcc0.ɵɵdirectiveInject(DataTable, 8)); };
Paginator.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: Paginator, selectors: [["mfPaginator"]], inputs: { inputMfTable: ["mfTable", "inputMfTable"] }, features: [ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function Paginator_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Paginator, [{
        type: core.Component,
        args: [{
                selector: "mfPaginator",
                template: "<ng-content></ng-content>"
            }]
    }], function () { return [{ type: DataTable, decorators: [{
                type: core.Optional
            }] }]; }, { inputMfTable: [{
            type: core.Input,
            args: ["mfTable"]
        }] }); })();
        return Paginator;
    }());
    Paginator.ctorParameters = function () { return [
        { type: DataTable, decorators: [{ type: core.Optional }] }
    ]; };
    Paginator.propDecorators = {
        inputMfTable: [{ type: core.Input, args: ["mfTable",] }]
    };

    var DefaultSorter = /** @class */ (function () {
        function DefaultSorter(mfTable) {
            this.mfTable = mfTable;
            this.isSortedByMeAsc = false;
            this.isSortedByMeDesc = false;
        }
        DefaultSorter.prototype.ngOnInit = function () {
            var _this = this;
            this.mfTable.onSortChange.subscribe(function (event) {
                // tslint:disable:triple-equals
                _this.isSortedByMeAsc = (event.sortBy == _this.sortBy && event.sortOrder == "asc");
                _this.isSortedByMeDesc = (event.sortBy == _this.sortBy && event.sortOrder == "desc");
                // tslint:enable:triple-equals
            });
        };
        DefaultSorter.prototype.sort = function () {
            if (this.isSortedByMeAsc) {
                this.mfTable.setSort(this.sortBy, "desc");
            }
            else {
                this.mfTable.setSort(this.sortBy, "asc");
            }
        };
DefaultSorter.ɵfac = function DefaultSorter_Factory(t) { return new (t || DefaultSorter)(ɵngcc0.ɵɵdirectiveInject(DataTable)); };
DefaultSorter.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DefaultSorter, selectors: [["mfDefaultSorter"]], inputs: { sortBy: ["by", "sortBy"] }, ngContentSelectors: _c0, decls: 4, vars: 2, consts: [[1, "text-nowrap", 2, "cursor", "pointer", 3, "click"], ["class", "glyphicon glyphicon-triangle-top", "aria-hidden", "true", 4, "ngIf"], ["class", "glyphicon glyphicon-triangle-bottom", "aria-hidden", "true", 4, "ngIf"], ["aria-hidden", "true", 1, "glyphicon", "glyphicon-triangle-top"], ["aria-hidden", "true", 1, "glyphicon", "glyphicon-triangle-bottom"]], template: function DefaultSorter_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "a", 0);
        ɵngcc0.ɵɵlistener("click", function DefaultSorter_Template_a_click_0_listener() { return ctx.sort(); });
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵtemplate(2, DefaultSorter_span_2_Template, 1, 0, "span", 1);
        ɵngcc0.ɵɵtemplate(3, DefaultSorter_span_3_Template, 1, 0, "span", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isSortedByMeAsc);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isSortedByMeDesc);
    } }, directives: [ɵngcc1.NgIf], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DefaultSorter, [{
        type: core.Component,
        args: [{
                selector: "mfDefaultSorter",
                template: "\n        <a style=\"cursor: pointer\" (click)=\"sort()\" class=\"text-nowrap\">\n            <ng-content></ng-content>\n            <span *ngIf=\"isSortedByMeAsc\" class=\"glyphicon glyphicon-triangle-top\" aria-hidden=\"true\"></span>\n            <span *ngIf=\"isSortedByMeDesc\" class=\"glyphicon glyphicon-triangle-bottom\" aria-hidden=\"true\"></span>\n        </a>"
            }]
    }], function () { return [{ type: DataTable }]; }, { sortBy: [{
            type: core.Input,
            args: ["by"]
        }] }); })();
        return DefaultSorter;
    }());
    DefaultSorter.ctorParameters = function () { return [
        { type: DataTable }
    ]; };
    DefaultSorter.propDecorators = {
        sortBy: [{ type: core.Input, args: ["by",] }]
    };

    var DataTableModule = /** @class */ (function () {
        function DataTableModule() {
        }
DataTableModule.ɵfac = function DataTableModule_Factory(t) { return new (t || DataTableModule)(); };
DataTableModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DataTableModule });
DataTableModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DataTableModule, { declarations: [DataTable, DefaultSorter, Paginator, BootstrapPaginator], imports: [ɵngcc1.CommonModule], exports: [DataTable, DefaultSorter, Paginator, BootstrapPaginator] }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DataTableModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    DataTable,
                    DefaultSorter,
                    Paginator,
                    BootstrapPaginator
                ],
                exports: [
                    DataTable,
                    DefaultSorter,
                    Paginator,
                    BootstrapPaginator
                ]
            }]
    }], function () { return []; }, null); })();
        return DataTableModule;
    }());

    /*
     * Public API Surface of ng-datatable
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.BootstrapPaginator = BootstrapPaginator;
    exports.DataTable = DataTable;
    exports.DataTableModule = DataTableModule;
    exports.DefaultSorter = DefaultSorter;
    exports.Paginator = Paginator;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
