// @coreui/angular/esm2015/lib/widget/widget.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { WidgetSimpleComponent } from './widget-simple/widget-simple.component';
import { WidgetBrandComponent } from './widget-brand/widget-brand.component';
import { WidgetIconComponent } from './widget-icon/widget-icon.component';
import { WidgetDropdownComponent } from './widget-dropdown/widget-dropdown.component';
import { WidgetProgressComponent } from './widget-progress/widget-progress.component';
import { WidgetProgressIconComponent } from './widget-progress-icon/widget-progress-icon.component';
import { ProgressModule } from '../progress/progress.module';
import * as ɵngcc0 from '@angular/core';
export class WidgetModule {
}
WidgetModule.ɵfac = function WidgetModule_Factory(t) { return new (t || WidgetModule)(); };
WidgetModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WidgetModule });
WidgetModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule,
            ProgressModule
        ], ProgressModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WidgetModule, { declarations: function () { return [WidgetSimpleComponent,
        WidgetBrandComponent,
        WidgetIconComponent,
        WidgetDropdownComponent,
        WidgetProgressComponent,
        WidgetProgressIconComponent]; }, imports: function () { return [CommonModule,
        ProgressModule]; }, exports: function () { return [WidgetSimpleComponent,
        WidgetBrandComponent,
        WidgetIconComponent,
        WidgetDropdownComponent,
        WidgetProgressComponent,
        WidgetProgressIconComponent,
        ProgressModule]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(WidgetModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    WidgetSimpleComponent,
                    WidgetBrandComponent,
                    WidgetIconComponent,
                    WidgetDropdownComponent,
                    WidgetProgressComponent,
                    WidgetProgressIconComponent
                ],
                exports: [
                    WidgetSimpleComponent,
                    WidgetBrandComponent,
                    WidgetIconComponent,
                    WidgetDropdownComponent,
                    WidgetProgressComponent,
                    WidgetProgressIconComponent,
                    ProgressModule
                ],
                imports: [
                    CommonModule,
                    ProgressModule
                ]
            }]
    }], null, null); })();
