// @coreui/angular/esm2015/lib/layout/layout.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FooterComponent } from './footer';
import { HeaderComponent, HeaderBrandComponent } from './header';
import { SubheaderComponent } from './subheader/subheader.component';
import { WrapperComponent } from './wrapper/wrapper.component';
import { BodyComponent } from './body/body.component';
import * as ɵngcc0 from '@angular/core';
export class LayoutModule {
}
LayoutModule.ɵfac = function LayoutModule_Factory(t) { return new (t || LayoutModule)(); };
LayoutModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: LayoutModule });
LayoutModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ providers: [], imports: [[
            CommonModule,
            RouterModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(LayoutModule, { declarations: function () { return [FooterComponent,
        HeaderComponent,
        HeaderBrandComponent,
        SubheaderComponent,
        WrapperComponent,
        BodyComponent]; }, imports: function () { return [CommonModule,
        RouterModule]; }, exports: function () { return [BodyComponent,
        FooterComponent,
        HeaderComponent,
        HeaderBrandComponent,
        SubheaderComponent,
        WrapperComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(LayoutModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    RouterModule,
                ],
                exports: [
                    BodyComponent,
                    FooterComponent,
                    HeaderComponent,
                    HeaderBrandComponent,
                    SubheaderComponent,
                    WrapperComponent
                ],
                declarations: [
                    FooterComponent,
                    HeaderComponent,
                    HeaderBrandComponent,
                    SubheaderComponent,
                    WrapperComponent,
                    BodyComponent,
                ],
                providers: []
            }]
    }], null, null); })();
