// ng-chartjs/esm2015/lib/ng-chartjs.service.js
import { Injectable } from '@angular/core';
import * as Chart from 'chart.js';
import { PluginConfig } from './plugins-config';
import { StoreService } from './store.service';
import * as i0 from "@angular/core";
import * as i1 from "./store.service";
import * as i2 from "./plugins-config";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './store.service';
import * as ɵngcc2 from './plugins-config';
export class NgChartjsService {
    constructor(storeService, pluginConfig) {
        this.storeService = storeService;
        this.pluginConfig = pluginConfig;
        const plugins = pluginConfig.plugins;
        if (plugins.length !== 0 || plugins) {
            for (let i = 0; i < plugins.length; i++) {
                if (plugins[i]) {
                    Chart.plugins.register(plugins[i]);
                }
            }
        }
    }
    // get chart instance by id
    getChart(id) {
        return this.storeService.getChart(id);
    }
}
NgChartjsService.ɵfac = function NgChartjsService_Factory(t) { return new (t || NgChartjsService)(ɵngcc0.ɵɵinject(ɵngcc1.StoreService), ɵngcc0.ɵɵinject(ɵngcc2.PluginConfig)); };
NgChartjsService.ɵprov = i0.ɵɵdefineInjectable({ factory: function NgChartjsService_Factory() { return new NgChartjsService(i0.ɵɵinject(i1.StoreService), i0.ɵɵinject(i2.PluginConfig)); }, token: NgChartjsService, providedIn: "root" });
NgChartjsService.ctorParameters = () => [
    { type: StoreService },
    { type: PluginConfig }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgChartjsService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: ɵngcc1.StoreService }, { type: ɵngcc2.PluginConfig }]; }, null); })();
