// @coreui/angular/esm2015/lib/list-group/list-group.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ListGroupDirective } from './list-group.directive';
import { ListGroupItemDirective } from './list-group-item.directive';
import * as ɵngcc0 from '@angular/core';
export class ListGroupModule {
}
ListGroupModule.ɵfac = function ListGroupModule_Factory(t) { return new (t || ListGroupModule)(); };
ListGroupModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ListGroupModule });
ListGroupModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ListGroupModule, { declarations: function () { return [ListGroupDirective, ListGroupItemDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ListGroupDirective,
        ListGroupItemDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ListGroupModule, [{
        type: NgModule,
        args: [{
                declarations: [ListGroupDirective, ListGroupItemDirective],
                exports: [
                    ListGroupDirective,
                    ListGroupItemDirective
                ],
                imports: [
                    CommonModule
                ]
            }]
    }], null, null); })();
