// @coreui/angular/esm2015/lib/list-group/list-group-item.directive.js
import { Directive, ElementRef, HostBinding, Input, Renderer2 } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';
export class ListGroupItemDirective {
    constructor(renderer, hostElement) {
        this.renderer = renderer;
        this.hostElement = hostElement;
        this.host = this.hostElement.nativeElement;
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames({
            'list-group-item': true,
            'list-group-item-action': this.action,
            'list-group-item-divider': this.divider,
            active: this.active,
            disabled: this.disabled,
            [`list-group-item-${this.color}`]: !!this.color,
            [`list-group-item-accent-${this.accent}`]: !!this.accent
        });
        return classes;
    }
    ngOnInit() { }
    ngOnChanges(changes) {
        if (changes.color) {
            this.renderer.addClass(this.host, `list-group-item-${changes.color.currentValue}`);
            this.renderer.removeClass(this.host, `list-group-item-${changes.color.previousValue}`);
        }
        if (changes.accent) {
            this.renderer.addClass(this.host, `list-group-item-accent-${changes.accent.currentValue}`);
            this.renderer.removeClass(this.host, `list-group-item-accent-${changes.accent.previousValue}`);
        }
    }
}
ListGroupItemDirective.ɵfac = function ListGroupItemDirective_Factory(t) { return new (t || ListGroupItemDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ListGroupItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ListGroupItemDirective, selectors: [["", "cListGroupItem", ""], ["c-list-group-item"]], hostVars: 3, hostBindings: function ListGroupItemDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("disabled", ctx.disabled);
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { accent: "accent", action: "action", active: "active", color: "color", divider: "divider", disabled: "disabled" }, exportAs: ["cListGroupItem"], features: [ɵngcc0.ɵɵNgOnChangesFeature] });
ListGroupItemDirective.ctorParameters = () => [
    { type: Renderer2 },
    { type: ElementRef }
];
ListGroupItemDirective.propDecorators = {
    accent: [{ type: Input }],
    action: [{ type: Input }],
    active: [{ type: Input }],
    color: [{ type: Input }],
    divider: [{ type: Input }],
    disabled: [{ type: HostBinding, args: ['attr.disabled',] }, { type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ListGroupItemDirective, [{
        type: Directive,
        args: [{
                selector: '[cListGroupItem], c-list-group-item',
                exportAs: 'cListGroupItem'
            }]
    }], function () { return [{ type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ElementRef }]; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], accent: [{
            type: Input
        }], action: [{
            type: Input
        }], active: [{
            type: Input
        }], color: [{
            type: Input
        }], divider: [{
            type: Input
        }], disabled: [{
            type: HostBinding,
            args: ['attr.disabled']
        }, {
            type: Input
        }] }); })();
