// @coreui/angular/esm2015/lib/accordion/accordion.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AccordionComponent } from './accordion.component';
import { AccordionGroupComponent } from './accordion-group.component';
import { CardModule } from '../card';
import { CollapseModule } from '../collapse';
import * as ɵngcc0 from '@angular/core';
export class AccordionModule {
}
AccordionModule.ɵfac = function AccordionModule_Factory(t) { return new (t || AccordionModule)(); };
AccordionModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AccordionModule });
AccordionModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule,
            CardModule,
            CollapseModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AccordionModule, { declarations: function () { return [AccordionComponent, AccordionGroupComponent]; }, imports: function () { return [CommonModule,
        CardModule,
        CollapseModule]; }, exports: function () { return [AccordionComponent,
        AccordionGroupComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AccordionModule, [{
        type: NgModule,
        args: [{
                declarations: [AccordionComponent, AccordionGroupComponent],
                exports: [
                    AccordionComponent,
                    AccordionGroupComponent
                ],
                imports: [
                    CommonModule,
                    CardModule,
                    CollapseModule
                ]
            }]
    }], null, null); })();
