// @coreui/angular/esm2015/lib/accordion/accordion-group.component.js
import { Component, HostBinding, Inject, Input } from '@angular/core';
import { AccordionComponent } from './accordion.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../card/card-header.component';
import * as ɵngcc2 from '../card/card-body.component';
import * as ɵngcc3 from '../collapse/collapse.directive';
import * as ɵngcc4 from './accordion.component';

const _c0 = [[["", "c-accordion-header", ""]], "*"];
const _c1 = ["[c-accordion-header]", "*"];
export class AccordionGroupComponent {
    constructor(accordion) {
        this.isOpen = false;
        this.groupClass = '';
        this.cardClass = true;
        this.accordion = accordion;
    }
    ngOnInit() {
        this.accordion.addGroup(this);
    }
    ngOnDestroy() {
        this.accordion.removeGroup(this);
    }
    toggleGroup() {
        this.isOpen = !this.isOpen;
        this.accordion.closeOtherGroups(this);
    }
}
AccordionGroupComponent.ɵfac = function AccordionGroupComponent_Factory(t) { return new (t || AccordionGroupComponent)(ɵngcc0.ɵɵdirectiveInject(AccordionComponent)); };
AccordionGroupComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AccordionGroupComponent, selectors: [["c-accordion-group"]], hostVars: 2, hostBindings: function AccordionGroupComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("card", ctx.cardClass);
    } }, inputs: { isOpen: "isOpen", groupClass: "groupClass" }, ngContentSelectors: _c1, decls: 4, vars: 1, consts: [[3, "click"], ["cCollapse", "", 3, "show"]], template: function AccordionGroupComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c0);
        ɵngcc0.ɵɵelementStart(0, "c-card-header", 0);
        ɵngcc0.ɵɵlistener("click", function AccordionGroupComponent_Template_c_card_header_click_0_listener() { return ctx.toggleGroup(); });
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(2, "c-card-body", 1);
        ɵngcc0.ɵɵprojection(3, 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("show", !ctx.isOpen);
    } }, directives: [ɵngcc1.CardHeaderComponent, ɵngcc2.CardBodyComponent, ɵngcc3.CollapseDirective], styles: ["[_nghost-%COMP%]{margin-bottom:0}"] });
AccordionGroupComponent.ctorParameters = () => [
    { type: AccordionComponent, decorators: [{ type: Inject, args: [AccordionComponent,] }] }
];
AccordionGroupComponent.propDecorators = {
    isOpen: [{ type: Input }],
    groupClass: [{ type: Input }],
    cardClass: [{ type: HostBinding, args: ['class.card',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AccordionGroupComponent, [{
        type: Component,
        args: [{
                selector: 'c-accordion-group',
                template: "<c-card-header (click)=\"toggleGroup()\">\n  <ng-content select=\"[c-accordion-header]\"></ng-content>\n</c-card-header>\n<c-card-body cCollapse [show]=\"!isOpen\">\n  <ng-content></ng-content>\n</c-card-body>\n",
                styles: [":host{margin-bottom:0}"]
            }]
    }], function () { return [{ type: ɵngcc4.AccordionComponent, decorators: [{
                type: Inject,
                args: [AccordionComponent]
            }] }]; }, { isOpen: [{
            type: Input
        }], groupClass: [{
            type: Input
        }], cardClass: [{
            type: HostBinding,
            args: ['class.card']
        }] }); })();
