// @coreui/angular/esm2015/lib/toast/toast-body/toast-body.component.js
import { Component, HostBinding, Optional } from '@angular/core';
import { ToastComponent } from '../toast/toast.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../toast/toast.component';

const _c0 = ["*"];
export class ToastBodyComponent {
    constructor(toast) {
        this.toast = toast;
        this.toastBodyClass = true;
    }
    ngOnInit() {
    }
}
ToastBodyComponent.ɵfac = function ToastBodyComponent_Factory(t) { return new (t || ToastBodyComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ToastComponent, 8)); };
ToastBodyComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ToastBodyComponent, selectors: [["c-toast-body"]], hostVars: 2, hostBindings: function ToastBodyComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("toast-body", ctx.toastBodyClass);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function ToastBodyComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]{display:block}"] });
ToastBodyComponent.ctorParameters = () => [
    { type: ToastComponent, decorators: [{ type: Optional }] }
];
ToastBodyComponent.propDecorators = {
    toastBodyClass: [{ type: HostBinding, args: ['class.toast-body',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ToastBodyComponent, [{
        type: Component,
        args: [{
                selector: 'c-toast-body',
                template: "<ng-content></ng-content>\n",
                styles: [":host{display:block}"]
            }]
    }], function () { return [{ type: ɵngcc1.ToastComponent, decorators: [{
                type: Optional
            }] }]; }, { toastBodyClass: [{
            type: HostBinding,
            args: ['class.toast-body']
        }] }); })();
