// @coreui/angular/esm2015/lib/grid/col.component.js
import { Component } from '@angular/core';
import { ColDirective } from './col.directive';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class ColComponent extends ColDirective {
}
ColComponent.ɵfac = function ColComponent_Factory(t) { return ɵColComponent_BaseFactory(t || ColComponent); };
ColComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ColComponent, selectors: [["c-col"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function ColComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]{display:block}"] });
const ɵColComponent_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(ColComponent);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ColComponent, [{
        type: Component,
        args: [{
                selector: 'c-col',
                template: '<ng-content></ng-content>',
                styles: [":host{display:block}"]
            }]
    }], null, null); })();
