// @coreui/angular/esm2015/lib/grid/col.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';
export class ColDirective {
    constructor() {
        this.suffix = {
            xs: '-xs',
            sm: '-sm',
            md: '-md',
            lg: '-lg',
            xl: '-xl',
            xxl: '-xxl',
            col: ''
        };
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        let classes = Object.create(null);
        const suffix = this.suffix;
        let addColClass = false;
        Object.keys(suffix).forEach((key) => {
            const prop = this[key];
            const breakpoint = suffix[key];
            classes[`col${breakpoint}`] = prop === true;
            classes[`col${breakpoint}-${prop}`] = (typeof prop === 'number') || (typeof prop === 'string');
            if (typeof prop === 'object') {
                classes[`col${breakpoint}-${prop.size}`] = !!prop.size;
                classes[`offset${breakpoint}-${prop.offset}`] = !!prop.offset;
                classes[`order${breakpoint}-${prop.order}`] = !!prop.order;
                addColClass = addColClass || !!prop.offset || !!prop.order;
            }
        });
        classes = classNames(addColClass ? 'col' : '', classes);
        return classes || 'col';
    }
}
ColDirective.ɵfac = function ColDirective_Factory(t) { return new (t || ColDirective)(); };
ColDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ColDirective, selectors: [["", "cCol", ""]], hostVars: 2, hostBindings: function ColDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { xs: "xs", sm: "sm", md: "md", lg: "lg", xl: "xl", xxl: "xxl", col: "col" } });
ColDirective.propDecorators = {
    xs: [{ type: Input }],
    sm: [{ type: Input }],
    md: [{ type: Input }],
    lg: [{ type: Input }],
    xl: [{ type: Input }],
    xxl: [{ type: Input }],
    col: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ColDirective, [{
        type: Directive,
        args: [{
                selector: '[cCol]'
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], xs: [{
            type: Input
        }], sm: [{
            type: Input
        }], md: [{
            type: Input
        }], lg: [{
            type: Input
        }], xl: [{
            type: Input
        }], xxl: [{
            type: Input
        }], col: [{
            type: Input
        }] }); })();
