// @coreui/angular/esm2015/lib/grid/grid.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ContainerComponent } from './container.component';
import { RowComponent } from './row.component';
import { ColComponent } from './col.component';
import { RowDirective } from './row.directive';
import { ColDirective } from './col.directive';
import * as ɵngcc0 from '@angular/core';
export class GridModule {
}
GridModule.ɵfac = function GridModule_Factory(t) { return new (t || GridModule)(); };
GridModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: GridModule });
GridModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ providers: [], imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(GridModule, { declarations: function () { return [ContainerComponent,
        RowComponent,
        ColComponent,
        RowDirective,
        ColDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ContainerComponent,
        RowComponent,
        ColComponent,
        RowDirective,
        ColDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(GridModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [
                    ContainerComponent,
                    RowComponent,
                    ColComponent,
                    RowDirective,
                    ColDirective,
                ],
                declarations: [
                    ContainerComponent,
                    RowComponent,
                    ColComponent,
                    RowDirective,
                    ColDirective,
                ],
                providers: []
            }]
    }], null, null); })();
