// @coreui/angular/esm2015/lib/grid/row.component.js
import { Component } from '@angular/core';
import { RowDirective } from './row.directive';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class RowComponent extends RowDirective {
}
RowComponent.ɵfac = function RowComponent_Factory(t) { return ɵRowComponent_BaseFactory(t || RowComponent); };
RowComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: RowComponent, selectors: [["c-row"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function RowComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
const ɵRowComponent_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(RowComponent);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(RowComponent, [{
        type: Component,
        args: [{
                selector: 'c-row',
                template: '<ng-content></ng-content>',
                styles: [""]
            }]
    }], null, null); })();
