// @coreui/angular/esm2015/lib/grid/row.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';
export class RowDirective {
    constructor() {
        this.noGutters = false;
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const alignV = `align-items-${this.alignV}`;
        const alignH = `justify-content-${this.alignH}`;
        const classes = { row: true };
        classes['no-gutters'] = !!this.noGutters;
        classes[alignV] = !!this.alignV;
        classes[alignH] = !!this.alignH;
        return classNames(classes);
    }
}
RowDirective.ɵfac = function RowDirective_Factory(t) { return new (t || RowDirective)(); };
RowDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: RowDirective, selectors: [["", "cRow", ""]], hostVars: 2, hostBindings: function RowDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { noGutters: "noGutters", alignV: "alignV", alignH: "alignH" } });
RowDirective.propDecorators = {
    noGutters: [{ type: Input }],
    alignV: [{ type: Input }],
    alignH: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(RowDirective, [{
        type: Directive,
        args: [{
                selector: '[cRow]'
            }]
    }], function () { return []; }, { noGutters: [{
            type: Input
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], alignV: [{
            type: Input
        }], alignH: [{
            type: Input
        }] }); })();
