// @coreui/angular/esm2015/lib/sidebar/sidebar-nav/sidebar-nav-dropdown.directive.js
import { Directive, HostListener } from '@angular/core';
import { SidebarNavDropdownComponent } from './sidebar-nav-dropdown.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './sidebar-nav-dropdown.component';
export class SidebarNavDropdownToggleDirective {
    constructor(sidebarNavDropdown) {
        this.sidebarNavDropdown = sidebarNavDropdown;
    }
    toggleOpen($event) {
        $event.preventDefault();
        this.sidebarNavDropdown.toggleDropdown();
    }
}
SidebarNavDropdownToggleDirective.ɵfac = function SidebarNavDropdownToggleDirective_Factory(t) { return new (t || SidebarNavDropdownToggleDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.SidebarNavDropdownComponent)); };
SidebarNavDropdownToggleDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SidebarNavDropdownToggleDirective, selectors: [["", "cSidebarNavDropdownToggle", ""]], hostBindings: function SidebarNavDropdownToggleDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function SidebarNavDropdownToggleDirective_click_HostBindingHandler($event) { return ctx.toggleOpen($event); });
    } }, exportAs: ["cSidebarNavDropdownToggle"] });
SidebarNavDropdownToggleDirective.ctorParameters = () => [
    { type: SidebarNavDropdownComponent }
];
SidebarNavDropdownToggleDirective.propDecorators = {
    toggleOpen: [{ type: HostListener, args: ['click', ['$event'],] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarNavDropdownToggleDirective, [{
        type: Directive,
        args: [{
                selector: '[cSidebarNavDropdownToggle]',
                exportAs: 'cSidebarNavDropdownToggle'
            }]
    }], function () { return [{ type: ɵngcc1.SidebarNavDropdownComponent }]; }, { toggleOpen: [{
            type: HostListener,
            args: ['click', ['$event']]
        }] }); })();
