// @coreui/angular/esm2015/lib/sidebar/sidebar-nav/sidebar-nav-item-class.pipe.js
import { Pipe } from '@angular/core';
import { SidebarNavHelper } from './sidebar-nav.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './sidebar-nav.service';
export class SidebarNavItemClassPipe {
    constructor(helper) {
        this.helper = helper;
    }
    transform(item, args) {
        const itemType = this.helper.itemType(item);
        let itemClass;
        if (['divider', 'title'].includes(itemType)) {
            itemClass = `c-sidebar-nav-${itemType}`;
        }
        else if (itemType === 'dropdown') {
            itemClass = '';
        }
        else {
            itemClass = 'c-sidebar-nav-item';
        }
        return item.class ? `${itemClass} ${item.class}` : itemClass;
    }
}
SidebarNavItemClassPipe.ɵfac = function SidebarNavItemClassPipe_Factory(t) { return new (t || SidebarNavItemClassPipe)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.SidebarNavHelper)); };
SidebarNavItemClassPipe.ɵpipe = ɵngcc0.ɵɵdefinePipe({ name: "cSidebarNavItemClass", type: SidebarNavItemClassPipe, pure: true });
SidebarNavItemClassPipe.ctorParameters = () => [
    { type: SidebarNavHelper }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarNavItemClassPipe, [{
        type: Pipe,
        args: [{
                name: 'cSidebarNavItemClass'
            }]
    }], function () { return [{ type: ɵngcc1.SidebarNavHelper }]; }, null); })();
