// @coreui/angular/esm2015/lib/sidebar/sidebar-nav/sidebar-nav-items.component.js
import { Component, Input } from '@angular/core';
import { Router } from '@angular/router';
import { SidebarNavHelper } from './sidebar-nav.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/router';
import * as ɵngcc2 from './sidebar-nav.service';
import * as ɵngcc3 from '@angular/common';
import * as ɵngcc4 from './sidebar-nav-dropdown.component';
import * as ɵngcc5 from './sidebar-nav-divider.component';
import * as ɵngcc6 from '../../shared/html-attr.directive';
import * as ɵngcc7 from './sidebar-nav-title.component';
import * as ɵngcc8 from './sidebar-nav-label.component';
import * as ɵngcc9 from './sidebar-nav-link.component';
import * as ɵngcc10 from './sidebar-nav-item-class.pipe';

function SidebarNavItemsComponent_ng_container_0_c_sidebar_nav_dropdown_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "c-sidebar-nav-dropdown", 7);
    ɵngcc0.ɵɵpipe(1, "cSidebarNavItemClass");
} if (rf & 2) {
    const item_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassProp("c-show", ctx_r2.helper.isActive(ctx_r2.router, item_r1));
    ɵngcc0.ɵɵproperty("item", item_r1)("ngClass", ɵngcc0.ɵɵpipeBind1(1, 4, item_r1));
} }
function SidebarNavItemsComponent_ng_container_0_c_sidebar_nav_divider_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "c-sidebar-nav-divider", 8);
    ɵngcc0.ɵɵpipe(1, "cSidebarNavItemClass");
} if (rf & 2) {
    const item_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("item", item_r1)("ngClass", ɵngcc0.ɵɵpipeBind1(1, 3, item_r1))("cHtmlAttr", item_r1.attributes);
} }
function SidebarNavItemsComponent_ng_container_0_c_sidebar_nav_title_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "c-sidebar-nav-title", 8);
    ɵngcc0.ɵɵpipe(1, "cSidebarNavItemClass");
} if (rf & 2) {
    const item_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("item", item_r1)("ngClass", ɵngcc0.ɵɵpipeBind1(1, 3, item_r1))("cHtmlAttr", item_r1.attributes);
} }
function SidebarNavItemsComponent_ng_container_0_c_sidebar_nav_label_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "c-sidebar-nav-label", 9);
    ɵngcc0.ɵɵpipe(1, "cSidebarNavItemClass");
} if (rf & 2) {
    const item_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("item", item_r1)("ngClass", ɵngcc0.ɵɵpipeBind1(1, 2, item_r1));
} }
function SidebarNavItemsComponent_ng_container_0_ng_container_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0);
} }
function SidebarNavItemsComponent_ng_container_0_c_sidebar_nav_link_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "c-sidebar-nav-link", 9);
    ɵngcc0.ɵɵpipe(1, "cSidebarNavItemClass");
} if (rf & 2) {
    const item_r1 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("item", item_r1)("ngClass", ɵngcc0.ɵɵpipeBind1(1, 2, item_r1));
} }
function SidebarNavItemsComponent_ng_container_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementContainerStart(1, 1);
    ɵngcc0.ɵɵtemplate(2, SidebarNavItemsComponent_ng_container_0_c_sidebar_nav_dropdown_2_Template, 2, 6, "c-sidebar-nav-dropdown", 2);
    ɵngcc0.ɵɵtemplate(3, SidebarNavItemsComponent_ng_container_0_c_sidebar_nav_divider_3_Template, 2, 5, "c-sidebar-nav-divider", 3);
    ɵngcc0.ɵɵtemplate(4, SidebarNavItemsComponent_ng_container_0_c_sidebar_nav_title_4_Template, 2, 5, "c-sidebar-nav-title", 3);
    ɵngcc0.ɵɵtemplate(5, SidebarNavItemsComponent_ng_container_0_c_sidebar_nav_label_5_Template, 2, 4, "c-sidebar-nav-label", 4);
    ɵngcc0.ɵɵtemplate(6, SidebarNavItemsComponent_ng_container_0_ng_container_6_Template, 1, 0, "ng-container", 5);
    ɵngcc0.ɵɵtemplate(7, SidebarNavItemsComponent_ng_container_0_c_sidebar_nav_link_7_Template, 2, 4, "c-sidebar-nav-link", 6);
    ɵngcc0.ɵɵelementContainerEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const item_r1 = ctx.$implicit;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitch", ctx_r0.helper.itemType(item_r1));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitchCase", "dropdown");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitchCase", "divider");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitchCase", "title");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitchCase", "label");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngSwitchCase", "empty");
} }
export class SidebarNavItemsComponent {
    constructor(router, helper) {
        this.router = router;
        this.helper = helper;
    }
    set items(items) {
        this._items = [...items];
    }
    get items() {
        return this._items;
    }
}
SidebarNavItemsComponent.ɵfac = function SidebarNavItemsComponent_Factory(t) { return new (t || SidebarNavItemsComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Router), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.SidebarNavHelper)); };
SidebarNavItemsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SidebarNavItemsComponent, selectors: [["c-sidebar-nav-items"]], inputs: { items: "items" }, decls: 1, vars: 1, consts: [[4, "ngFor", "ngForOf"], [3, "ngSwitch"], ["cSidebarNavDropdown", "", "routerLinkActive", "c-show", 3, "item", "c-show", "ngClass", 4, "ngSwitchCase"], [3, "item", "ngClass", "cHtmlAttr", 4, "ngSwitchCase"], [3, "item", "ngClass", 4, "ngSwitchCase"], [4, "ngSwitchCase"], [3, "item", "ngClass", 4, "ngSwitchDefault"], ["cSidebarNavDropdown", "", "routerLinkActive", "c-show", 3, "item", "ngClass"], [3, "item", "ngClass", "cHtmlAttr"], [3, "item", "ngClass"]], template: function SidebarNavItemsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SidebarNavItemsComponent_ng_container_0_Template, 8, 6, "ng-container", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngForOf", ctx.items);
    } }, directives: [ɵngcc3.NgForOf, ɵngcc3.NgSwitch, ɵngcc3.NgSwitchCase, ɵngcc3.NgSwitchDefault, ɵngcc4.SidebarNavDropdownComponent, ɵngcc1.RouterLinkActive, ɵngcc3.NgClass, ɵngcc5.SidebarNavDividerComponent, ɵngcc6.HtmlAttributesDirective, ɵngcc7.SidebarNavTitleComponent, ɵngcc8.SidebarNavLabelComponent, ɵngcc9.SidebarNavLinkComponent], pipes: [ɵngcc10.SidebarNavItemClassPipe], encapsulation: 2 });
SidebarNavItemsComponent.ctorParameters = () => [
    { type: Router },
    { type: SidebarNavHelper }
];
SidebarNavItemsComponent.propDecorators = {
    items: [{ type: Input }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarNavItemsComponent, [{
        type: Component,
        args: [{
                selector: 'c-sidebar-nav-items',
                template: `
    <ng-container *ngFor="let item of items">
      <ng-container [ngSwitch]="helper.itemType(item)">
        <c-sidebar-nav-dropdown
          *ngSwitchCase="'dropdown'"
          [item]="item"
          [class.c-show]="helper.isActive(router, item)"
          [ngClass]="item | cSidebarNavItemClass"
          cSidebarNavDropdown
          routerLinkActive="c-show">
        </c-sidebar-nav-dropdown>
        <c-sidebar-nav-divider
          *ngSwitchCase="'divider'"
          [item]="item"
          [ngClass]="item | cSidebarNavItemClass"
          [cHtmlAttr]="item.attributes">
        </c-sidebar-nav-divider>
        <c-sidebar-nav-title
          *ngSwitchCase="'title'"
          [item]="item"
          [ngClass]="item | cSidebarNavItemClass"
          [cHtmlAttr]="item.attributes">
        </c-sidebar-nav-title>
        <c-sidebar-nav-label
          *ngSwitchCase="'label'"
          [item]="item"
          [ngClass]="item | cSidebarNavItemClass">
        </c-sidebar-nav-label>
        <ng-container
          *ngSwitchCase="'empty'">
        </ng-container>
        <c-sidebar-nav-link
          *ngSwitchDefault
          [item]="item"
          [ngClass]="item | cSidebarNavItemClass"
        >
        </c-sidebar-nav-link>
      </ng-container>
    </ng-container>
  `
            }]
    }], function () { return [{ type: ɵngcc1.Router }, { type: ɵngcc2.SidebarNavHelper }]; }, { items: [{
            type: Input
        }] }); })();
