// @coreui/angular/esm2015/lib/sidebar/sidebar-nav/sidebar-nav-divider.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class SidebarNavDividerComponent {
    constructor() { }
    ngOnInit() { }
}
SidebarNavDividerComponent.ɵfac = function SidebarNavDividerComponent_Factory(t) { return new (t || SidebarNavDividerComponent)(); };
SidebarNavDividerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SidebarNavDividerComponent, selectors: [["c-sidebar-nav-divider"]], inputs: { item: "item" }, decls: 0, vars: 0, template: function SidebarNavDividerComponent_Template(rf, ctx) { }, encapsulation: 2 });
SidebarNavDividerComponent.ctorParameters = () => [];
SidebarNavDividerComponent.propDecorators = {
    item: [{ type: Input }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SidebarNavDividerComponent, [{
        type: Component,
        args: [{
                selector: 'c-sidebar-nav-divider',
                template: ``
            }]
    }], function () { return []; }, { item: [{
            type: Input
        }] }); })();
