// @coreui/angular/esm2015/lib/widget/widget-brand/widget-brand.component.js
import { Component, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function WidgetBrandComponent_div_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 5);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r0.rightHeader);
} }
function WidgetBrandComponent_div_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r1.rightFooter);
} }
function WidgetBrandComponent_div_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 5);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r2.leftHeader);
} }
function WidgetBrandComponent_div_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r3.leftFooter);
} }
const _c0 = ["*"];
export class WidgetBrandComponent {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames({
            card: true,
        });
        return classes;
    }
    ngOnInit() {
    }
    get headerClasses() {
        const classes = classNames({
            'card-header': true,
            'content-center': true,
            'text-white': true,
            'p-0': true,
            [`bg-${this.color}`]: !!this.color,
        }, this.addHeaderClasses);
        return classes;
    }
}
WidgetBrandComponent.ɵfac = function WidgetBrandComponent_Factory(t) { return new (t || WidgetBrandComponent)(); };
WidgetBrandComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: WidgetBrandComponent, selectors: [["c-widget-brand"]], hostVars: 2, hostBindings: function WidgetBrandComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { color: "color", rightHeader: "rightHeader", leftHeader: "leftHeader", rightFooter: "rightFooter", leftFooter: "leftFooter", addHeaderClasses: "addHeaderClasses" }, ngContentSelectors: _c0, decls: 11, vars: 6, consts: [[1, "card-body", "row", "text-center"], [1, "col"], ["class", "text-value-lg", 4, "ngIf"], ["class", "text-uppercase text-muted small", 4, "ngIf"], [1, "c-vr"], [1, "text-value-lg"], [1, "text-uppercase", "text-muted", "small"]], template: function WidgetBrandComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementContainerStart(0);
        ɵngcc0.ɵɵelementStart(1, "div");
        ɵngcc0.ɵɵprojection(2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 0);
        ɵngcc0.ɵɵelementStart(4, "div", 1);
        ɵngcc0.ɵɵtemplate(5, WidgetBrandComponent_div_5_Template, 2, 1, "div", 2);
        ɵngcc0.ɵɵtemplate(6, WidgetBrandComponent_div_6_Template, 2, 1, "div", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(7, "div", 4);
        ɵngcc0.ɵɵelementStart(8, "div", 1);
        ɵngcc0.ɵɵtemplate(9, WidgetBrandComponent_div_9_Template, 2, 1, "div", 2);
        ɵngcc0.ɵɵtemplate(10, WidgetBrandComponent_div_10_Template, 2, 1, "div", 3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementContainerEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵclassMap(ctx.headerClasses);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("ngIf", !!ctx.rightHeader);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !!ctx.rightFooter);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngIf", !!ctx.leftHeader);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !!ctx.leftFooter);
    } }, directives: [ɵngcc1.NgIf], styles: [""] });
WidgetBrandComponent.ctorParameters = () => [];
WidgetBrandComponent.propDecorators = {
    color: [{ type: Input }],
    rightHeader: [{ type: Input }],
    leftHeader: [{ type: Input }],
    rightFooter: [{ type: Input }],
    leftFooter: [{ type: Input }],
    addHeaderClasses: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(WidgetBrandComponent, [{
        type: Component,
        args: [{
                selector: 'c-widget-brand',
                template: "<ng-container>\n  <div [class]=\"headerClasses\">\n    <ng-content></ng-content>\n  </div>\n  <!--todo: <ng-content select=\"[slot=body]\"></ng-content>-->\n  <div class=\"card-body row text-center\">\n    <div class=\"col\">\n      <div *ngIf=\"!!rightHeader\" class=\"text-value-lg\">{{rightHeader}}</div>\n      <div *ngIf=\"!!rightFooter\" class=\"text-uppercase text-muted small\">{{rightFooter}}</div>\n    </div>\n    <div class=\"c-vr\"></div>\n    <div class=\"col\">\n      <div *ngIf=\"!!leftHeader\" class=\"text-value-lg\">{{leftHeader}}</div>\n      <div *ngIf=\"!!leftFooter\" class=\"text-uppercase text-muted small\">{{leftFooter}}</div>\n    </div>\n  </div>\n</ng-container>\n",
                styles: [""]
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], color: [{
            type: Input
        }], rightHeader: [{
            type: Input
        }], leftHeader: [{
            type: Input
        }], rightFooter: [{
            type: Input
        }], leftFooter: [{
            type: Input
        }], addHeaderClasses: [{
            type: Input
        }] }); })();
