// @coreui/angular/esm2015/lib/navbar/navbar.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { NavbarComponent } from './navbar.component';
import { NavbarBrandComponent } from './navbar-brand/navbar-brand.component';
import { NavbarNavComponent } from './navbar-nav/navbar-nav.component';
import { NavbarTextComponent } from './navbar-text/navbar-text.component';
import * as ɵngcc0 from '@angular/core';
export class NavbarModule {
}
NavbarModule.ɵfac = function NavbarModule_Factory(t) { return new (t || NavbarModule)(); };
NavbarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NavbarModule });
NavbarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule,
            RouterModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NavbarModule, { declarations: function () { return [NavbarComponent,
        NavbarBrandComponent,
        NavbarNavComponent,
        NavbarTextComponent]; }, imports: function () { return [CommonModule,
        RouterModule]; }, exports: function () { return [NavbarComponent,
        NavbarBrandComponent,
        NavbarNavComponent,
        NavbarTextComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NavbarModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    NavbarComponent,
                    NavbarBrandComponent,
                    NavbarNavComponent,
                    NavbarTextComponent
                ],
                exports: [
                    NavbarComponent,
                    NavbarBrandComponent,
                    NavbarNavComponent,
                    NavbarTextComponent
                ],
                imports: [
                    CommonModule,
                    RouterModule
                ]
            }]
    }], null, null); })();
