// @coreui/angular/esm2015/lib/button/button.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonComponent } from './button.component';
import { ButtonDirective } from './button.directive';
import { ButtonGroupComponent } from './button-group.component';
import { ButtonToolbarComponent } from './button-toolbar.component';
import { ButtonRadioDirective } from './button-radio.directive';
import { ButtonRadioGroupDirective } from './button-radio-group.directive';
import * as ɵngcc0 from '@angular/core';
export class ButtonModule {
}
ButtonModule.ɵfac = function ButtonModule_Factory(t) { return new (t || ButtonModule)(); };
ButtonModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ButtonModule });
ButtonModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule,
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ButtonModule, { declarations: function () { return [ButtonComponent,
        ButtonDirective,
        ButtonGroupComponent,
        ButtonToolbarComponent,
        ButtonRadioDirective,
        ButtonRadioGroupDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ButtonComponent,
        ButtonDirective,
        ButtonGroupComponent,
        ButtonToolbarComponent,
        ButtonRadioDirective,
        ButtonRadioGroupDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ButtonModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    ButtonComponent,
                    ButtonDirective,
                    ButtonGroupComponent,
                    ButtonToolbarComponent,
                    ButtonRadioDirective,
                    ButtonRadioGroupDirective
                ],
                imports: [
                    CommonModule,
                ],
                exports: [
                    ButtonComponent,
                    ButtonDirective,
                    ButtonGroupComponent,
                    ButtonToolbarComponent,
                    ButtonRadioDirective,
                    ButtonRadioGroupDirective
                ]
            }]
    }], null, null); })();
