// @coreui/angular/esm2015/lib/button/button-toolbar.component.js
import { Component, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class ButtonToolbarComponent {
    constructor() {
        this.justify = '';
        this.class = '';
        this.classes = Object.create(null);
        this.aria = 'Button toolbar';
    }
    get hostRole() {
        return 'toolbar';
    }
    get hostClasses() {
        return this.getClasses();
    }
    ngOnInit() {
    }
    getClasses() {
        this.classes['btn-toolbar'] = true;
        this.classes[`justify-content-start`] = this.justify === 'start';
        this.classes[`justify-content-end`] = this.justify === 'end';
        this.classes[`justify-content-center}`] = this.justify === 'center';
        this.classes[`justify-content-between`] = this.justify === 'between';
        return classNames(this.classes, this.class, this.ngClass);
    }
}
ButtonToolbarComponent.ɵfac = function ButtonToolbarComponent_Factory(t) { return new (t || ButtonToolbarComponent)(); };
ButtonToolbarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ButtonToolbarComponent, selectors: [["c-button-toolbar"]], hostVars: 4, hostBindings: function ButtonToolbarComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("aria-label", ctx.aria)("role", ctx.hostRole);
        ɵngcc0.ɵɵclassMap(ctx.hostClasses);
    } }, inputs: { justify: "justify", class: "class", ngClass: "ngClass" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function ButtonToolbarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]{display:flex}"] });
ButtonToolbarComponent.ctorParameters = () => [];
ButtonToolbarComponent.propDecorators = {
    justify: [{ type: Input }],
    class: [{ type: Input }],
    ngClass: [{ type: Input }],
    aria: [{ type: HostBinding, args: ['attr.aria-label',] }],
    hostRole: [{ type: HostBinding, args: ['attr.role',] }],
    hostClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ButtonToolbarComponent, [{
        type: Component,
        args: [{
                selector: 'c-button-toolbar',
                template: '<ng-content></ng-content>',
                styles: [":host{display:flex}"]
            }]
    }], function () { return []; }, { justify: [{
            type: Input
        }], class: [{
            type: Input
        }], aria: [{
            type: HostBinding,
            args: ['attr.aria-label']
        }], hostRole: [{
            type: HostBinding,
            args: ['attr.role']
        }], hostClasses: [{
            type: HostBinding,
            args: ['class']
        }], ngClass: [{
            type: Input
        }] }); })();
