// @coreui/angular/esm2015/lib/button/button-group.component.js
import { Component, ElementRef, HostBinding, Input, Renderer2 } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class ButtonGroupComponent {
    constructor(renderer, hostElement) {
        this.renderer = renderer;
        this.hostElement = hostElement;
        this.size = '';
        this.vertical = null;
        this.classes = Object.create(null);
        this.aria = 'Button group';
        this.host = this.hostElement.nativeElement;
    }
    get hostRole() {
        return 'group';
    }
    ngOnInit() {
        this.addClasses();
    }
    addClasses() {
        this.setClasses();
        for (const [key, value] of Object.entries(this.classes)) {
            if (value) {
                this.renderer.addClass(this.host, key);
            }
        }
    }
    setClasses() {
        this.classes['btn-group-vertical'] = Boolean(this.vertical) || this.vertical != null;
        this.classes['btn-group'] = !this.classes['btn-group-vertical'];
        this.classes[`btn-group-${this.size}`] = Boolean(this.size);
    }
}
ButtonGroupComponent.ɵfac = function ButtonGroupComponent_Factory(t) { return new (t || ButtonGroupComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ButtonGroupComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ButtonGroupComponent, selectors: [["c-button-group"]], hostVars: 2, hostBindings: function ButtonGroupComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("aria-label", ctx.aria)("role", ctx.hostRole);
    } }, inputs: { size: "size", vertical: "vertical" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function ButtonGroupComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
ButtonGroupComponent.ctorParameters = () => [
    { type: Renderer2 },
    { type: ElementRef }
];
ButtonGroupComponent.propDecorators = {
    size: [{ type: Input }],
    vertical: [{ type: Input }],
    aria: [{ type: HostBinding, args: ['attr.aria-label',] }],
    hostRole: [{ type: HostBinding, args: ['attr.role',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ButtonGroupComponent, [{
        type: Component,
        args: [{
                selector: 'c-button-group',
                template: '<ng-content></ng-content>',
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ElementRef }]; }, { size: [{
            type: Input
        }], vertical: [{
            type: Input
        }], aria: [{
            type: HostBinding,
            args: ['attr.aria-label']
        }], hostRole: [{
            type: HostBinding,
            args: ['attr.role']
        }] }); })();
