// @coreui/angular/esm2015/lib/toast/toast-header/toast-header.component.js
import { Component, HostBinding, Input, Optional } from '@angular/core';
import { ToastComponent } from '../toast/toast.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../toast/toast.component';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../toast-close.directive';

const _c0 = function () { return { outline: 0 }; };
function ToastHeaderComponent_button_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "button", 1);
    ɵngcc0.ɵɵtext(1, "\u00D7");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleMap(ɵngcc0.ɵɵpureFunction0(3, _c0));
    ɵngcc0.ɵɵproperty("cToastClose", ctx_r0.toast);
} }
const _c1 = ["*"];
export class ToastHeaderComponent {
    constructor(toast) {
        this.toast = toast;
        this.closeButton = true;
        this.toastHeaderClass = true;
    }
}
ToastHeaderComponent.ɵfac = function ToastHeaderComponent_Factory(t) { return new (t || ToastHeaderComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ToastComponent, 8)); };
ToastHeaderComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ToastHeaderComponent, selectors: [["c-toast-header"]], hostVars: 2, hostBindings: function ToastHeaderComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("toast-header", ctx.toastHeaderClass);
    } }, inputs: { closeButton: "closeButton" }, ngContentSelectors: _c1, decls: 3, vars: 1, consts: [["class", "close mfs-auto", "aria-label", "close", 3, "style", "cToastClose", 4, "ngIf"], ["aria-label", "close", 1, "close", "mfs-auto", 3, "cToastClose"]], template: function ToastHeaderComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementContainerStart(0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵtemplate(2, ToastHeaderComponent_button_2_Template, 2, 4, "button", 0);
        ɵngcc0.ɵɵelementContainerEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.closeButton);
    } }, directives: [ɵngcc2.NgIf, ɵngcc3.ToastCloseDirective], styles: [""] });
ToastHeaderComponent.ctorParameters = () => [
    { type: ToastComponent, decorators: [{ type: Optional }] }
];
ToastHeaderComponent.propDecorators = {
    closeButton: [{ type: Input }],
    toastHeaderClass: [{ type: HostBinding, args: ['class.toast-header',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ToastHeaderComponent, [{
        type: Component,
        args: [{
                selector: 'c-toast-header',
                template: "<ng-container>\r\n  <ng-content></ng-content>\r\n  <button *ngIf=\"closeButton\" class=\"close mfs-auto\" aria-label=\"close\" [style]=\"{outline: 0}\" [cToastClose]=\"toast\">\u00D7</button>\r\n</ng-container>\r\n",
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc1.ToastComponent, decorators: [{
                type: Optional
            }] }]; }, { closeButton: [{
            type: Input
        }], toastHeaderClass: [{
            type: HostBinding,
            args: ['class.toast-header']
        }] }); })();
