// @coreui/angular/esm2015/lib/form/input-group-text/input-group-text.component.js
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class InputGroupTextComponent {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = {
            'input-group-text': true,
        };
        return classes;
    }
    ngOnInit() { }
}
InputGroupTextComponent.ɵfac = function InputGroupTextComponent_Factory(t) { return new (t || InputGroupTextComponent)(); };
InputGroupTextComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: InputGroupTextComponent, selectors: [["c-input-group-text"]], hostVars: 2, hostBindings: function InputGroupTextComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function InputGroupTextComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
InputGroupTextComponent.ctorParameters = () => [];
InputGroupTextComponent.propDecorators = {
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(InputGroupTextComponent, [{
        type: Component,
        args: [{
                selector: 'c-input-group-text',
                template: "<ng-content></ng-content>\n",
                styles: [""]
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }] }); })();
