// @coreui/angular/esm2015/lib/toggler/toggler.component.js
import { Component, ChangeDetectorRef, HostBinding, ViewChild, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

const _c0 = ["content"];
function TogglerComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 2, 3);
    ɵngcc0.ɵɵprojection(2);
    ɵngcc0.ɵɵelementEnd();
} }
function TogglerComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "button", 4);
    ɵngcc0.ɵɵelement(1, "span");
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap(ctx_r2.buttonClass);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ctx_r2.iconClass);
} }
const _c1 = ["*"];
export class TogglerComponent {
    constructor(changeDetector) {
        this.changeDetector = changeDetector;
        this.displayFlex = true;
        this.hasContent = true;
    }
    get buttonClass() {
        if (this.inHeader) {
            return 'c-header-toggler';
        }
        else if (this.inNavbar) {
            return 'navbar-toggler';
        }
    }
    get iconClass() {
        if (this.inHeader) {
            return 'c-header-toggler-icon';
        }
        else if (this.inNavbar) {
            return 'navbar-toggler-icon';
        }
    }
    ngAfterViewInit() {
        this.hasContent = this.content.nativeElement.childNodes.length;
        this.changeDetector.detectChanges();
    }
}
TogglerComponent.ɵfac = function TogglerComponent_Factory(t) { return new (t || TogglerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef)); };
TogglerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TogglerComponent, selectors: [["c-toggler"]], viewQuery: function TogglerComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, 1);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.content = _t.first);
    } }, hostVars: 2, hostBindings: function TogglerComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("d-flex", ctx.displayFlex);
    } }, inputs: { inHeader: "inHeader", inNavbar: "inNavbar" }, ngContentSelectors: _c1, decls: 3, vars: 2, consts: [["class", "d-flex", 4, "ngIf", "ngIfElse"], ["default", ""], [1, "d-flex"], ["content", ""], ["type", "button"]], template: function TogglerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵtemplate(0, TogglerComponent_div_0_Template, 3, 0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, TogglerComponent_ng_template_1_Template, 2, 4, "ng-template", null, 1, ɵngcc0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        const _r1 = ɵngcc0.ɵɵreference(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasContent)("ngIfElse", _r1);
    } }, directives: [ɵngcc1.NgIf], styles: [""] });
TogglerComponent.ctorParameters = () => [
    { type: ChangeDetectorRef }
];
TogglerComponent.propDecorators = {
    inHeader: [{ type: Input }],
    inNavbar: [{ type: Input }],
    displayFlex: [{ type: HostBinding, args: ['class.d-flex',] }],
    content: [{ type: ViewChild, args: ['content',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TogglerComponent, [{
        type: Component,
        args: [{
                selector: 'c-toggler',
                template: "<div #content *ngIf = \"hasContent; else default\" class=\"d-flex\"><ng-content></ng-content></div>\n<ng-template #default>\n  <button [class]=\"buttonClass\" type=\"button\">\n    <span [class]=\"iconClass\"></span>\n  </button>\n</ng-template>\n",
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc0.ChangeDetectorRef }]; }, { displayFlex: [{
            type: HostBinding,
            args: ['class.d-flex']
        }], inHeader: [{
            type: Input
        }], inNavbar: [{
            type: Input
        }], content: [{
            type: ViewChild,
            args: ['content']
        }] }); })();
