// @coreui/angular/esm2015/lib/toggler/toggler.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TogglerComponent } from './toggler.component';
import * as ɵngcc0 from '@angular/core';
export class TogglerModule {
}
TogglerModule.ɵfac = function TogglerModule_Factory(t) { return new (t || TogglerModule)(); };
TogglerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TogglerModule });
TogglerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TogglerModule, { declarations: function () { return [TogglerComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [TogglerComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TogglerModule, [{
        type: NgModule,
        args: [{
                declarations: [TogglerComponent],
                exports: [
                    TogglerComponent
                ],
                imports: [
                    CommonModule
                ]
            }]
    }], null, null); })();
