// ngx-spinner/esm5/lib/ngx-spinner.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxSpinnerComponent } from './ngx-spinner.component';
import * as ɵngcc0 from '@angular/core';
var NgxSpinnerModule = /** @class */ (function () {
    function NgxSpinnerModule() {
    }
NgxSpinnerModule.ɵfac = function NgxSpinnerModule_Factory(t) { return new (t || NgxSpinnerModule)(); };
NgxSpinnerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxSpinnerModule });
NgxSpinnerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxSpinnerModule, { declarations: function () { return [NgxSpinnerComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxSpinnerComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxSpinnerModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [NgxSpinnerComponent],
                exports: [NgxSpinnerComponent]
            }]
    }], function () { return []; }, null); })();
    return NgxSpinnerModule;
}());
export { NgxSpinnerModule };
