// @coreui/angular/esm2015/lib/form/label/label.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import { chkBoolProp } from '../../shared/check-bool-prop';
import * as ɵngcc0 from '@angular/core';
export class LabelDirective {
    constructor() {
        this.col = '';
        this.sizing = '';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const isCustom = chkBoolProp(this.custom);
        const classes = classNames({
            'custom-control-label': isCustom,
            'custom-file-label': isCustom && this.variant === 'file',
            'form-check-label': !isCustom && ['checkbox', 'radio', 'switch'].includes(this.variant),
            'sr-only': chkBoolProp(this.hidden),
            'visually-hidden': chkBoolProp(this.hidden),
            'col-form-label': this.col === 'col',
            [`col-form-label-${this.sizing}`]: !!this.sizing && this.col === 'col',
        });
        return classes;
    }
}
LabelDirective.ɵfac = function LabelDirective_Factory(t) { return new (t || LabelDirective)(); };
LabelDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: LabelDirective, selectors: [["", "cLabel", ""]], hostVars: 2, hostBindings: function LabelDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { col: ["cLabel", "col"], sizing: "sizing", hidden: "hidden", variant: "variant", custom: "custom" } });
LabelDirective.ctorParameters = () => [];
LabelDirective.propDecorators = {
    col: [{ type: Input, args: ['cLabel',] }],
    hidden: [{ type: Input }],
    variant: [{ type: Input }],
    sizing: [{ type: Input }],
    custom: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(LabelDirective, [{
        type: Directive,
        args: [{
                selector: '[cLabel]'
            }]
    }], function () { return []; }, { col: [{
            type: Input,
            args: ['cLabel']
        }], sizing: [{
            type: Input
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], hidden: [{
            type: Input
        }], variant: [{
            type: Input
        }], custom: [{
            type: Input
        }] }); })();
