// @coreui/angular/esm2015/lib/tabset/tab-pane/tab-pane.component.js
import { Component, HostBinding, Input } from '@angular/core';
import { TabsetService } from '../tabset.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../tabset.service';
import * as ɵngcc2 from '@angular/common';

function TabPaneComponent_ng_content_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0, 0, ["*ngIf", "active"]);
} }
const _c0 = ["*"];
export class TabPaneComponent {
    constructor(tabsetService) {
        this.tabsetService = tabsetService;
        this.fade = true;
        this._tabPane = true;
        this._tabActive = true;
        this._tabShow = false;
        this._tabFade = this.fade;
    }
    set active(active) {
        this._tabActive = active;
        this._tabShow = active;
    }
    get active() {
        return this._tabActive;
    }
    ngOnInit() {
        this.activeTabSubscribe();
    }
    ngOnDestroy() {
        this.activeTabUnsubscribe();
    }
    activeTabSubscribe() {
        this.activeTabSubscription = this.tabsetService.activeTab$.subscribe((activeTab) => {
            setTimeout(() => {
                this.active = activeTab.indexKey === this.indexKey;
            });
        });
    }
    activeTabUnsubscribe() {
        this.activeTabSubscription.unsubscribe();
    }
}
TabPaneComponent.ɵfac = function TabPaneComponent_Factory(t) { return new (t || TabPaneComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.TabsetService)); };
TabPaneComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TabPaneComponent, selectors: [["c-tab-pane"]], hostVars: 8, hostBindings: function TabPaneComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("tab-pane", ctx._tabPane)("active", ctx._tabActive)("show", ctx._tabShow)("fade", ctx._tabFade);
    } }, inputs: { active: "active" }, ngContentSelectors: _c0, decls: 1, vars: 1, consts: [[4, "ngIf"]], template: function TabPaneComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵtemplate(0, TabPaneComponent_ng_content_0_Template, 1, 0, "ng-content", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.active);
    } }, directives: [ɵngcc2.NgIf], styles: [""] });
TabPaneComponent.ctorParameters = () => [
    { type: TabsetService }
];
TabPaneComponent.propDecorators = {
    active: [{ type: Input }],
    _tabPane: [{ type: HostBinding, args: ['class.tab-pane',] }],
    _tabActive: [{ type: HostBinding, args: ['class.active',] }],
    _tabShow: [{ type: HostBinding, args: ['class.show',] }],
    _tabFade: [{ type: HostBinding, args: ['class.fade',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TabPaneComponent, [{
        type: Component,
        args: [{
                selector: 'c-tab-pane',
                template: "<ng-content *ngIf=\"active\"></ng-content>\n\n",
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc1.TabsetService }]; }, { _tabPane: [{
            type: HostBinding,
            args: ['class.tab-pane']
        }], _tabActive: [{
            type: HostBinding,
            args: ['class.active']
        }], _tabShow: [{
            type: HostBinding,
            args: ['class.show']
        }], _tabFade: [{
            type: HostBinding,
            args: ['class.fade']
        }], active: [{
            type: Input
        }] }); })();
