// @coreui/angular/esm2015/lib/spinner/spinner.component.js
import { Component, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';
export class SpinnerComponent {
    constructor() {
        this.color = '';
        this.grow = false;
        this.size = '';
        this.role = 'status';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const shape = this.grow ? 'grow' : 'border';
        const classes = classNames({
            [`spinner-${shape}`]: true,
            [`text-${this.color}`]: !!this.color,
            [`spinner-${shape}-${this.size}`]: !!this.size
        });
        return classes;
    }
}
SpinnerComponent.ɵfac = function SpinnerComponent_Factory(t) { return new (t || SpinnerComponent)(); };
SpinnerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SpinnerComponent, selectors: [["c-spinner"]], hostVars: 3, hostBindings: function SpinnerComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("role", ctx.role);
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { color: "color", grow: "grow", size: "size", role: "role" }, decls: 2, vars: 0, consts: [[1, "sr-only"]], template: function SpinnerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "span", 0);
        ɵngcc0.ɵɵtext(1, "Loading...");
        ɵngcc0.ɵɵelementEnd();
    } }, styles: [""] });
SpinnerComponent.propDecorators = {
    color: [{ type: Input }],
    grow: [{ type: Input }],
    size: [{ type: Input }],
    role: [{ type: Input }, { type: HostBinding, args: ['attr.role',] }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinnerComponent, [{
        type: Component,
        args: [{
                selector: 'c-spinner',
                template: "<!--<div [ngClass]=\"classes\" [attr.role]=\"role\" [ngStyle]=\"styles\">-->\n  <span class=\"sr-only\">Loading...</span>\n<!--</div>-->\n",
                styles: [""]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }], grow: [{
            type: Input
        }], size: [{
            type: Input
        }], role: [{
            type: Input
        }, {
            type: HostBinding,
            args: ['attr.role']
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }] }); })();
