// @coreui/angular/esm2015/lib/navbar/navbar-brand/navbar-brand.component.js
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class NavbarBrandComponent {
    constructor() {
        this.navbarBrand = true;
        this.role = 'button';
    }
    ngOnInit() { }
}
NavbarBrandComponent.ɵfac = function NavbarBrandComponent_Factory(t) { return new (t || NavbarBrandComponent)(); };
NavbarBrandComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NavbarBrandComponent, selectors: [["c-navbar-brand"]], hostVars: 3, hostBindings: function NavbarBrandComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("role", ctx.role);
        ɵngcc0.ɵɵclassProp("navbar-brand", ctx.navbarBrand);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function NavbarBrandComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]{outline:inherit}"] });
NavbarBrandComponent.ctorParameters = () => [];
NavbarBrandComponent.propDecorators = {
    navbarBrand: [{ type: HostBinding, args: ['class.navbar-brand',] }],
    role: [{ type: HostBinding, args: ['attr.role',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NavbarBrandComponent, [{
        type: Component,
        args: [{
                selector: 'c-navbar-brand',
                template: "<ng-content></ng-content>\n",
                styles: [":host{outline:inherit}"]
            }]
    }], function () { return []; }, { navbarBrand: [{
            type: HostBinding,
            args: ['class.navbar-brand']
        }], role: [{
            type: HostBinding,
            args: ['attr.role']
        }] }); })();
