// @coreui/angular/esm2015/lib/carousel/carousel-inner/carousel-inner.component.js
import { Component, ContentChildren, HostBinding, } from '@angular/core';
import { slideAnimation } from '../carousel.animation';
import { CarouselItemComponent } from '../carousel-item/carousel-item.component';
import { CarouselState } from '../carousel-state';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../carousel-state';

const _c0 = ["*"];
export class CarouselInnerComponent {
    constructor(carouselState) {
        this.carouselState = carouselState;
        this.carouselInnerClass = true;
        this.slide = { left: true };
    }
    ngOnInit() { }
    ngAfterContentInit() {
        this.setItems();
    }
    ngAfterContentChecked() {
        var _a;
        this.setItems();
        const state = (_a = this.carouselState) === null || _a === void 0 ? void 0 : _a.state;
        const nextIndex = state === null || state === void 0 ? void 0 : state.activeItemIndex;
        const nextDirection = state === null || state === void 0 ? void 0 : state.direction;
        if (this.activeIndex !== nextIndex) {
            this.animate = state === null || state === void 0 ? void 0 : state.animate;
            this.slide = { left: nextDirection === 'next' };
            this.activeIndex = state === null || state === void 0 ? void 0 : state.activeItemIndex;
        }
    }
    setItems() {
        if (this.prevContentItems !== this.contentItems) {
            this.prevContentItems = this.contentItems;
            this.carouselState.setItems(this.contentItems);
        }
    }
}
CarouselInnerComponent.ɵfac = function CarouselInnerComponent_Factory(t) { return new (t || CarouselInnerComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CarouselState)); };
CarouselInnerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CarouselInnerComponent, selectors: [["c-carousel-inner"]], contentQueries: function CarouselInnerComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, CarouselItemComponent, 0);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.contentItems = _t);
    } }, hostVars: 2, hostBindings: function CarouselInnerComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("carousel-inner", ctx.carouselInnerClass);
    } }, ngContentSelectors: _c0, decls: 2, vars: 2, template: function CarouselInnerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div");
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("@slideAnimation", ctx.slide)("@.disabled", !ctx.animate);
    } }, styles: ["[_nghost-%COMP%]{display:block}"], data: { animation: [slideAnimation] } });
CarouselInnerComponent.ctorParameters = () => [
    { type: CarouselState }
];
CarouselInnerComponent.propDecorators = {
    carouselInnerClass: [{ type: HostBinding, args: ['class.carousel-inner',] }],
    contentItems: [{ type: ContentChildren, args: [CarouselItemComponent,] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CarouselInnerComponent, [{
        type: Component,
        args: [{
                selector: 'c-carousel-inner',
                template: "<div [@slideAnimation]=\"slide\" [@.disabled]=\"!animate\">\n  <ng-content></ng-content>\n</div>\n",
                animations: [slideAnimation],
                styles: [":host{display:block}"]
            }]
    }], function () { return [{ type: ɵngcc1.CarouselState }]; }, { carouselInnerClass: [{
            type: HostBinding,
            args: ['class.carousel-inner']
        }], contentItems: [{
            type: ContentChildren,
            args: [CarouselItemComponent]
        }] }); })();
