// @coreui/angular/esm2015/lib/image/img.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ImgDirective } from './img.directive';
import * as ɵngcc0 from '@angular/core';
export class ImgModule {
}
ImgModule.ɵfac = function ImgModule_Factory(t) { return new (t || ImgModule)(); };
ImgModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ImgModule });
ImgModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ImgModule, { declarations: function () { return [ImgDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ImgDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ImgModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                exports: [
                    ImgDirective
                ],
                declarations: [
                    ImgDirective
                ]
            }]
    }], null, null); })();
