// @coreui/angular/esm2015/lib/image/img.directive.js
import { Directive, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';
export class ImgDirective {
    constructor() {
        this.align = '';
        this.placeholderColor = 'transparent';
    }
    get getStyles() {
        return { backgroundColor: this.placeholderColor };
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames(this.alignClass, {
            'img-thumbnail': this.thumbnail,
            'img-fluid': this.fluid || this.fluidGrow,
            'w-100': this.fluidGrow,
            'd-block': this.block,
            [`${this.shape}`]: this.shape,
        });
        return classes;
    }
    get alignClass() {
        return this.align === 'center' ? 'mx-auto' :
            this.align === 'right' ? 'float-right' :
                this.align === 'left' ? 'float-left' : '';
    }
    ngOnInit() { }
}
ImgDirective.ɵfac = function ImgDirective_Factory(t) { return new (t || ImgDirective)(); };
ImgDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ImgDirective, selectors: [["", "cImg", ""]], hostVars: 4, hostBindings: function ImgDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵstyleMap(ctx.getStyles);
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { align: "align", placeholderColor: "placeholderColor", block: "block", fluid: "fluid", fluidGrow: "fluidGrow", shape: "shape", thumbnail: "thumbnail" } });
ImgDirective.ctorParameters = () => [];
ImgDirective.propDecorators = {
    block: [{ type: Input }],
    fluid: [{ type: Input }],
    fluidGrow: [{ type: Input }],
    shape: [{ type: Input }],
    thumbnail: [{ type: Input }],
    align: [{ type: Input }],
    placeholderColor: [{ type: Input }],
    getStyles: [{ type: HostBinding, args: ['style',] }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ImgDirective, [{
        type: Directive,
        args: [{
                selector: '[cImg]'
            }]
    }], function () { return []; }, { align: [{
            type: Input
        }], placeholderColor: [{
            type: Input
        }], getStyles: [{
            type: HostBinding,
            args: ['style']
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }], block: [{
            type: Input
        }], fluid: [{
            type: Input
        }], fluidGrow: [{
            type: Input
        }], shape: [{
            type: Input
        }], thumbnail: [{
            type: Input
        }] }); })();
