// ngx-tippy-wrapper/esm2015/lib/ngx-tippy.directive.js
import { Directive, ElementRef, Input, Renderer2, Inject, PLATFORM_ID } from '@angular/core';
import { isPlatformServer } from '@angular/common';
import tippy from 'tippy.js';
import { NgxTippyService } from './ngx-tippy.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './ngx-tippy.service';
export class NgxTippyDirective {
    constructor(tippyEl, ngxTippyService, renderer, platform) {
        this.tippyEl = tippyEl;
        this.ngxTippyService = ngxTippyService;
        this.renderer = renderer;
        this.platform = platform;
    }
    ngOnInit() {
        if (isPlatformServer(this.platform))
            return;
        this.initTippy();
    }
    /**
     * Tooltip initialize
     * Content can be directly passed through `ngxTippy` selector
     */
    initTippy() {
        if (this.ngxTippy === null || this.ngxTippy === undefined)
            return;
        const tippyTarget = this.tippyEl.nativeElement;
        const tippyTemplate = this.ngxTippy;
        tippy(tippyTarget, Object.assign(Object.assign({}, (this.tippyProps || {})), (tippyTemplate && { content: tippyTemplate })));
        this.ngxTippyService.setTemplateVisible(tippyTemplate);
        this.setTippyInstance(tippyTarget);
    }
    setTippyInstance(tippyTarget) {
        const tippyInstance = tippyTarget._tippy;
        this.writeInstancesToStorage(tippyInstance);
        this.setClassName(tippyInstance);
    }
    setClassName(tippyInstance) {
        if (!this.tippyClassName)
            return;
        const classNames = this.tippyClassName.split(' ');
        classNames.length &&
            classNames.forEach((className) => {
                this.renderer.addClass(tippyInstance.popper.firstElementChild, className);
            });
    }
    /**
     * To manipulate tooltips, write all instances to storage
     * `tippyName` used as unique key
     * If `tippyName` does not provided - it will be generated using `tippyInstance.id`
     *
     * @param tippyInstance { NgxTippyInstance }
     */
    writeInstancesToStorage(tippyInstance) {
        this.ngxTippyService.setInstance(this.tippyName || `tippy-${tippyInstance.id}`, tippyInstance);
    }
}
NgxTippyDirective.ɵfac = function NgxTippyDirective_Factory(t) { return new (t || NgxTippyDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NgxTippyService), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(PLATFORM_ID)); };
NgxTippyDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: NgxTippyDirective, selectors: [["", "ngxTippy", ""]], inputs: { ngxTippy: "ngxTippy", tippyProps: "tippyProps", tippyName: "tippyName", tippyClassName: "tippyClassName" } });
NgxTippyDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: NgxTippyService },
    { type: Renderer2 },
    { type: Object, decorators: [{ type: Inject, args: [PLATFORM_ID,] }] }
];
NgxTippyDirective.propDecorators = {
    ngxTippy: [{ type: Input }],
    tippyProps: [{ type: Input }],
    tippyName: [{ type: Input }],
    tippyClassName: [{ type: Input }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxTippyDirective, [{
        type: Directive,
        args: [{
                selector: '[ngxTippy]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc1.NgxTippyService }, { type: ɵngcc0.Renderer2 }, { type: Object, decorators: [{
                type: Inject,
                args: [PLATFORM_ID]
            }] }]; }, { ngxTippy: [{
            type: Input
        }], tippyProps: [{
            type: Input
        }], tippyName: [{
            type: Input
        }], tippyClassName: [{
            type: Input
        }] }); })();
