// ngx-tippy-wrapper/esm2015/lib/ngx-tippy.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NgxTippySingletonComponent } from './ngx-tippy-singleton.component';
import { NgxTippyGroupComponent } from './ngx-tippy-group.component';
import { NgxTippyDirective } from './ngx-tippy.directive';
import * as ɵngcc0 from '@angular/core';
export class NgxTippyModule {
}
NgxTippyModule.ɵfac = function NgxTippyModule_Factory(t) { return new (t || NgxTippyModule)(); };
NgxTippyModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgxTippyModule });
NgxTippyModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ providers: [], imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgxTippyModule, { declarations: function () { return [NgxTippyDirective, NgxTippyGroupComponent, NgxTippySingletonComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NgxTippyDirective, NgxTippyGroupComponent, NgxTippySingletonComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxTippyModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [NgxTippyDirective, NgxTippyGroupComponent, NgxTippySingletonComponent],
                exports: [NgxTippyDirective, NgxTippyGroupComponent, NgxTippySingletonComponent],
                providers: []
            }]
    }], null, null); })();
