// ngx-tippy-wrapper/esm2015/lib/ngx-tippy-group.component.js
import { Component, ElementRef, Input, ViewChild, Inject, PLATFORM_ID } from '@angular/core';
import { isPlatformServer } from '@angular/common';
import tippy from 'tippy.js';
/**
 * This component implements next case: different tooltip content to many different elements, while only needing to initialize once with shared props
 */
import * as ɵngcc0 from '@angular/core';

const _c0 = ["contentWrapper"];
const _c1 = ["*"];
export class NgxTippyGroupComponent {
    constructor(platform) {
        this.platform = platform;
    }
    ngAfterViewInit() {
        if (isPlatformServer(this.platform))
            return;
        this.setTooltips();
    }
    setTooltips() {
        const contentWrapperNativeEl = this.contentWrapper.nativeElement;
        const tooltips = Array.from(contentWrapperNativeEl.querySelectorAll('[data-grouped]'));
        this.initTippy(tooltips);
    }
    initTippy(tooltips) {
        tippy(tooltips, this.tippyProps);
    }
}
NgxTippyGroupComponent.ɵfac = function NgxTippyGroupComponent_Factory(t) { return new (t || NgxTippyGroupComponent)(ɵngcc0.ɵɵdirectiveInject(PLATFORM_ID)); };
NgxTippyGroupComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NgxTippyGroupComponent, selectors: [["ngx-tippy-group"]], viewQuery: function NgxTippyGroupComponent_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(_c0, 1, ElementRef);
    } if (rf & 2) {
        let _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.contentWrapper = _t.first);
    } }, inputs: { tippyProps: "tippyProps" }, ngContentSelectors: _c1, decls: 3, vars: 0, consts: [["contentWrapper", ""]], template: function NgxTippyGroupComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", null, 0);
        ɵngcc0.ɵɵprojection(2);
        ɵngcc0.ɵɵelementEnd();
    } }, encapsulation: 2 });
NgxTippyGroupComponent.ctorParameters = () => [
    { type: Object, decorators: [{ type: Inject, args: [PLATFORM_ID,] }] }
];
NgxTippyGroupComponent.propDecorators = {
    tippyProps: [{ type: Input }],
    contentWrapper: [{ type: ViewChild, args: ['contentWrapper', { read: ElementRef, static: false },] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NgxTippyGroupComponent, [{
        type: Component,
        args: [{
                selector: 'ngx-tippy-group',
                template: `
    <div #contentWrapper>
      <ng-content></ng-content>
    </div>
  `
            }]
    }], function () { return [{ type: Object, decorators: [{
                type: Inject,
                args: [PLATFORM_ID]
            }] }]; }, { tippyProps: [{
            type: Input
        }], contentWrapper: [{
            type: ViewChild,
            args: ['contentWrapper', { read: ElementRef, static: false }]
        }] }); })();
