// ngx-perfect-scrollbar/esm2015/lib/perfect-scrollbar-force-native-scroll.directive.js
import { Directive, ElementRef, Renderer2 } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class ForceNativeScrollDirective {
    constructor(renderer, el) {
        this.renderer = renderer;
        ['ps__child', 'ps__child--consume'].forEach((className) => {
            this.renderer.addClass(el === null || el === void 0 ? void 0 : el.nativeElement, className);
        });
    }
}
ForceNativeScrollDirective.ɵfac = function ForceNativeScrollDirective_Factory(t) { return new (t || ForceNativeScrollDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ForceNativeScrollDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ForceNativeScrollDirective, selectors: [["", "forceNativeScrolling", ""]] });
ForceNativeScrollDirective.ctorParameters = () => [
    { type: Renderer2 },
    { type: ElementRef }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ForceNativeScrollDirective, [{
        type: Directive,
        args: [{
                selector: '[forceNativeScrolling]'
            }]
    }], function () { return [{ type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ElementRef }]; }, null); })();
