// @coreui/angular/esm2015/lib/toast/toast-close.directive.js
import { Directive, HostListener, Input } from '@angular/core';
import { ToasterService } from './toaster/toaster.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './toaster/toaster.service';
export class ToastCloseDirective {
    constructor(toasterService) {
        this.toasterService = toasterService;
    }
    toggleOpen($event) {
        $event.preventDefault();
        this.toasterService.setState({ show: false, toast: this.toast });
    }
}
ToastCloseDirective.ɵfac = function ToastCloseDirective_Factory(t) { return new (t || ToastCloseDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ToasterService)); };
ToastCloseDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ToastCloseDirective, selectors: [["", "cToastClose", ""]], hostBindings: function ToastCloseDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function ToastCloseDirective_click_HostBindingHandler($event) { return ctx.toggleOpen($event); });
    } }, inputs: { toast: ["cToastClose", "toast"] }, exportAs: ["cToastClose"] });
ToastCloseDirective.ctorParameters = () => [
    { type: ToasterService }
];
ToastCloseDirective.propDecorators = {
    toast: [{ type: Input, args: ['cToastClose',] }],
    toggleOpen: [{ type: HostListener, args: ['click', ['$event'],] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(ToastCloseDirective, [{
        type: Directive,
        args: [{
                selector: '[cToastClose]',
                exportAs: 'cToastClose'
            }]
    }], function () { return [{ type: ɵngcc1.ToasterService }]; }, { toggleOpen: [{
            type: HostListener,
            args: ['click', ['$event']]
        }], toast: [{
            type: Input,
            args: ['cToastClose']
        }] }); })();
