// @pascalhonegger/ng-datatable/esm2015/lib/DataTableModule.js
import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { DataTable } from "./DataTable";
import { DefaultSorter } from "./DefaultSorter";
import { Paginator } from "./Paginator";
import { BootstrapPaginator } from "./BootstrapPaginator";
import * as ɵngcc0 from '@angular/core';
export class DataTableModule {
}
DataTableModule.ɵfac = function DataTableModule_Factory(t) { return new (t || DataTableModule)(); };
DataTableModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DataTableModule });
DataTableModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DataTableModule, { declarations: function () { return [DataTable,
        DefaultSorter,
        Paginator,
        BootstrapPaginator]; }, imports: function () { return [CommonModule]; }, exports: function () { return [DataTable,
        DefaultSorter,
        Paginator,
        BootstrapPaginator]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DataTableModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    DataTable,
                    DefaultSorter,
                    Paginator,
                    BootstrapPaginator
                ],
                exports: [
                    DataTable,
                    DefaultSorter,
                    Paginator,
                    BootstrapPaginator
                ]
            }]
    }], null, null); })();
