// @pascalhonegger/ng-datatable/esm2015/lib/DefaultSorter.js
import { Component, Input } from "@angular/core";
import { DataTable } from "./DataTable";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './DataTable';
import * as ɵngcc2 from '@angular/common';

function DefaultSorter_span_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 3);
} }
function DefaultSorter_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 4);
} }
const _c0 = ["*"];
export class DefaultSorter {
    constructor(mfTable) {
        this.mfTable = mfTable;
        this.isSortedByMeAsc = false;
        this.isSortedByMeDesc = false;
    }
    ngOnInit() {
        this.mfTable.onSortChange.subscribe((event) => {
            // tslint:disable:triple-equals
            this.isSortedByMeAsc = (event.sortBy == this.sortBy && event.sortOrder == "asc");
            this.isSortedByMeDesc = (event.sortBy == this.sortBy && event.sortOrder == "desc");
            // tslint:enable:triple-equals
        });
    }
    sort() {
        if (this.isSortedByMeAsc) {
            this.mfTable.setSort(this.sortBy, "desc");
        }
        else {
            this.mfTable.setSort(this.sortBy, "asc");
        }
    }
}
DefaultSorter.ɵfac = function DefaultSorter_Factory(t) { return new (t || DefaultSorter)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DataTable)); };
DefaultSorter.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DefaultSorter, selectors: [["mfDefaultSorter"]], inputs: { sortBy: ["by", "sortBy"] }, ngContentSelectors: _c0, decls: 4, vars: 2, consts: [[1, "text-nowrap", 2, "cursor", "pointer", 3, "click"], ["class", "glyphicon glyphicon-triangle-top", "aria-hidden", "true", 4, "ngIf"], ["class", "glyphicon glyphicon-triangle-bottom", "aria-hidden", "true", 4, "ngIf"], ["aria-hidden", "true", 1, "glyphicon", "glyphicon-triangle-top"], ["aria-hidden", "true", 1, "glyphicon", "glyphicon-triangle-bottom"]], template: function DefaultSorter_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "a", 0);
        ɵngcc0.ɵɵlistener("click", function DefaultSorter_Template_a_click_0_listener() { return ctx.sort(); });
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵtemplate(2, DefaultSorter_span_2_Template, 1, 0, "span", 1);
        ɵngcc0.ɵɵtemplate(3, DefaultSorter_span_3_Template, 1, 0, "span", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isSortedByMeAsc);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.isSortedByMeDesc);
    } }, directives: [ɵngcc2.NgIf], encapsulation: 2 });
DefaultSorter.ctorParameters = () => [
    { type: DataTable }
];
DefaultSorter.propDecorators = {
    sortBy: [{ type: Input, args: ["by",] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DefaultSorter, [{
        type: Component,
        args: [{
                selector: "mfDefaultSorter",
                template: `
        <a style="cursor: pointer" (click)="sort()" class="text-nowrap">
            <ng-content></ng-content>
            <span *ngIf="isSortedByMeAsc" class="glyphicon glyphicon-triangle-top" aria-hidden="true"></span>
            <span *ngIf="isSortedByMeDesc" class="glyphicon glyphicon-triangle-bottom" aria-hidden="true"></span>
        </a>`
            }]
    }], function () { return [{ type: ɵngcc1.DataTable }]; }, { sortBy: [{
            type: Input,
            args: ["by"]
        }] }); })();
