// @pascalhonegger/ng-datatable/esm2015/lib/Paginator.js
import { Component, Input, Optional } from "@angular/core";
import { DataTable } from "./DataTable";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './DataTable';

const _c0 = ["*"];
export class Paginator {
    constructor(injectMfTable) {
        this.injectMfTable = injectMfTable;
        this.dataLength = 0;
        this.onPageChangeSubscriber = (event) => {
            this.activePage = event.activePage;
            this.rowsOnPage = event.rowsOnPage;
            this.dataLength = event.dataLength;
            this.lastPage = Math.ceil(this.dataLength / this.rowsOnPage);
        };
    }
    ngOnChanges(changes) {
        this.mfTable = this.inputMfTable || this.injectMfTable;
        this.onPageChangeSubscriber(this.mfTable.getPage());
        this.mfTable.onPageChange.subscribe(this.onPageChangeSubscriber);
    }
    setPage(pageNumber) {
        this.mfTable.setPage(pageNumber, this.rowsOnPage);
    }
    setRowsOnPage(rowsOnPage) {
        this.mfTable.setPage(this.activePage, rowsOnPage);
    }
}
Paginator.ɵfac = function Paginator_Factory(t) { return new (t || Paginator)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DataTable, 8)); };
Paginator.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: Paginator, selectors: [["mfPaginator"]], inputs: { inputMfTable: ["mfTable", "inputMfTable"] }, features: [ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function Paginator_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
Paginator.ctorParameters = () => [
    { type: DataTable, decorators: [{ type: Optional }] }
];
Paginator.propDecorators = {
    inputMfTable: [{ type: Input, args: ["mfTable",] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(Paginator, [{
        type: Component,
        args: [{
                selector: "mfPaginator",
                template: `<ng-content></ng-content>`
            }]
    }], function () { return [{ type: ɵngcc1.DataTable, decorators: [{
                type: Optional
            }] }]; }, { inputMfTable: [{
            type: Input,
            args: ["mfTable"]
        }] }); })();
