// @coreui/angular/esm2015/lib/dropdown/dropdown.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropdownDirective } from './dropdown.directive';
import { DropdownService } from './dropdown.service';
import { DropdownState } from './dropdown.state';
import { DropdownMenuDirective } from './dropdown-menu.directive';
import { DropdownToggleDirective } from './dropdown-toggle.directive';
import { DropdownHeaderDirective } from './dropdown-header.directive';
import { DropdownDividerDirective } from './dropdown-divider.directive';
import { DropdownItemDirective } from './dropdown-item.directive';
import { DropdownDividerComponent } from './dropdown-divider.component';
import * as ɵngcc0 from '@angular/core';
export class DropdownModule {
    static forRoot(config) {
        return {
            ngModule: DropdownModule,
            providers: []
        };
    }
}
DropdownModule.ɵfac = function DropdownModule_Factory(t) { return new (t || DropdownModule)(); };
DropdownModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DropdownModule });
DropdownModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ providers: [
        DropdownService,
        DropdownState
    ], imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DropdownModule, { declarations: function () { return [DropdownDirective,
        DropdownToggleDirective,
        DropdownMenuDirective,
        DropdownHeaderDirective,
        DropdownDividerDirective,
        DropdownItemDirective,
        DropdownDividerComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [DropdownDirective,
        DropdownToggleDirective,
        DropdownMenuDirective,
        DropdownItemDirective,
        DropdownDividerDirective,
        DropdownHeaderDirective,
        DropdownDividerComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DropdownModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    DropdownDirective,
                    DropdownToggleDirective,
                    DropdownMenuDirective,
                    DropdownHeaderDirective,
                    DropdownDividerDirective,
                    DropdownItemDirective,
                    DropdownDividerComponent
                ],
                exports: [
                    DropdownDirective,
                    DropdownToggleDirective,
                    DropdownMenuDirective,
                    DropdownItemDirective,
                    DropdownDividerDirective,
                    DropdownHeaderDirective,
                    DropdownDividerComponent
                ],
                imports: [
                    CommonModule
                ],
                providers: [
                    DropdownService,
                    DropdownState
                ]
            }]
    }], null, null); })();
