// @coreui/angular/esm2015/lib/dropdown/dropdown-item.directive.js
import { Directive, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class DropdownItemDirective {
    constructor() {
        this.dropdownItem = true;
    }
}
DropdownItemDirective.ɵfac = function DropdownItemDirective_Factory(t) { return new (t || DropdownItemDirective)(); };
DropdownItemDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: DropdownItemDirective, selectors: [["", "cDropdownItem", ""], ["c-dropdown-item"]], hostVars: 2, hostBindings: function DropdownItemDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("dropdown-item", ctx.dropdownItem);
    } }, exportAs: ["cDropdownItem"] });
DropdownItemDirective.propDecorators = {
    dropdownItem: [{ type: HostBinding, args: ['class.dropdown-item',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DropdownItemDirective, [{
        type: Directive,
        args: [{
                selector: '[cDropdownItem], c-dropdown-item',
                exportAs: 'cDropdownItem'
            }]
    }], function () { return []; }, { dropdownItem: [{
            type: HostBinding,
            args: ['class.dropdown-item']
        }] }); })();
