// @coreui/angular/esm2015/lib/dropdown/dropdown-toggle.directive.js
import { Directive, ElementRef, HostListener, Input, Renderer2 } from '@angular/core';
import { DropdownState } from './dropdown.state';
import { DropdownService } from './dropdown.service';
import { DropdownDirective } from './dropdown.directive';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './dropdown.directive';
import * as ɵngcc2 from './dropdown.service';
import * as ɵngcc3 from './dropdown.state';
export class DropdownToggleDirective {
    constructor(renderer, hostElement, dropdown, dropdownService, dropdownState) {
        this.renderer = renderer;
        this.hostElement = hostElement;
        this.dropdown = dropdown;
        this.dropdownService = dropdownService;
        this.dropdownState = dropdownState;
        this.cDropdownCaret = 'caret';
        this.host = this.hostElement.nativeElement;
    }
    ngOnInit() {
        this.state = this.dropdownState.setState({ caret: this.cDropdownCaret });
        const toggleClass = 'dropdown-toggle';
        if (this.state.caret === 'none') {
            this.renderer.removeClass(this.host, toggleClass);
        }
        else {
            this.renderer.addClass(this.host, toggleClass);
            if (this.state.caret === 'split') {
                this.renderer.addClass(this.host, `${toggleClass}-split`);
            }
        }
    }
    toggleShow($event) {
        $event.preventDefault();
        this.state = this.dropdownState.getState();
        this.dropdownService.toggle({ show: !this.state.show });
    }
}
DropdownToggleDirective.ɵfac = function DropdownToggleDirective_Factory(t) { return new (t || DropdownToggleDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DropdownDirective), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.DropdownService), ɵngcc0.ɵɵdirectiveInject(ɵngcc3.DropdownState)); };
DropdownToggleDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: DropdownToggleDirective, selectors: [["", "cDropdownToggle", ""], ["c-dropdown-toggle"]], hostBindings: function DropdownToggleDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function DropdownToggleDirective_click_HostBindingHandler($event) { return ctx.toggleShow($event); });
    } }, inputs: { cDropdownCaret: "cDropdownCaret" }, exportAs: ["cDropdownToggle"] });
DropdownToggleDirective.ctorParameters = () => [
    { type: Renderer2 },
    { type: ElementRef },
    { type: DropdownDirective },
    { type: DropdownService },
    { type: DropdownState }
];
DropdownToggleDirective.propDecorators = {
    cDropdownCaret: [{ type: Input }],
    toggleShow: [{ type: HostListener, args: ['click', ['$event'],] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DropdownToggleDirective, [{
        type: Directive,
        args: [{
                selector: '[cDropdownToggle], c-dropdown-toggle',
                exportAs: 'cDropdownToggle'
            }]
    }], function () { return [{ type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ElementRef }, { type: ɵngcc1.DropdownDirective }, { type: ɵngcc2.DropdownService }, { type: ɵngcc3.DropdownState }]; }, { cDropdownCaret: [{
            type: Input
        }], toggleShow: [{
            type: HostListener,
            args: ['click', ['$event']]
        }] }); })();
