// @coreui/angular/esm2015/lib/dropdown/dropdown-menu.directive.js
import { Directive, ElementRef, HostBinding, Renderer2 } from '@angular/core';
import { DropdownState } from './dropdown.state';
import { DropdownService } from './dropdown.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './dropdown.state';
import * as ɵngcc2 from './dropdown.service';
export class DropdownMenuDirective {
    constructor(renderer, hostElement, dropdownState, dropdownService) {
        this.renderer = renderer;
        this.hostElement = hostElement;
        this.dropdownState = dropdownState;
        this.dropdownService = dropdownService;
        this.role = 'menu';
        this.dropdownMenu = true;
        this.host = this.hostElement.nativeElement;
    }
    ngOnInit() {
        this.stateToggleSubscribe();
    }
    ngOnDestroy() {
        this.stateToggleSubscribe(false);
    }
    show(show) {
        const className = 'show';
        show ?
            this.renderer.addClass(this.host, className) :
            this.renderer.removeClass(this.host, className);
    }
    align(align) {
        if (align) {
            const className = `dropdown-menu-${align}`;
            this.renderer.addClass(this.host, className);
        }
    }
    stateToggleSubscribe(subscribe = true) {
        if (subscribe) {
            this.stateToggleSubscription = this.dropdownService.dropdownState$.subscribe((state) => {
                if ('show' in state) {
                    this.show(state.show);
                }
                if ('align' in state) {
                    this.align(state.align);
                }
            });
        }
        else {
            this.stateToggleSubscription.unsubscribe();
        }
    }
}
DropdownMenuDirective.ɵfac = function DropdownMenuDirective_Factory(t) { return new (t || DropdownMenuDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Renderer2), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.DropdownState), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.DropdownService)); };
DropdownMenuDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: DropdownMenuDirective, selectors: [["", "cDropdownMenu", ""], ["c-dropdown-menu"]], hostVars: 3, hostBindings: function DropdownMenuDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵattribute("role", ctx.role);
        ɵngcc0.ɵɵclassProp("dropdown-menu", ctx.dropdownMenu);
    } }, exportAs: ["cDropdownMenu"] });
DropdownMenuDirective.ctorParameters = () => [
    { type: Renderer2 },
    { type: ElementRef },
    { type: DropdownState },
    { type: DropdownService }
];
DropdownMenuDirective.propDecorators = {
    role: [{ type: HostBinding, args: ['attr.role',] }],
    dropdownMenu: [{ type: HostBinding, args: ['class.dropdown-menu',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DropdownMenuDirective, [{
        type: Directive,
        args: [{
                selector: '[cDropdownMenu], c-dropdown-menu',
                exportAs: 'cDropdownMenu'
            }]
    }], function () { return [{ type: ɵngcc0.Renderer2 }, { type: ɵngcc0.ElementRef }, { type: ɵngcc1.DropdownState }, { type: ɵngcc2.DropdownService }]; }, { role: [{
            type: HostBinding,
            args: ['attr.role']
        }], dropdownMenu: [{
            type: HostBinding,
            args: ['class.dropdown-menu']
        }] }); })();
