// @coreui/angular/esm2015/lib/dropdown/dropdown-header.directive.js
import { Directive, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class DropdownHeaderDirective {
    constructor() {
        this.dropdownHeader = true;
    }
}
DropdownHeaderDirective.ɵfac = function DropdownHeaderDirective_Factory(t) { return new (t || DropdownHeaderDirective)(); };
DropdownHeaderDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: DropdownHeaderDirective, selectors: [["", "cDropdownHeader", ""], ["c-dropdown-header"]], hostVars: 2, hostBindings: function DropdownHeaderDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("dropdown-header", ctx.dropdownHeader);
    } }, exportAs: ["cDropdownHeader"] });
DropdownHeaderDirective.propDecorators = {
    dropdownHeader: [{ type: HostBinding, args: ['class.dropdown-header',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DropdownHeaderDirective, [{
        type: Directive,
        args: [{
                selector: '[cDropdownHeader], c-dropdown-header',
                exportAs: 'cDropdownHeader'
            }]
    }], function () { return []; }, { dropdownHeader: [{
            type: HostBinding,
            args: ['class.dropdown-header']
        }] }); })();
