// @coreui/angular/esm2015/lib/dropdown/dropdown-divider.directive.js
import { Directive, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class DropdownDividerDirective {
    constructor() {
        this.dropdownDivider = true;
    }
}
DropdownDividerDirective.ɵfac = function DropdownDividerDirective_Factory(t) { return new (t || DropdownDividerDirective)(); };
DropdownDividerDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: DropdownDividerDirective, selectors: [["", "cDropdownDivider", ""], ["c-dropdown-divider"]], hostVars: 2, hostBindings: function DropdownDividerDirective_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("dropdown-divider", ctx.dropdownDivider);
    } }, exportAs: ["cDropdownDivider"] });
DropdownDividerDirective.propDecorators = {
    dropdownDivider: [{ type: HostBinding, args: ['class.dropdown-divider',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(DropdownDividerDirective, [{
        type: Directive,
        args: [{
                selector: '[cDropdownDivider], c-dropdown-divider',
                exportAs: 'cDropdownDivider'
            }]
    }], function () { return []; }, { dropdownDivider: [{
            type: HostBinding,
            args: ['class.dropdown-divider']
        }] }); })();
