// @coreui/angular/esm2015/lib/carousel/carousel-indicators/carousel-indicators.component.js
import { Component } from '@angular/core';
import { CarouselState } from '../carousel-state';
import { CarouselService } from '../carousel.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../carousel.service';
import * as ɵngcc2 from '../carousel-state';
import * as ɵngcc3 from '@angular/common';

const _c0 = function (a0) { return { active: a0 }; };
function CarouselIndicatorsComponent_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    const _r4 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "li", 2);
    ɵngcc0.ɵɵlistener("click", function CarouselIndicatorsComponent_ng_container_1_Template_li_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r4); const i_r2 = ctx.index; const ctx_r3 = ɵngcc0.ɵɵnextContext(); return ctx_r3.onClick(i_r2); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const i_r2 = ctx.index;
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵclassMap(ɵngcc0.ɵɵpureFunction1(2, _c0, ctx_r0.active === i_r2));
} }
export class CarouselIndicatorsComponent {
    constructor(carouselService, carouselState) {
        this.carouselService = carouselService;
        this.carouselState = carouselState;
        this.items = [];
        this.active = 0;
    }
    ngOnInit() {
        this.carouselStateSubscribe();
    }
    ngOnDestroy() {
        this.carouselStateSubscribe(false);
    }
    onClick(index) {
        if (index !== this.active) {
            const direction = index < this.active ? 'prev' : 'next';
            this.carouselState.state = { direction, activeItemIndex: index };
        }
    }
    carouselStateSubscribe(subscribe = true) {
        if (subscribe) {
            this.carouselIndexSubscription = this.carouselService.carouselIndex$.subscribe((nextIndex) => {
                var _a, _b, _c;
                this.items = (_c = (_b = (_a = this.carouselState) === null || _a === void 0 ? void 0 : _a.state) === null || _b === void 0 ? void 0 : _b.items) === null || _c === void 0 ? void 0 : _c.map(item => item.index);
                if ('active' in nextIndex) {
                    this.active = nextIndex.active;
                }
            });
        }
        else {
            this.carouselIndexSubscription.unsubscribe();
        }
    }
}
CarouselIndicatorsComponent.ɵfac = function CarouselIndicatorsComponent_Factory(t) { return new (t || CarouselIndicatorsComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CarouselService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.CarouselState)); };
CarouselIndicatorsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CarouselIndicatorsComponent, selectors: [["c-carousel-indicators"]], decls: 2, vars: 1, consts: [[1, "carousel-indicators"], [4, "ngFor", "ngForOf"], [3, "click"]], template: function CarouselIndicatorsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "ol", 0);
        ɵngcc0.ɵɵtemplate(1, CarouselIndicatorsComponent_ng_container_1_Template, 2, 4, "ng-container", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.items);
    } }, directives: [ɵngcc3.NgForOf], styles: [""] });
CarouselIndicatorsComponent.ctorParameters = () => [
    { type: CarouselService },
    { type: CarouselState }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CarouselIndicatorsComponent, [{
        type: Component,
        args: [{
                selector: 'c-carousel-indicators',
                template: "<ol class=\"carousel-indicators\">\n  <ng-container *ngFor=\"let item of items; let i=index\">\n    <li (click)=\"onClick(i)\" [class]=\"{active: active === i}\"></li>\n  </ng-container>\n</ol>\n",
                styles: [""]
            }]
    }], function () { return [{ type: ɵngcc1.CarouselService }, { type: ɵngcc2.CarouselState }]; }, null); })();
