// @coreui/angular/esm2015/lib/carousel/carousel/carousel.component.js
import { Component, EventEmitter, HostBinding, HostListener, Inject, Input, Output, } from '@angular/core';
import { CarouselState } from '../carousel-state';
import { CarouselService } from '../carousel.service';
import { CarouselConfig } from '../carousel.config';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../carousel.service';
import * as ɵngcc2 from '../carousel-state';
import * as ɵngcc3 from '../carousel.config';

const _c0 = ["*"];
export class CarouselComponent {
    constructor(config, carouselService, carouselState) {
        this.config = config;
        this.carouselService = carouselService;
        this.carouselState = carouselState;
        this.activeIndex = 0;
        this.animate = true;
        this.direction = 'next';
        this.interval = 0;
        this.itemChange = new EventEmitter();
        this.carouselClass = true;
        this.carouselSlideClass = true;
        Object.assign(this, config);
    }
    onMouseenter($event) {
        this.resetTimer();
    }
    onMouseleave($event) {
        this.setTimer();
    }
    ngOnInit() {
        this.carouselStateSubscribe();
    }
    ngOnDestroy() {
        this.carouselStateSubscribe(false);
    }
    ngAfterContentInit() {
        this.carouselState.state = { activeItemIndex: this.activeIndex, animate: this.animate };
        this.setTimer();
    }
    setTimer() {
        this.resetTimer();
        if (this.interval > 0) {
            this.timerId = setTimeout(() => {
                const nextIndex = this.carouselState.direction(this.direction);
                this.carouselState.state = { activeItemIndex: nextIndex };
            }, this.interval);
        }
    }
    resetTimer() {
        clearTimeout(this.timerId);
    }
    carouselStateSubscribe(subscribe = true) {
        if (subscribe) {
            this.carouselIndexSubscription = this.carouselService.carouselIndex$.subscribe((nextIndex) => {
                if ('active' in nextIndex) {
                    this.setTimer();
                }
            });
        }
        else {
            this.carouselIndexSubscription.unsubscribe();
        }
    }
}
CarouselComponent.ɵfac = function CarouselComponent_Factory(t) { return new (t || CarouselComponent)(ɵngcc0.ɵɵdirectiveInject(CarouselConfig), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.CarouselService), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.CarouselState)); };
CarouselComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CarouselComponent, selectors: [["c-carousel"]], hostVars: 4, hostBindings: function CarouselComponent_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("mouseenter", function CarouselComponent_mouseenter_HostBindingHandler($event) { return ctx.onMouseenter($event); })("mousedown", function CarouselComponent_mousedown_HostBindingHandler($event) { return ctx.onMouseenter($event); })("mouseleave", function CarouselComponent_mouseleave_HostBindingHandler($event) { return ctx.onMouseleave($event); })("mouseup", function CarouselComponent_mouseup_HostBindingHandler($event) { return ctx.onMouseleave($event); });
    } if (rf & 2) {
        ɵngcc0.ɵɵclassProp("carousel", ctx.carouselClass)("slide", ctx.carouselSlideClass);
    } }, inputs: { activeIndex: "activeIndex", animate: "animate", direction: "direction", interval: "interval" }, outputs: { itemChange: "itemChange" }, features: [ɵngcc0.ɵɵProvidersFeature([CarouselService, CarouselState, CarouselConfig])], ngContentSelectors: _c0, decls: 1, vars: 0, template: function CarouselComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]{display:block}"] });
CarouselComponent.ctorParameters = () => [
    { type: CarouselConfig, decorators: [{ type: Inject, args: [CarouselConfig,] }] },
    { type: CarouselService },
    { type: CarouselState }
];
CarouselComponent.propDecorators = {
    activeIndex: [{ type: Input }],
    animate: [{ type: Input }],
    direction: [{ type: Input }],
    interval: [{ type: Input }],
    itemChange: [{ type: Output }],
    carouselClass: [{ type: HostBinding, args: ['class.carousel',] }],
    carouselSlideClass: [{ type: HostBinding, args: ['class.slide',] }],
    onMouseenter: [{ type: HostListener, args: ['mouseenter', ['$event'],] }, { type: HostListener, args: ['mousedown', ['$event'],] }],
    onMouseleave: [{ type: HostListener, args: ['mouseleave', ['$event'],] }, { type: HostListener, args: ['mouseup', ['$event'],] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CarouselComponent, [{
        type: Component,
        args: [{
                selector: 'c-carousel',
                template: "<ng-content></ng-content>\n",
                providers: [CarouselService, CarouselState, CarouselConfig],
                styles: [":host{display:block}"]
            }]
    }], function () { return [{ type: ɵngcc3.CarouselConfig, decorators: [{
                type: Inject,
                args: [CarouselConfig]
            }] }, { type: ɵngcc1.CarouselService }, { type: ɵngcc2.CarouselState }]; }, { activeIndex: [{
            type: Input
        }], animate: [{
            type: Input
        }], direction: [{
            type: Input
        }], interval: [{
            type: Input
        }], itemChange: [{
            type: Output
        }], carouselClass: [{
            type: HostBinding,
            args: ['class.carousel']
        }], carouselSlideClass: [{
            type: HostBinding,
            args: ['class.slide']
        }], onMouseenter: [{
            type: HostListener,
            args: ['mouseenter', ['$event']]
        }, {
            type: HostListener,
            args: ['mousedown', ['$event']]
        }], onMouseleave: [{
            type: HostListener,
            args: ['mouseleave', ['$event']]
        }, {
            type: HostListener,
            args: ['mouseup', ['$event']]
        }] }); })();
