// @coreui/angular/esm2015/lib/form/input-group-addon/input-group-addon.component.js
import { Component, HostBinding, Input } from '@angular/core';
import { chkBoolProp } from '../../shared/check-bool-prop';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
const _c1 = "[_nghost-%COMP%]     .btn:not(:hover):not(:focus){box-shadow:none!important}";
export class InputGroupAddonComponent {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const addon = chkBoolProp(this.prepend) ? 'prepend' : 'append';
        const classes = {
            [`input-group-${addon}`]: true,
        };
        return classes;
    }
    ngOnInit() { }
}
InputGroupAddonComponent.ɵfac = function InputGroupAddonComponent_Factory(t) { return new (t || InputGroupAddonComponent)(); };
InputGroupAddonComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: InputGroupAddonComponent, selectors: [["ng-component"]], hostVars: 2, hostBindings: function InputGroupAddonComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { prepend: "prepend" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function InputGroupAddonComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [_c1] });
InputGroupAddonComponent.ctorParameters = () => [];
InputGroupAddonComponent.propDecorators = {
    prepend: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(InputGroupAddonComponent, [{
        type: Component,
        args: [{
                template: "<ng-content></ng-content>\n",
                styles: [":host ::ng-deep .btn:not(:hover):not(:focus){box-shadow:none!important}"]
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], prepend: [{
            type: Input
        }] }); })();
export class InputGroupPrependComponent extends InputGroupAddonComponent {
    constructor() {
        super(...arguments);
        this.prepend = true;
    }
}
InputGroupPrependComponent.ɵfac = function InputGroupPrependComponent_Factory(t) { return ɵInputGroupPrependComponent_BaseFactory(t || InputGroupPrependComponent); };
InputGroupPrependComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: InputGroupPrependComponent, selectors: [["c-input-group-prepend"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function InputGroupPrependComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [_c1] });
const ɵInputGroupPrependComponent_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(InputGroupPrependComponent);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(InputGroupPrependComponent, [{
        type: Component,
        args: [{
                selector: 'c-input-group-prepend',
                template: "<ng-content></ng-content>\n",
                styles: [":host ::ng-deep .btn:not(:hover):not(:focus){box-shadow:none!important}"]
            }]
    }], null, null); })();
export class InputGroupAppendComponent extends InputGroupAddonComponent {
    constructor() {
        super(...arguments);
        this.prepend = false;
    }
}
InputGroupAppendComponent.ɵfac = function InputGroupAppendComponent_Factory(t) { return ɵInputGroupAppendComponent_BaseFactory(t || InputGroupAppendComponent); };
InputGroupAppendComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: InputGroupAppendComponent, selectors: [["c-input-group-append"]], features: [ɵngcc0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function InputGroupAppendComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [_c1] });
const ɵInputGroupAppendComponent_BaseFactory = /*@__PURE__*/ ɵngcc0.ɵɵgetInheritedFactory(InputGroupAppendComponent);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(InputGroupAppendComponent, [{
        type: Component,
        args: [{
                selector: 'c-input-group-append',
                template: "<ng-content></ng-content>\n",
                styles: [":host ::ng-deep .btn:not(:hover):not(:focus){box-shadow:none!important}"]
            }]
    }], null, null); })();
