// @coreui/angular/esm2015/lib/breadcrumb/breadcrumb.component.js
import { Component, HostBinding, Input } from '@angular/core';
import { Observable } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/router';
import * as ɵngcc3 from '../shared/html-attr.directive';

function BreadcrumbComponent_ng_template_1_ng_container_0_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵelementStart(1, "a", 5);
    ɵngcc0.ɵɵi18n(2, 6);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const breadcrumb_r1 = ɵngcc0.ɵɵnextContext(2).$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("routerLink", breadcrumb_r1.url)("cHtmlAttr", breadcrumb_r1.attributes)("target", breadcrumb_r1.attributes == null ? null : breadcrumb_r1.attributes.target)("queryParams", breadcrumb_r1.linkProps == null ? null : breadcrumb_r1.linkProps.queryParams)("fragment", breadcrumb_r1.linkProps == null ? null : breadcrumb_r1.linkProps.fragment)("queryParamsHandling", breadcrumb_r1.linkProps == null ? null : breadcrumb_r1.linkProps.queryParamsHandling)("preserveFragment", breadcrumb_r1.linkProps == null ? null : breadcrumb_r1.linkProps.preserveFragment)("skipLocationChange", breadcrumb_r1.linkProps == null ? null : breadcrumb_r1.linkProps.skipLocationChange)("replaceUrl", breadcrumb_r1.linkProps == null ? null : breadcrumb_r1.linkProps.replaceUrl)("state", breadcrumb_r1.linkProps == null ? null : breadcrumb_r1.linkProps.state);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵi18nExp(breadcrumb_r1.label);
    ɵngcc0.ɵɵi18nApply(2);
} }
function BreadcrumbComponent_ng_template_1_ng_container_0_span_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r5 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r5.divider, " ");
} }
function BreadcrumbComponent_ng_template_1_ng_container_0_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 7);
    ɵngcc0.ɵɵi18n(1, 8);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const breadcrumb_r1 = ɵngcc0.ɵɵnextContext(2).$implicit;
    ɵngcc0.ɵɵproperty("cHtmlAttr", breadcrumb_r1.attributes);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵi18nExp(breadcrumb_r1.label);
    ɵngcc0.ɵɵi18nApply(1);
} }
function BreadcrumbComponent_ng_template_1_ng_container_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵtemplate(1, BreadcrumbComponent_ng_template_1_ng_container_0_div_1_Template, 3, 11, "div", 2);
    ɵngcc0.ɵɵtemplate(2, BreadcrumbComponent_ng_template_1_ng_container_0_span_2_Template, 2, 1, "span", 1);
    ɵngcc0.ɵɵtemplate(3, BreadcrumbComponent_ng_template_1_ng_container_0_span_3_Template, 2, 2, "span", 3);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const last_r2 = ɵngcc0.ɵɵnextContext().last;
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !last_r2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !last_r2 && ctx_r3.divider);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", last_r2);
} }
function BreadcrumbComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵtemplate(0, BreadcrumbComponent_ng_template_1_ng_container_0_Template, 4, 3, "ng-container", 1);
} if (rf & 2) {
    const breadcrumb_r1 = ctx.$implicit;
    const last_r2 = ctx.last;
    ɵngcc0.ɵɵproperty("ngIf", breadcrumb_r1.label && (breadcrumb_r1.url.slice(-1) == "/" || last_r2));
} }
const _c4 = ["*"];
export class BreadcrumbComponent {
    constructor() {
        this.divider = false;
        this.breadcrumbClass = true;
    }
    ngOnInit() {
    }
    ngOnChanges(changes) {
        if (changes.items) {
            this.setup();
        }
    }
    setup() {
        if (this.items && this.items.length > 0) {
            this.breadcrumbs = new Observable((observer) => {
                if (this.items) {
                    observer.next(this.items);
                }
            });
        }
    }
    ngOnDestroy() {
        this.breadcrumbs = null;
    }
}
BreadcrumbComponent.ɵfac = function BreadcrumbComponent_Factory(t) { return new (t || BreadcrumbComponent)(); };
BreadcrumbComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: BreadcrumbComponent, selectors: [["c-breadcrumb"]], hostVars: 2, hostBindings: function BreadcrumbComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("breadcrumb", ctx.breadcrumbClass);
    } }, inputs: { divider: "divider", items: "items" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c4, decls: 4, vars: 3, consts: function () { let i18n_0; if (typeof ngI18nClosureMode !== "undefined" && ngI18nClosureMode) {
        const MSG_EXTERNAL_1573340103535294293$$ESM2015_LIB_BREADCRUMB_BREADCRUMB_COMPONENT_JS____1 = goog.getMsg("{$interpolation} ", { "interpolation": "\uFFFD0\uFFFD" });
        i18n_0 = MSG_EXTERNAL_1573340103535294293$$ESM2015_LIB_BREADCRUMB_BREADCRUMB_COMPONENT_JS____1;
    }
    else {
        i18n_0 = $localize `:␟615f5863f83a316b692cb88dc36570691be4d586␟1573340103535294293:${"\uFFFD0\uFFFD"}:INTERPOLATION: `;
    } let i18n_2; if (typeof ngI18nClosureMode !== "undefined" && ngI18nClosureMode) {
        const MSG_EXTERNAL_2790673009523224325$$ESM2015_LIB_BREADCRUMB_BREADCRUMB_COMPONENT_JS____3 = goog.getMsg("{$interpolation}", { "interpolation": "\uFFFD0\uFFFD" });
        i18n_2 = MSG_EXTERNAL_2790673009523224325$$ESM2015_LIB_BREADCRUMB_BREADCRUMB_COMPONENT_JS____3;
    }
    else {
        i18n_2 = $localize `:␟8d663c4b400232e6d8c54ffd3c710c82f206cebb␟2790673009523224325:${"\uFFFD0\uFFFD"}:INTERPOLATION:`;
    } return [["ngFor", "", 3, "ngForOf"], [4, "ngIf"], ["class", "breadcrumb-item", 4, "ngIf"], ["class", "breadcrumb-item active", 3, "cHtmlAttr", 4, "ngIf"], [1, "breadcrumb-item"], [3, "routerLink", "cHtmlAttr", "target", "queryParams", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state"], i18n_0, [1, "breadcrumb-item", "active", 3, "cHtmlAttr"], i18n_2]; }, template: function BreadcrumbComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementContainerStart(0);
        ɵngcc0.ɵɵtemplate(1, BreadcrumbComponent_ng_template_1_Template, 1, 1, "ng-template", 0);
        ɵngcc0.ɵɵpipe(2, "async");
        ɵngcc0.ɵɵprojection(3);
        ɵngcc0.ɵɵelementContainerEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ɵngcc0.ɵɵpipeBind1(2, 1, ctx.breadcrumbs));
    } }, directives: [ɵngcc1.NgForOf, ɵngcc1.NgIf, ɵngcc2.RouterLinkWithHref, ɵngcc3.HtmlAttributesDirective], pipes: [ɵngcc1.AsyncPipe], styles: [""] });
BreadcrumbComponent.ctorParameters = () => [];
BreadcrumbComponent.propDecorators = {
    items: [{ type: Input }],
    divider: [{ type: Input }],
    breadcrumbClass: [{ type: HostBinding, args: ['class.breadcrumb',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BreadcrumbComponent, [{
        type: Component,
        args: [{
                selector: 'c-breadcrumb',
                template: "<ng-container>\n  <ng-template ngFor let-breadcrumb [ngForOf]=\"breadcrumbs | async\" let-last = last>\n    <ng-container *ngIf=\"breadcrumb.label && (breadcrumb.url.slice(-1) == '/' || last)\">\n      <div *ngIf=\"!last\" class=\"breadcrumb-item\">\n        <a [routerLink]=\"breadcrumb.url\"\n           [cHtmlAttr]=\"breadcrumb.attributes\"\n           [target]=\"breadcrumb.attributes?.target\"\n           [queryParams]=\"breadcrumb.linkProps?.queryParams\"\n           [fragment]=\"breadcrumb.linkProps?.fragment\"\n           [queryParamsHandling]=\"breadcrumb.linkProps?.queryParamsHandling\"\n           [preserveFragment]=\"breadcrumb.linkProps?.preserveFragment\"\n           [skipLocationChange]=\"breadcrumb.linkProps?.skipLocationChange\"\n           [replaceUrl]=\"breadcrumb.linkProps?.replaceUrl\"\n           [state]=\"breadcrumb.linkProps?.state\"\n           i18n>{{breadcrumb.label}}\n        </a>\n      </div>\n      <span *ngIf=\"!last && divider\" ngPreserveWhitespaces=\"true\"> {{divider}} </span>\n      <span class=\"breadcrumb-item active\" *ngIf=\"last\" [cHtmlAttr]=\"breadcrumb.attributes\" i18n>{{breadcrumb.label}}</span>\n    </ng-container>\n  </ng-template>\n  <ng-content></ng-content>\n</ng-container>\n",
                styles: [""]
            }]
    }], function () { return []; }, { divider: [{
            type: Input
        }], breadcrumbClass: [{
            type: HostBinding,
            args: ['class.breadcrumb']
        }], items: [{
            type: Input
        }] }); })();
