// @coreui/angular/esm2015/lib/breadcrumb/breadcrumb.directive.js
import { Directive, Host } from '@angular/core';
import { BreadcrumbService } from './breadcrumb.service';
import { BreadcrumbComponent } from './breadcrumb.component';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './breadcrumb.component';
import * as ɵngcc2 from './breadcrumb.service';
export class BreadcrumbDirective {
    constructor(breadcrumbComponent, service) {
        this.breadcrumbComponent = breadcrumbComponent;
        this.service = service;
    }
    ngOnInit() {
        this.breadcrumbComponent.breadcrumbs = this.service.breadcrumbs$;
    }
}
BreadcrumbDirective.ɵfac = function BreadcrumbDirective_Factory(t) { return new (t || BreadcrumbDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.BreadcrumbComponent, 1), ɵngcc0.ɵɵdirectiveInject(ɵngcc2.BreadcrumbService)); };
BreadcrumbDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BreadcrumbDirective, selectors: [["", "cBreadcrumbRouter", ""]], exportAs: ["cBreadcrumbRouter"] });
BreadcrumbDirective.ctorParameters = () => [
    { type: BreadcrumbComponent, decorators: [{ type: Host }] },
    { type: BreadcrumbService }
];
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BreadcrumbDirective, [{
        type: Directive,
        args: [{
                selector: '[cBreadcrumbRouter]',
                exportAs: 'cBreadcrumbRouter'
            }]
    }], function () { return [{ type: ɵngcc1.BreadcrumbComponent, decorators: [{
                type: Host
            }] }, { type: ɵngcc2.BreadcrumbService }]; }, null); })();
