// @coreui/angular/esm2015/lib/breadcrumb/breadcrumb.module.js
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { BreadcrumbService } from './breadcrumb.service';
import { BreadcrumbDirective } from './breadcrumb.directive';
import { BreadcrumbComponent } from './breadcrumb.component';
import { SharedModule } from '../shared';
import * as ɵngcc0 from '@angular/core';
export class BreadcrumbModule {
}
BreadcrumbModule.ɵfac = function BreadcrumbModule_Factory(t) { return new (t || BreadcrumbModule)(); };
BreadcrumbModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BreadcrumbModule });
BreadcrumbModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ providers: [BreadcrumbService], imports: [[CommonModule, RouterModule, SharedModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BreadcrumbModule, { declarations: function () { return [BreadcrumbComponent, BreadcrumbDirective]; }, imports: function () { return [CommonModule, RouterModule, SharedModule]; }, exports: function () { return [BreadcrumbComponent, BreadcrumbDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BreadcrumbModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, RouterModule, SharedModule],
                exports: [BreadcrumbComponent, BreadcrumbDirective],
                declarations: [BreadcrumbComponent, BreadcrumbDirective],
                providers: [BreadcrumbService]
            }]
    }], null, null); })();
