// @coreui/angular/esm2015/lib/text-mask/text-mask.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TextMaskDirective } from './text-mask.directive';
import * as ɵngcc0 from '@angular/core';
export class TextMaskModule {
    static forRoot() {
        return {
            ngModule: TextMaskModule
        };
    }
}
TextMaskModule.ɵfac = function TextMaskModule_Factory(t) { return new (t || TextMaskModule)(); };
TextMaskModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TextMaskModule });
TextMaskModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TextMaskModule, { declarations: function () { return [TextMaskDirective]; }, imports: function () { return [CommonModule]; }, exports: function () { return [TextMaskDirective]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TextMaskModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    TextMaskDirective
                ],
                exports: [
                    TextMaskDirective
                ]
            }]
    }], null, null); })();
