// @coreui/angular/esm2015/lib/form/form-text/form-text.component.js
import { Component, HostBinding, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class FormTextComponent {
    constructor() {
        this.tag = 'small';
        this.color = 'muted';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = {
            'form-text': true,
            [`${this.tag}`]: true,
            [`text-${this.color}`]: !!this.color,
        };
        return classes;
    }
}
FormTextComponent.ɵfac = function FormTextComponent_Factory(t) { return new (t || FormTextComponent)(); };
FormTextComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FormTextComponent, selectors: [["c-form-text"]], hostVars: 2, hostBindings: function FormTextComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { tag: "tag", color: "color" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function FormTextComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
FormTextComponent.ctorParameters = () => [];
FormTextComponent.propDecorators = {
    tag: [{ type: Input }],
    color: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FormTextComponent, [{
        type: Component,
        args: [{
                selector: 'c-form-text',
                template: "<ng-content></ng-content>\n",
                styles: [""]
            }]
    }], function () { return []; }, { tag: [{
            type: Input
        }], color: [{
            type: Input
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }] }); })();
