// @coreui/angular/esm2015/lib/form/form-feedback/form-feedback.component.js
import { Component, HostBinding, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class FormFeedbackComponent {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const validMode = this.tooltip ? 'tooltip' : 'feedback';
        const classes = {
            [`valid-${validMode}`]: this.valid === true,
            [`invalid-${validMode}`]: this.valid !== true,
        };
        return classes;
    }
}
FormFeedbackComponent.ɵfac = function FormFeedbackComponent_Factory(t) { return new (t || FormFeedbackComponent)(); };
FormFeedbackComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FormFeedbackComponent, selectors: [["c-form-feedback"]], hostVars: 2, hostBindings: function FormFeedbackComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { valid: "valid", tooltip: "tooltip" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function FormFeedbackComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
FormFeedbackComponent.ctorParameters = () => [];
FormFeedbackComponent.propDecorators = {
    valid: [{ type: Input }],
    tooltip: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FormFeedbackComponent, [{
        type: Component,
        args: [{
                selector: 'c-form-feedback',
                template: "<ng-content></ng-content>\n",
                styles: [""]
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], valid: [{
            type: Input
        }], tooltip: [{
            type: Input
        }] }); })();
