// @coreui/angular/esm2015/lib/spinkit/spinkit.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SpinkitComponent } from './spinkit.component';
import * as ɵngcc0 from '@angular/core';
export class SpinkitModule {
}
SpinkitModule.ɵfac = function SpinkitModule_Factory(t) { return new (t || SpinkitModule)(); };
SpinkitModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SpinkitModule });
SpinkitModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SpinkitModule, { declarations: function () { return [SpinkitComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [SpinkitComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinkitModule, [{
        type: NgModule,
        args: [{
                declarations: [SpinkitComponent],
                exports: [
                    SpinkitComponent
                ],
                imports: [
                    CommonModule
                ]
            }]
    }], null, null); })();
