// @coreui/angular/esm2015/lib/spinkit/spinkit.component.js
import { Component, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function SpinkitComponent_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0);
} }
function SpinkitComponent_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "div", 3);
    ɵngcc0.ɵɵelement(2, "div", 3);
    ɵngcc0.ɵɵelement(3, "div", 3);
    ɵngcc0.ɵɵelement(4, "div", 3);
    ɵngcc0.ɵɵelement(5, "div", 3);
    ɵngcc0.ɵɵelement(6, "div", 3);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SpinkitComponent_ng_container_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "div", 4);
    ɵngcc0.ɵɵelement(2, "div", 4);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SpinkitComponent_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "div", 5);
    ɵngcc0.ɵɵelement(2, "div", 5);
    ɵngcc0.ɵɵelement(3, "div", 5);
    ɵngcc0.ɵɵelement(4, "div", 5);
    ɵngcc0.ɵɵelement(5, "div", 5);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SpinkitComponent_ng_container_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0);
} }
function SpinkitComponent_ng_container_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "div", 6);
    ɵngcc0.ɵɵelement(2, "div", 6);
    ɵngcc0.ɵɵelement(3, "div", 6);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SpinkitComponent_ng_container_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "div", 7);
    ɵngcc0.ɵɵelement(2, "div", 7);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SpinkitComponent_ng_container_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "div", 8);
    ɵngcc0.ɵɵelement(2, "div", 8);
    ɵngcc0.ɵɵelement(3, "div", 8);
    ɵngcc0.ɵɵelement(4, "div", 8);
    ɵngcc0.ɵɵelement(5, "div", 8);
    ɵngcc0.ɵɵelement(6, "div", 8);
    ɵngcc0.ɵɵelement(7, "div", 8);
    ɵngcc0.ɵɵelement(8, "div", 8);
    ɵngcc0.ɵɵelement(9, "div", 8);
    ɵngcc0.ɵɵelement(10, "div", 8);
    ɵngcc0.ɵɵelement(11, "div", 8);
    ɵngcc0.ɵɵelement(12, "div", 8);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SpinkitComponent_ng_container_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "div", 9);
    ɵngcc0.ɵɵelement(2, "div", 9);
    ɵngcc0.ɵɵelement(3, "div", 9);
    ɵngcc0.ɵɵelement(4, "div", 9);
    ɵngcc0.ɵɵelement(5, "div", 9);
    ɵngcc0.ɵɵelement(6, "div", 9);
    ɵngcc0.ɵɵelement(7, "div", 9);
    ɵngcc0.ɵɵelement(8, "div", 9);
    ɵngcc0.ɵɵelement(9, "div", 9);
    ɵngcc0.ɵɵelement(10, "div", 9);
    ɵngcc0.ɵɵelement(11, "div", 9);
    ɵngcc0.ɵɵelement(12, "div", 9);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SpinkitComponent_ng_container_10_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "div", 10);
    ɵngcc0.ɵɵelement(2, "div", 10);
    ɵngcc0.ɵɵelement(3, "div", 10);
    ɵngcc0.ɵɵelement(4, "div", 10);
    ɵngcc0.ɵɵelement(5, "div", 10);
    ɵngcc0.ɵɵelement(6, "div", 10);
    ɵngcc0.ɵɵelement(7, "div", 10);
    ɵngcc0.ɵɵelement(8, "div", 10);
    ɵngcc0.ɵɵelement(9, "div", 10);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SpinkitComponent_ng_container_11_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "div", 11);
    ɵngcc0.ɵɵelement(2, "div", 11);
    ɵngcc0.ɵɵelement(3, "div", 11);
    ɵngcc0.ɵɵelement(4, "div", 11);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SpinkitComponent_ng_container_12_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelement(1, "div", 12);
    ɵngcc0.ɵɵelement(2, "div", 12);
    ɵngcc0.ɵɵelement(3, "div", 12);
    ɵngcc0.ɵɵelement(4, "div", 12);
    ɵngcc0.ɵɵelementContainerEnd();
} }
function SpinkitComponent_ng_container_13_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainer(0);
} }
export class SpinkitComponent {
    constructor() {
        this.center = true;
        this.name = 'plane';
    }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames({
            [`sk-${this.name}`]: true,
            'sk-center': this.center,
        });
        return classes;
    }
    ngOnInit() { }
}
SpinkitComponent.ɵfac = function SpinkitComponent_Factory(t) { return new (t || SpinkitComponent)(); };
SpinkitComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SpinkitComponent, selectors: [["c-spinkit"]], hostVars: 2, hostBindings: function SpinkitComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { center: "center", name: "name" }, decls: 14, vars: 13, consts: [[3, "ngSwitch"], [4, "ngSwitchCase"], [4, "ngSwitchDefault"], [1, "sk-chase-dot"], [1, "sk-bounce-dot"], [1, "sk-wave-rect"], [1, "sk-flow-dot"], [1, "sk-swing-dot"], [1, "sk-circle-dot"], [1, "sk-circle-fade-dot"], [1, "sk-grid-cube"], [1, "sk-fold-cube"], [1, "sk-wander-cube"]], template: function SpinkitComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementContainerStart(0, 0);
        ɵngcc0.ɵɵtemplate(1, SpinkitComponent_ng_container_1_Template, 1, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(2, SpinkitComponent_ng_container_2_Template, 7, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(3, SpinkitComponent_ng_container_3_Template, 3, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(4, SpinkitComponent_ng_container_4_Template, 6, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(5, SpinkitComponent_ng_container_5_Template, 1, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(6, SpinkitComponent_ng_container_6_Template, 4, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(7, SpinkitComponent_ng_container_7_Template, 3, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(8, SpinkitComponent_ng_container_8_Template, 13, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(9, SpinkitComponent_ng_container_9_Template, 13, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(10, SpinkitComponent_ng_container_10_Template, 10, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(11, SpinkitComponent_ng_container_11_Template, 5, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(12, SpinkitComponent_ng_container_12_Template, 5, 0, "ng-container", 1);
        ɵngcc0.ɵɵtemplate(13, SpinkitComponent_ng_container_13_Template, 1, 0, "ng-container", 2);
        ɵngcc0.ɵɵelementContainerEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngSwitch", ctx.name);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "plane");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "chase");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "bounce");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "wave");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "pulse");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "flow");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "swing");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "circle");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "circle-fade");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "grid");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "fold");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngSwitchCase", "wander");
    } }, directives: [ɵngcc1.NgSwitch, ɵngcc1.NgSwitchCase, ɵngcc1.NgSwitchDefault], styles: ["[_nghost-%COMP%]{display:block}.sk-wave[_nghost-%COMP%]{display:flex}.sk-flow[_nghost-%COMP%]{display:flex;align-items:center}"] });
SpinkitComponent.ctorParameters = () => [];
SpinkitComponent.propDecorators = {
    center: [{ type: Input }],
    name: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(SpinkitComponent, [{
        type: Component,
        args: [{
                selector: 'c-spinkit',
                template: "<ng-container [ngSwitch]=\"name\">\n  <ng-container *ngSwitchCase=\"'plane'\"></ng-container>\n  <ng-container *ngSwitchCase=\"'chase'\">\n    <div class=\"sk-chase-dot\"></div>\n    <div class=\"sk-chase-dot\"></div>\n    <div class=\"sk-chase-dot\"></div>\n    <div class=\"sk-chase-dot\"></div>\n    <div class=\"sk-chase-dot\"></div>\n    <div class=\"sk-chase-dot\"></div>\n  </ng-container>\n  <ng-container *ngSwitchCase=\"'bounce'\">\n    <div class=\"sk-bounce-dot\"></div>\n    <div class=\"sk-bounce-dot\"></div>\n  </ng-container>\n  <ng-container *ngSwitchCase=\"'wave'\">\n    <div class=\"sk-wave-rect\"></div>\n    <div class=\"sk-wave-rect\"></div>\n    <div class=\"sk-wave-rect\"></div>\n    <div class=\"sk-wave-rect\"></div>\n    <div class=\"sk-wave-rect\"></div>\n  </ng-container>\n  <ng-container *ngSwitchCase=\"'pulse'\"></ng-container>\n  <ng-container *ngSwitchCase=\"'flow'\">\n    <div class=\"sk-flow-dot\"></div>\n    <div class=\"sk-flow-dot\"></div>\n    <div class=\"sk-flow-dot\"></div>\n  </ng-container>\n  <ng-container *ngSwitchCase=\"'swing'\">\n    <div class=\"sk-swing-dot\"></div>\n    <div class=\"sk-swing-dot\"></div>\n  </ng-container>\n  <ng-container *ngSwitchCase=\"'circle'\">\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n    <div class=\"sk-circle-dot\"></div>\n  </ng-container>\n  <ng-container *ngSwitchCase=\"'circle-fade'\">\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n    <div class=\"sk-circle-fade-dot\"></div>\n  </ng-container>\n  <ng-container *ngSwitchCase=\"'grid'\">\n    <div class=\"sk-grid-cube\"></div>\n    <div class=\"sk-grid-cube\"></div>\n    <div class=\"sk-grid-cube\"></div>\n    <div class=\"sk-grid-cube\"></div>\n    <div class=\"sk-grid-cube\"></div>\n    <div class=\"sk-grid-cube\"></div>\n    <div class=\"sk-grid-cube\"></div>\n    <div class=\"sk-grid-cube\"></div>\n    <div class=\"sk-grid-cube\"></div>\n  </ng-container>\n  <ng-container *ngSwitchCase=\"'fold'\">\n    <div class=\"sk-fold-cube\"></div>\n    <div class=\"sk-fold-cube\"></div>\n    <div class=\"sk-fold-cube\"></div>\n    <div class=\"sk-fold-cube\"></div>\n  </ng-container>\n  <ng-container *ngSwitchCase=\"'wander'\">\n    <div class=\"sk-wander-cube\"></div>\n    <div class=\"sk-wander-cube\"></div>\n    <div class=\"sk-wander-cube\"></div>\n    <div class=\"sk-wander-cube\"></div>\n  </ng-container>\n  <ng-container *ngSwitchDefault></ng-container>\n</ng-container>\n",
                styles: [":host{display:block}:host.sk-wave{display:flex}:host.sk-flow{display:flex;align-items:center}"]
            }]
    }], function () { return []; }, { center: [{
            type: Input
        }], name: [{
            type: Input
        }], getClasses: [{
            type: HostBinding,
            args: ['class']
        }] }); })();
