// @coreui/angular/esm2015/lib/callout/callout.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CalloutComponent } from './callout.component';
import * as ɵngcc0 from '@angular/core';
export class CalloutModule {
}
CalloutModule.ɵfac = function CalloutModule_Factory(t) { return new (t || CalloutModule)(); };
CalloutModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CalloutModule });
CalloutModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CalloutModule, { declarations: function () { return [CalloutComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CalloutComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CalloutModule, [{
        type: NgModule,
        args: [{
                declarations: [CalloutComponent],
                exports: [CalloutComponent],
                imports: [
                    CommonModule
                ]
            }]
    }], null, null); })();
