// @coreui/angular/esm2015/lib/callout/callout.component.js
import { Component, HostBinding, Input } from '@angular/core';
import classNames from 'classnames';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class CalloutComponent {
    constructor() { }
    get getClasses() {
        return this.hostClasses;
    }
    get hostClasses() {
        const classes = classNames({
            'c-callout': true,
            [`c-callout-${this.color}`]: !!this.color
        });
        return classes;
    }
}
CalloutComponent.ɵfac = function CalloutComponent_Factory(t) { return new (t || CalloutComponent)(); };
CalloutComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CalloutComponent, selectors: [["c-callout"], ["", "c-callout", ""]], hostVars: 2, hostBindings: function CalloutComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.getClasses);
    } }, inputs: { color: "color" }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function CalloutComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: ["[_nghost-%COMP%]{display:block}"] });
CalloutComponent.ctorParameters = () => [];
CalloutComponent.propDecorators = {
    color: [{ type: Input }],
    getClasses: [{ type: HostBinding, args: ['class',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(CalloutComponent, [{
        type: Component,
        args: [{
                selector: 'c-callout, [c-callout]',
                template: `<ng-content></ng-content>`,
                styles: [":host{display:block}"]
            }]
    }], function () { return []; }, { getClasses: [{
            type: HostBinding,
            args: ['class']
        }], color: [{
            type: Input
        }] }); })();
