// @fullcalendar/angular/esm2015/lib/fullcalendar.module.js
import { NgModule } from '@angular/core';
import { globalPlugins } from '@fullcalendar/core';
import { FullCalendarComponent } from './fullcalendar.component';
import * as ɵngcc0 from '@angular/core';
export class FullCalendarModule {
    static registerPlugins(plugins) {
        globalPlugins.push(...plugins);
    }
}
FullCalendarModule.ɵfac = function FullCalendarModule_Factory(t) { return new (t || FullCalendarModule)(); };
FullCalendarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FullCalendarModule });
FullCalendarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FullCalendarModule, { declarations: function () { return [FullCalendarComponent]; }, exports: function () { return [FullCalendarComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(FullCalendarModule, [{
        type: NgModule,
        args: [{
                declarations: [FullCalendarComponent],
                imports: [],
                exports: [FullCalendarComponent]
            }]
    }], null, null); })();
