// @coreui/angular/esm2015/lib/shared/out-click.directive.js
import { Directive, ElementRef, HostListener, } from '@angular/core';
import { OutClickService } from './out-click.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './out-click.service';
export class OutClickDirective {
    constructor(elementRef, outClickService) {
        this.elementRef = elementRef;
        this.outClickService = outClickService;
    }
    onDocumentClick($event) {
        const targetElement = $event.target;
        if (targetElement && !this.elementRef.nativeElement.contains(targetElement)) {
            this.outClickService.onClick({ event: $event });
        }
    }
    ngOnInit() {
        this.outClickSubscription = this.outClickService.outClick$.subscribe((message) => {
        });
    }
    ngOnDestroy() {
        this.outClickSubscription.unsubscribe();
    }
}
OutClickDirective.ɵfac = function OutClickDirective_Factory(t) { return new (t || OutClickDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.OutClickService)); };
OutClickDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: OutClickDirective, selectors: [["", "cOutClick", ""]], hostBindings: function OutClickDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function OutClickDirective_click_HostBindingHandler($event) { return ctx.onDocumentClick($event); }, false, ɵngcc0.ɵɵresolveDocument);
    } }, exportAs: ["cOutClick"] });
OutClickDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: OutClickService }
];
OutClickDirective.propDecorators = {
    onDocumentClick: [{ type: HostListener, args: ['document:click', ['$event'],] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(OutClickDirective, [{
        type: Directive,
        args: [{
                selector: '[cOutClick]',
                exportAs: 'cOutClick'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc1.OutClickService }]; }, { onDocumentClick: [{
            type: HostListener,
            args: ['document:click', ['$event']]
        }] }); })();
