// @coreui/angular/esm2015/lib/tabset/tabset.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TabsetComponent } from './tabset.component';
import { TabListComponent } from './tab-list/tab-list.component';
import { TabComponent } from './tab/tab.component';
import { TabContentComponent } from './tab-content/tab-content.component';
import { TabPaneComponent } from './tab-pane/tab-pane.component';
import { TabsetService } from './tabset.service';
import * as ɵngcc0 from '@angular/core';
export class TabsetModule {
}
TabsetModule.ɵfac = function TabsetModule_Factory(t) { return new (t || TabsetModule)(); };
TabsetModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TabsetModule });
TabsetModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ providers: [
        TabsetService
    ], imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TabsetModule, { declarations: function () { return [TabsetComponent, TabListComponent, TabComponent, TabContentComponent, TabPaneComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [TabsetComponent,
        TabListComponent,
        TabComponent,
        TabContentComponent,
        TabPaneComponent]; } }); })();
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(TabsetModule, [{
        type: NgModule,
        args: [{
                declarations: [TabsetComponent, TabListComponent, TabComponent, TabContentComponent, TabPaneComponent],
                exports: [
                    TabsetComponent,
                    TabListComponent,
                    TabComponent,
                    TabContentComponent,
                    TabPaneComponent,
                ],
                imports: [
                    CommonModule
                ],
                providers: [
                    TabsetService
                ]
            }]
    }], null, null); })();
