// @coreui/angular/esm2015/lib/navbar/navbar-text/navbar-text.component.js
import { Component, HostBinding } from '@angular/core';
import * as ɵngcc0 from '@angular/core';

const _c0 = ["*"];
export class NavbarTextComponent {
    constructor() {
        this.navbarTextClass = true;
    }
    ngOnInit() { }
}
NavbarTextComponent.ɵfac = function NavbarTextComponent_Factory(t) { return new (t || NavbarTextComponent)(); };
NavbarTextComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NavbarTextComponent, selectors: [["c-navbar-text"]], hostVars: 2, hostBindings: function NavbarTextComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵclassProp("navbar-text", ctx.navbarTextClass);
    } }, ngContentSelectors: _c0, decls: 1, vars: 0, template: function NavbarTextComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, encapsulation: 2 });
NavbarTextComponent.ctorParameters = () => [];
NavbarTextComponent.propDecorators = {
    navbarTextClass: [{ type: HostBinding, args: ['class.navbar-text',] }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(NavbarTextComponent, [{
        type: Component,
        args: [{
                selector: 'c-navbar-text',
                template: "<ng-content></ng-content>\n"
            }]
    }], function () { return []; }, { navbarTextClass: [{
            type: HostBinding,
            args: ['class.navbar-text']
        }] }); })();
