// @coreui/angular/esm2015/lib/widget/widget-simple/widget-simple.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function WidgetSimpleComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 4);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.header, " ");
} }
function WidgetSimpleComponent_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 5);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r1.text, " ");
} }
const _c0 = ["*"];
export class WidgetSimpleComponent {
    constructor() { }
    ngOnInit() { }
}
WidgetSimpleComponent.ɵfac = function WidgetSimpleComponent_Factory(t) { return new (t || WidgetSimpleComponent)(); };
WidgetSimpleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: WidgetSimpleComponent, selectors: [["c-widget-simple"]], inputs: { header: "header", text: "text" }, ngContentSelectors: _c0, decls: 5, vars: 2, consts: [[1, "card"], [1, "card-body", "text-center"], ["class", "text-muted small text-uppercase font-weight-bold", 4, "ngIf"], ["class", "h2 py-3", 4, "ngIf"], [1, "text-muted", "small", "text-uppercase", "font-weight-bold"], [1, "h2", "py-3"]], template: function WidgetSimpleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵtemplate(2, WidgetSimpleComponent_div_2_Template, 2, 1, "div", 2);
        ɵngcc0.ɵɵtemplate(3, WidgetSimpleComponent_div_3_Template, 2, 1, "div", 3);
        ɵngcc0.ɵɵprojection(4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", !!ctx.header);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !!ctx.text);
    } }, directives: [ɵngcc1.NgIf], styles: [""] });
WidgetSimpleComponent.ctorParameters = () => [];
WidgetSimpleComponent.propDecorators = {
    header: [{ type: Input }],
    text: [{ type: Input }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(WidgetSimpleComponent, [{
        type: Component,
        args: [{
                selector: 'c-widget-simple',
                template: "<div class=\"card\">\n  <div class=\"card-body text-center\">\n    <div *ngIf=\"!!header\" class=\"text-muted small text-uppercase font-weight-bold\">\n      {{header}}\n    </div>\n    <div *ngIf=\"!!text\" class=\"h2 py-3\">\n      {{text}}\n    </div>\n    <ng-content></ng-content>\n  </div>\n</div>\n",
                styles: [""]
            }]
    }], function () { return []; }, { header: [{
            type: Input
        }], text: [{
            type: Input
        }] }); })();
